{
  "cells": [
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "CLLeQw9Mxab4"
      },
      "source": [
        "## –°—Å—ã–ª–∫–∞ –Ω–∞ –¥–∏—Å–∫ —Å –∫–∞—Ä—Ç–∏—Ä–æ–≤–∞–Ω–Ω—ã–º–∏ —á—Ç–µ–Ω–∏—è–º–∏"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "iX7NcoC_CLY0"
      },
      "source": [
        "\n",
        "| | SRA accession  | Stage    |\n",
        "|---:|:-------------|:-----------|\n",
        "| 1 | SRR5836473  | 8 cell       |\n",
        "| 2 | SRR3824222 | Epiblast    |\n",
        "| 3 | SRR5836475  | ICM       |\n"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "RQAK_lggxsXi"
      },
      "source": [
        "[Bam-—Ñ–∞–π–ª—ã –∏ –æ—Ç—á–µ—Ç—ã](https://drive.google.com/drive/folders/1hGzGjuUZNLIeUsCdSxbEzgthcV_BaqMw?usp=sharing)"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "5l0a7C6JHe40"
      },
      "source": [
        "# –£—Å—Ç–∞–Ω–æ–≤–∫–∞ FastQC"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "7NGOIflnTzGi"
      },
      "source": [
        "–ü–µ—Ä–µ–∑–∞–ø—É—Å–∫–∞–ª –Ω–æ—É—Ç–±—É–∫, –æ—Ç—á–µ—Ç—ã –µ—Å—Ç—å –≤ readme"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "esYQuHKO8nAg",
        "outputId": "dce3638e-c622-4a08-925a-8f6b5d8f7454"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "--2024-02-11 13:00:54--  https://www.bioinformatics.babraham.ac.uk/projects/fastqc/fastqc_v0.11.9.zip\n",
            "Resolving www.bioinformatics.babraham.ac.uk (www.bioinformatics.babraham.ac.uk)... 149.155.133.4\n",
            "Connecting to www.bioinformatics.babraham.ac.uk (www.bioinformatics.babraham.ac.uk)|149.155.133.4|:443... connected.\n",
            "HTTP request sent, awaiting response... 200 OK\n",
            "Length: 10249221 (9.8M) [application/zip]\n",
            "Saving to: ‚Äòfastqc_v0.11.9.zip‚Äô\n",
            "\n",
            "fastqc_v0.11.9.zip  100%[===================>]   9.77M  4.39MB/s    in 2.2s    \n",
            "\n",
            "2024-02-11 13:00:58 (4.39 MB/s) - ‚Äòfastqc_v0.11.9.zip‚Äô saved [10249221/10249221]\n",
            "\n",
            "Archive:  fastqc_v0.11.9.zip\n",
            "caution: filename not matched:  -y\n",
            "chmod: cannot access 'FastQC/fastqc': No such file or directory\n"
          ]
        }
      ],
      "source": [
        "!wget https://www.bioinformatics.babraham.ac.uk/projects/fastqc/fastqc_v0.11.9.zip\n",
        "!unzip fastqc_v0.11.9.zip -y\n",
        "!chmod a+x FastQC/fastqc"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "_zEx3USYPKoR"
      },
      "source": [
        "–ú–æ–∂–Ω–æ –ø–æ–ª—É—á–∞—Ç—å –¥–∞–Ω–Ω—ã–µ —Å–µ–∫–≤–µ–Ω–∏—Ä–æ–≤–∏—è —Å ENA. –ù–∞–ø—Ä–∏–º–µ—Ä, https://www.ebi.ac.uk/ena/browser/view/SRR5836473"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "background_save": true,
          "base_uri": "https://localhost:8080/"
        },
        "id": "jNHisprKBQm0",
        "outputId": "15016992-d2f6-44a7-95b9-9aaa2b35cc6e"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "--2024-02-11 13:01:03--  ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR583/003/SRR5836473/SRR5836473_1.fastq.gz\n",
            "           => ‚ÄòSRR5836473_1_latest.fastq.gz‚Äô\n",
            "Resolving ftp.sra.ebi.ac.uk (ftp.sra.ebi.ac.uk)... 193.62.193.165\n",
            "Connecting to ftp.sra.ebi.ac.uk (ftp.sra.ebi.ac.uk)|193.62.193.165|:21... connected.\n",
            "Logging in as anonymous ... Logged in!\n",
            "==> SYST ... done.    ==> PWD ... done.\n",
            "==> TYPE I ... done.  ==> CWD (1) /vol1/fastq/SRR583/003/SRR5836473 ... done.\n",
            "==> SIZE SRR5836473_1.fastq.gz ... 6856386252\n",
            "==> PASV ... done.    ==> RETR SRR5836473_1.fastq.gz ... done.\n",
            "Length: 6856386252 (6.4G) (unauthoritative)\n",
            "\n",
            "SRR5836473_1.fastq.  81%[===============>    ]   5.22G  1.09MB/s    eta 13m 22s"
          ]
        }
      ],
      "source": [
        "! wget -O SRR5836473_1_latest.fastq.gz ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR583/003/SRR5836473/SRR5836473_1.fastq.gz"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "tHiJQhQPBoft",
        "outputId": "a9d49e9f-a09e-4f7a-d111-7b0f78b500d0"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "--2024-02-10 22:04:48--  ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR583/003/SRR5836473/SRR5836473_2.fastq.gz\n",
            "           => ‚ÄòSRR5836473_2_new_2.fastq.gz‚Äô\n",
            "Resolving ftp.sra.ebi.ac.uk (ftp.sra.ebi.ac.uk)... 193.62.193.165\n",
            "Connecting to ftp.sra.ebi.ac.uk (ftp.sra.ebi.ac.uk)|193.62.193.165|:21... connected.\n",
            "Logging in as anonymous ... Logged in!\n",
            "==> SYST ... done.    ==> PWD ... done.\n",
            "==> TYPE I ... done.  ==> CWD (1) /vol1/fastq/SRR583/003/SRR5836473 ... done.\n",
            "==> SIZE SRR5836473_2.fastq.gz ... 8127008133\n",
            "==> PASV ... done.    ==> RETR SRR5836473_2.fastq.gz ... done.\n",
            "Length: 8127008133 (7.6G) (unauthoritative)\n",
            "\n",
            "SRR5836473_2.fastq.   0%[                    ]   5.13M   229KB/s    eta 8h 51m ^C\n"
          ]
        }
      ],
      "source": [
        "! wget -O SRR5836473_2_latest.fastq.gz ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR583/003/SRR5836473/SRR5836473_2.fastq.gz"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "PrmtffYs_6XP",
        "outputId": "9cc3ed4a-fb3a-452b-96f3-ef25382dfd82"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "\n",
            "gzip: SRR5836473_1.fastq.gz: unexpected end of file\n"
          ]
        }
      ],
      "source": [
        "! gzip -d SRR5836473_1_latest.fastq.gz"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "wKAegpNQUGXK"
      },
      "outputs": [],
      "source": [
        "! gzip -d SRR5836473_2_latest.fastq.gz"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "XV5AgySn__hP",
        "outputId": "d8569798-835a-4517-ac81-783f10e506ae"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "Skipping 'SRR5836473_1.fastq' which didn't exist, or couldn't be read\n"
          ]
        }
      ],
      "source": [
        "!./FastQC/fastqc SRR5836473_1_latest.fastq"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "3muIY0-IUKPM"
      },
      "outputs": [],
      "source": [
        "!./FastQC/fastqc SRR5836473_2_latest.fastq"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "Ugk3bV0LI5El"
      },
      "source": [
        "# –£—Å—Ç–∞–Ω–æ–≤–∫–∞ Bismark —Å –∑–∞–≤–∏—Å–∏–º–æ—Å—Ç—è–º–∏ (+bowtie2, samtools)"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "uSOZLyn5n5Nl",
        "outputId": "cf4e6e96-a69d-45de-985c-814db669a54c"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "\u001b[33mWARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv\u001b[0m\u001b[33m\n",
            "\u001b[0m‚ú®üç∞‚ú® Everything looks OK!\n"
          ]
        }
      ],
      "source": [
        "! pip install -q condacolab\n",
        "import condacolab\n",
        "condacolab.install()"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "npb--DNtn52o",
        "outputId": "f5309d98-f803-4f89-8d74-f38e3f9cb35b"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "‚ú®üç∞‚ú® Everything looks OK!\n"
          ]
        }
      ],
      "source": [
        "import condacolab\n",
        "condacolab.check()"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "MfBSN4YuoCL6",
        "outputId": "3d6db822-8acc-480a-a3c1-06adc4a258fe"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "Channels:\n",
            " - bioconda\n",
            " - conda-forge\n",
            "Platform: linux-64\n",
            "Collecting package metadata (repodata.json): - \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\bdone\n",
            "Solving environment: | \b\b/ \b\b- \b\bdone\n",
            "\n",
            "# All requested packages already installed.\n",
            "\n"
          ]
        }
      ],
      "source": [
        "! conda install -c bioconda bismark"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "13VM_exnoHEl",
        "outputId": "b113f8d4-670e-4606-e95a-3c32ad8e6d1e"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "\n",
            "Usage: samtools view [options] <in.bam>|<in.sam>|<in.cram> [region ...]\n",
            "\n",
            "Output options:\n",
            "  -b, --bam                  Output BAM\n",
            "  -C, --cram                 Output CRAM (requires -T)\n",
            "  -1, --fast                 Use fast BAM compression (and default to --bam)\n",
            "  -u, --uncompressed         Uncompressed BAM output (and default to --bam)\n",
            "  -h, --with-header          Include header in SAM output\n",
            "  -H, --header-only          Print SAM header only (no alignments)\n",
            "      --no-header            Print SAM alignment records only [default]\n",
            "  -c, --count                Print only the count of matching records\n",
            "  -o, --output FILE          Write output to FILE [standard output]\n",
            "  -U, --unoutput FILE, --output-unselected FILE\n",
            "                             Output reads not selected by filters to FILE\n",
            "  -p, --unmap                Set flag to UNMAP on reads not selected\n",
            "                             then write to output file.\n",
            "  -P, --fetch-pairs          Retrieve complete pairs even when outside of region\n",
            "Input options:\n",
            "  -t, --fai-reference FILE   FILE listing reference names and lengths\n",
            "  -M, --use-index            Use index and multi-region iterator for regions\n",
            "      --region[s]-file FILE  Use index to include only reads overlapping FILE\n",
            "  -X, --customized-index     Expect extra index file argument after <in.bam>\n",
            "\n",
            "Filtering options (Only include in output reads that...):\n",
            "  -L, --target[s]-file FILE  ...overlap (BED) regions in FILE\n",
            "  -N, --qname-file [^]FILE   ...whose read name is listed in FILE (\"^\" negates)\n",
            "  -r, --read-group STR       ...are in read group STR\n",
            "  -R, --read-group-file [^]FILE\n",
            "                             ...are in a read group listed in FILE\n",
            "  -d, --tag STR1[:STR2]      ...have a tag STR1 (with associated value STR2)\n",
            "  -D, --tag-file STR:FILE    ...have a tag STR whose value is listed in FILE\n",
            "  -q, --min-MQ INT           ...have mapping quality >= INT\n",
            "  -l, --library STR          ...are in library STR\n",
            "  -m, --min-qlen INT         ...cover >= INT query bases (as measured via CIGAR)\n",
            "  -e, --expr STR             ...match the filter expression STR\n",
            "  -f, --require-flags FLAG   ...have all of the FLAGs present\n",
            "  -F, --excl[ude]-flags FLAG ...have none of the FLAGs present\n",
            "      --rf, --incl-flags, --include-flags FLAG\n",
            "                             ...have some of the FLAGs present\n",
            "  -G FLAG                    EXCLUDE reads with all of the FLAGs present\n",
            "      --subsample FLOAT      Keep only FLOAT fraction of templates/read pairs\n",
            "      --subsample-seed INT   Influence WHICH reads are kept in subsampling [0]\n",
            "  -s INT.FRAC                Same as --subsample 0.FRAC --subsample-seed INT\n",
            "\n",
            "Processing options:\n",
            "      --add-flags FLAG       Add FLAGs to reads\n",
            "      --remove-flags FLAG    Remove FLAGs from reads\n",
            "  -x, --remove-tag STR\n",
            "               Comma-separated read tags to strip (repeatable) [null]\n",
            "      --keep-tag STR\n",
            "               Comma-separated read tags to preserve (repeatable) [null].\n",
            "               Equivalent to \"-x ^STR\"\n",
            "  -B, --remove-B             Collapse the backward CIGAR operation\n",
            "  -z, --sanitize FLAGS       Perform sanitity checking and fixing on records.\n",
            "                             FLAGS is comma separated (see manual). [off]\n",
            "\n",
            "General options:\n",
            "  -?, --help   Print long help, including note about region specification\n",
            "  -S           Ignored (input format is auto-detected)\n",
            "      --no-PG  Do not add a PG line\n",
            "      --input-fmt-option OPT[=VAL]\n",
            "               Specify a single input file format option in the form\n",
            "               of OPTION or OPTION=VALUE\n",
            "  -O, --output-fmt FORMAT[,OPT[=VAL]]...\n",
            "               Specify output format (SAM, BAM, CRAM)\n",
            "      --output-fmt-option OPT[=VAL]\n",
            "               Specify a single output file format option in the form\n",
            "               of OPTION or OPTION=VALUE\n",
            "  -T, --reference FILE\n",
            "               Reference sequence FASTA FILE [null]\n",
            "  -@, --threads INT\n",
            "               Number of additional threads to use [0]\n",
            "      --write-index\n",
            "               Automatically index the output files [off]\n",
            "      --verbosity INT\n",
            "               Set level of verbosity\n",
            "\n"
          ]
        }
      ],
      "source": [
        "! samtools view"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "XVeS3m8fWWNR"
      },
      "source": [
        "1. –°–∫–∞—á–∏–≤–∞–µ–º —á—Ç–µ–Ω–∏—è –∏–∑ SRA —Å –ø–æ–º–æ—â—å—é SRAtoolkit –∏–ª–∏ –ª—é–±—ã–º –¥—Ä—É–≥–∏–º —Å–ø–æ—Å–æ–±–æ–º\n",
        "\n",
        "2. –ò–Ω–¥–µ–∫—Å–∏—Ä—É–µ–º –≥–µ–Ω–æ–º\n",
        "\n",
        "3. –°—Ç—Ä–æ–∏–º –≤—ã—Ä–∞–≤–Ω–∏–≤–∞–Ω–∏–µ\n",
        "\n",
        "(–û–ø–∏—Å–∞–Ω–∏–µ —à–∞–≥–æ–≤ 2-3 —Å–º https://www.bioinformatics.babraham.ac.uk/projects/bismark/Bismark_User_Guide.pdf)\n",
        "\n",
        "#–í–Ω–∏–º–∞–Ω–∏–µ!\n",
        "\n",
        "–¢.–∫. –≤—ã—á–∏—Å–ª–∏—Ç–µ–ª—å–Ω—ã–µ –º–æ—â–Ω–æ—Å—Ç–∏ –Ω—É–∂–Ω—ã –±–æ–ª—å—à–∏–µ, —Ç–æ —ç—Ç–æ—Ç —à–∞–≥ –±—ã–ª —Å–¥–µ–ª–∞–Ω –∑–∞ –≤–∞—Å.\n",
        " –î–æ—Å—Ç–∞—Ç–æ—á–Ω–æ —Å–∫–∞—á–∞—Ç—å –≤—ã—Ä–∞–≤–Ω–∏–≤–∞–Ω–∏—è —Å –¥–∏—Å–∫–∞ https://drive.google.com/drive/folders/1hGzGjuUZNLIeUsCdSxbEzgthcV_BaqMw?usp=sharing –∏–ª–∏ —Å —Å–µ—Ä–≤–µ—Ä–∞ –º–∞–π–Ω–æ—Ä–∞)"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "4R93Nsuiu5_Y",
        "outputId": "6d407a09-e247-462a-bae5-474f690242b5"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "\u001b[?25l   \u001b[90m‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ\u001b[0m \u001b[32m0.0/147.9 kB\u001b[0m \u001b[31m?\u001b[0m eta \u001b[36m-:--:--\u001b[0m\r\u001b[2K   \u001b[90m‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ\u001b[0m \u001b[32m147.9/147.9 kB\u001b[0m \u001b[31m4.6 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25h\u001b[33mWARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv\u001b[0m\u001b[33m\n",
            "\u001b[0m"
          ]
        }
      ],
      "source": [
        "!pip install -q gdown"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "XoU1sYLxvDWH",
        "outputId": "b10b5290-d90b-4d4f-9bd6-d518c4570c4d"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "Downloading...\n",
            "From (original): https://drive.google.com/uc?id=1vZUsitBBYAYaVHbfonRuoHEK_toNZnN5\n",
            "From (redirected): https://drive.google.com/uc?id=1vZUsitBBYAYaVHbfonRuoHEK_toNZnN5&confirm=t&uuid=970ca863-2cbf-493d-aa1b-5acba86b6fc9\n",
            "To: /content/SRR3824222_1_bismark_bt2_pe.bam\n",
            "100% 2.04G/2.04G [00:19<00:00, 105MB/s] \n"
          ]
        }
      ],
      "source": [
        "url = 'https://drive.google.com/uc?id=1vZUsitBBYAYaVHbfonRuoHEK_toNZnN5'\n",
        "out = 'SRR3824222_1_bismark_bt2_pe.bam'\n",
        "!gdown {url} -O {out}"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "KLlEOrofvGAw",
        "outputId": "5352403b-8a0b-4def-ec39-5fc7fbfffdee"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "Downloading...\n",
            "From (original): https://drive.google.com/uc?id=1h8kqppLRjNoK2uwcnXfK7iQbe_sXQmor\n",
            "From (redirected): https://drive.google.com/uc?id=1h8kqppLRjNoK2uwcnXfK7iQbe_sXQmor&confirm=t&uuid=4a366256-7704-4e8a-b5ff-ba62cc7d5376\n",
            "To: /content/SRR5836473_1_bismark_bt2_pe.bam\n",
            "100% 806M/806M [00:05<00:00, 147MB/s]\n"
          ]
        }
      ],
      "source": [
        "url = 'https://drive.google.com/uc?id=1h8kqppLRjNoK2uwcnXfK7iQbe_sXQmor'\n",
        "out = 'SRR5836473_1_bismark_bt2_pe.bam'\n",
        "!gdown {url} -O {out}"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "YDDW0wnvu8Iu",
        "outputId": "a7230f29-fa44-46ce-fc0e-f05e3f543d69"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "Downloading...\n",
            "From (original): https://drive.google.com/uc?id=1yDVwkpPriz1ZMPSBkTm-5A6lNzXJJk6I\n",
            "From (redirected): https://drive.google.com/uc?id=1yDVwkpPriz1ZMPSBkTm-5A6lNzXJJk6I&confirm=t&uuid=cc746d4e-2682-4ff5-b565-d7cbb648ab14\n",
            "To: /content/SRR5836475_1_bismark_bt2_pe.bam\n",
            "100% 1.20G/1.20G [00:21<00:00, 56.2MB/s]\n"
          ]
        }
      ],
      "source": [
        "url = 'https://drive.google.com/uc?id=1yDVwkpPriz1ZMPSBkTm-5A6lNzXJJk6I'\n",
        "out = 'SRR5836475_1_bismark_bt2_pe.bam'\n",
        "!gdown {url} -O {out}"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "AxMpAQFVQFR8",
        "outputId": "128f7c63-0a68-4db9-a571-19d5a6737f21"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "total 263M\n",
            "-rw-r--r-- 1 root root  23K Feb 10 15:13 condacolab_install.log\n",
            "drwxr-xr-x 8 root root 4.0K Feb 10 15:04 FastQC\n",
            "-rw-r--r-- 1 root root 9.8M Jan 15  2020 fastqc_v0.11.9.zip\n",
            "drwxr-xr-x 1 root root 4.0K Feb  8 14:21 sample_data\n",
            "-rw-r--r-- 1 root root 5.0M Feb 10 15:20 SRR5836473_1_bismark_bt2_pe.bam\n",
            "-rw-r--r-- 1 root root 245M Feb 10 15:13 SRR5836473_1.fastq.gz\n",
            "-rw-r--r-- 1 root root 896K Feb 10 15:13 SRR5836473_2.fastq.gz\n",
            "-rw-r--r-- 1 root root 2.0M Feb 10 15:20 SRR5836475_1_bismark_bt2_pe.bam\n"
          ]
        }
      ],
      "source": [
        "! ls -lh"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "yMQy_LQ8xSOP",
        "outputId": "41ef1417-a79f-4f23-ff68-f8855a2a047d"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "SRR5836473.41_41_length=140\t99\tchr11\t40076664\t0\t140M\t=\t40076703\t178\tAATTTTTTTGGGGATTGTATTGAATTTGTAGATTGTTTTTGGTAAGAGAGTTATTTTTATTATATTGATTTTGTTAATTTATGAATATGGGAGATTTTTTTATTTTTTGAGATTTTTTTTAATTTTTTTTTTTAGAGATT\tKKKKKKKKKKKKKKKFKKKKKKKKKKKKKKFKKKKKKKKKFKKKKKKKKKFKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKF7<77FF\tNM:i:28\tMD:Z:6G0A9C7C9C6C4T2C0C7C9C0C2C0C3C0C5C9C3C0C2C2C6C2C8C3C2C5C1\tXM:Z:.................h.......x.........h......h.......hh.......h.........hx..hh...hh.....h.........h...hh..h..x......h..h........h...h..x.....h.\tXR:Z:CT\tXG:Z:CT\n",
            "SRR5836473.41_41_length=140\t147\tchr11\t40076703\t0\t139M\t=\t40076664\t-178\tTGGTAAGAGAGTTATTTTTATTATATTGATTTTGTTAATTTATGAATATGGGAGATTTTTTTATTTTTTGAGATTTTTTTTAATTTTTTTTTTTAGAGATTTGAAGTTTTTATTATATAGATTTTTTATTTTTTTAGTG\tKKKKKAAFFFKKKKKKKKKKKKKFFKKKKKKKKKKFAFKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKFKKKKKKKKKKKKFKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKK\tNM:i:31\tMD:Z:3C4T2C0C7C9C0C2C0C3C0C5C9C3C0C2C2C6C2C8C3C2C5C8C4C3A4C3C1C2C6T0\tXM:Z:...h.......hh.......h.........hx..hh...hh.....h.........h...hh..h..x......h..h........h...h..x.....h........h....h........h...h.h..h.......\tXR:Z:GA\tXG:Z:CT\n",
            "SRR5836473.231_231_length=140\t83\tchr11\t48401999\t0\t140M\t=\t48401992\t-147\tACAAATTACTCCTTCTAATTCATTAAAAAATTAAATTAAAATTATAATAAAAATTACATTAAATCTATAAATTACTTTTAACAAAATAACCATTTTTACAATATTAATCCTACCAATCCATAAACACAAAAAAACTTTCC\tKKKKKKKKF<KKKKKKKKKFKKKKKKKKKKFKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKFKKKKKKKKKKKKKKKK\tNM:i:34\tMD:Z:0G2G3G13G2G2G4G1G2G0G6G2G0G0G0G3G4G5G2G3G5G0G3G3G13G2G5G9G1G2T1G0G1G1T6\tXM:Z:h..x...h.............z..h..h....h.h..hh......h..hhhh...h....h.....x..h...h.....hh...h...h.............h..h.....x.........h.h....hh.h........\tXR:Z:CT\tXG:Z:GA\n",
            "SRR5836473.231_231_length=140\t163\tchr11\t48401992\t0\t139M\t=\t48401999\t147\tTAAATTTACAAATTACTCCTTCTAATTCATTAAAAAATTAAATTAAAATTATAATAAAAATTACATTAAATCTATAAATTACTTTTAACAAAATAACCATTTTTACAATATTAATCCTACCAATCCATAAACACAAAAA\tFFKFKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKAKFFKKKKKKKKKKKKKKKKFKAF<FF<FKKKKKKKKKKFKKKK<FFKAFF<FKF<FK7AF7A,7<AAF<KF\tNM:i:35\tMD:Z:1G3A1G2G3G13G2G2G4G1G2G0G6G2G0G0G0G3G4G5G2G3G5G0G3G3G13G2G5G9G1G2T1G0G1G0\tXM:Z:.h.....h..x...h.............z..h..h....h.h..hh......h..hhhh...h....h.....x..h...h.....hh...h...h.............h..h.....x.........h.h....hh.h\tXR:Z:GA\tXG:Z:GA\n",
            "SRR5836473.1421_1421_length=140\t99\tchr11\t30560802\t0\t140M\t=\t30560952\t289\tTGTTGAGGGGTATTTGGGTTTTTTTTAGTTTTCGGTTATTATAAATAAGGTTGTTATGAATATAGTGTAGTATGTGTTTTTTTTATTGGTTGGAATATTTTTTGGATATATGTTTAGATGAGGTATTGTGGGATTTTTTG\tKKKKKKKKFKFKKKKKKKFKKKKKFKKKKKAK,FAKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKFKKKKKKKKKKKKKKKKKKAKKKKKKKKKKKKKKKKKKKKKKKKKKKKKAKKKKAFFK<AFKKKKK\tNM:i:30\tMD:Z:10C2C6C3C0C2C2C0T2C9C4C2C6C6G2A6C3C3C0C8C2C2C10C0C0C3A9C5C0C1C2\tXM:Z:..........h..x......h...hx..h..x...h.........h....x..h......h................h...h...xz........h..h..x..........hhx.............z.....hh.x..\tXR:Z:CT\tXG:Z:CT\n",
            "SRR5836473.1421_1421_length=140\t147\tchr11\t30560952\t0\t139M\t=\t30560802\t-289\tGTTTAATTTTTTGAAGAATTGTTAGATTGATTTTTAGAGTGGTTGTAAAAGTTTGTAATTTTATTAATAATGGAGGAGTGTTTTTTTTTTTTTATATTTTTTTTAGTATTTGTTGTTATTTGAATTTTTGATTTTAGTT\t<<<<<<KF<KKKKKFFFKKAA<KAFKFKKKKKKKKKFFFFFFKKKFFAFAAKKKKFKKKKKKKKKKKKKKKKKKKKFKKKKKKKKKKKKKKKKFKKKKKKKKKKKKKKKKKKKKKKKKKKKKFKKKKKKKFKKKKKKFF\tNM:i:41\tMD:Z:2C0C6C3G3C2C0C3C6C0C12C3C3C3C0C0C1C0C2C14C0C1C3C1C0C1C1G0C0C1C0G0C0C2C2C2C3C1C0C12C5C0\tXM:Z:..hh......x.......x..hx...x......hx................h...h...hhh.hh..h..............hh.h...h.hh.x..hh.z.hx..h..x..x...h.hx............h.....h\tXR:Z:GA\tXG:Z:CT\n",
            "SRR5836473.2061_2061_length=140\t99\tchr11\t73270947\t0\t139M\t=\t73270962\t154\tATAAGAGATAGTTATATTAGGGTTTTGTTAGTAAAATTTTGTTGGTATTTGTAGTAGTGTTTGGGTTTGGTGGTTGTTTATGGGATGGATTTATGGGTGGGGTAGTTTTTGGATGGTTTTTTTTTTTATTTTAGTTTTG\tKKKKKKKKKKKFKKKKKKKKKKFKKKKKKKKKKKKKKFKKKKKKKKKKKKKKKKFKKFKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKFFKKKKKKKFKKKK7AFAFKKKKKF7FKKKKKKFKFAKKKKKKKKF7FFKF\tNM:i:31\tMD:Z:1C6C2C5C5C0C3C2C5C3C3C2A2C1A4A1C1A27C0C10C3C1C9C2C0C2C0C2C1C2C1C2\tXM:Z:.h......x..h.....x.....hx...x..h.....h...x...h.....h........h.............................hh..........x...h.x.........h..hh..hh..h.x..h.x..\tXR:Z:CT\tXG:Z:CT\n",
            "SRR5836473.2061_2061_length=140\t147\tchr11\t73270962\t0\t139M\t=\t73270947\t-154\tATTAGGGTTTTGTTAGTAAAATTTTGTTGGTATTTGTAGTAGTGTTTGGGTTTGGTGGTTGTTTATGGGATGGATTTATGGGTGGGGTAGTTTTTGGATGGTTTTTTTTTTTATTTTAGTTTTGAATTTTGTTTTTGTA\tA7<AAAAA77AKKKFKKKFKKKKAKF<KKKFKKFKKFKAAKKKKFKAKKKKKKKFKKKKKKKKKKKKKFKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKK\tNM:i:31\tMD:Z:2C5C0C3C2C5C3C3C2A2C1A4A1C1A27C0C10C3C1C9C2C0C2C0C2C1C2C1C4C5C1C4\tXM:Z:..x.....hx...x..h.....h...x...h.....h........h.............................hh..........x...h.x.........h..hh..hh..h.x..h.x....h.....h.x....\tXR:Z:GA\tXG:Z:CT\n",
            "SRR5836473.3331_3331_length=140\t99\tchr11\t43350802\t40\t140M\t=\t43350876\t214\tTTTTAATGTAAGTTGTATAAGGAGTATAATTAGGTTTAGGTGATGTTGAAAATAATATAATTTTATTTATAAATATTATTATTATTATGATAAAGGGAAAGGTGATTATTTTTAAAAGATTATGATGGTTTGTTGGTAAT\tKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKFKFFKKKKKKKKKF\tNM:i:29\tMD:Z:0C5C1C6C0T0C7G0C3C3C1C23C2C1C0C6C2C1C0C1C23C0C3C0C0C7C7C0C6C3\tXM:Z:h.....z.h......h.h........h...x...h.x.......................h..h.hh......h..h.hh.h.......................hh...hhh.......h.......hx......h...\tXR:Z:CT\tXG:Z:CT\n",
            "SRR5836473.3331_3331_length=140\t147\tchr11\t43350876\t40\t140M\t=\t43350802\t-214\tATTATTATTATTATGATAAAGGGAAAGGTGATTATTTTTAAAAGATTATGATGGTTTGTTGGTAATGTTATTGTTATACGTAGTATGTTTTGGTGTTATTTGGATTTTGTATTTTTTATAGTTTTTTTATTTTGTTAATT\tKFKKKKK<KKKKKFKAKFKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKFFKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKFKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKK\tNM:i:34\tMD:Z:2C1C0C1C23C0C3C0C0C7C7C0C6C4C2C2C6C2C3C0C7C2C0C3C0C1C1C4C0C11C1C0C4C3C0\tXM:Z:..h.hh.h.......................hh...hhh.......h.......hx......h....h..x..h....Z.x..h...hh.......h..xz...hh.z.h....hh...........h.hh....h...h\tXR:Z:GA\tXG:Z:CT\n"
          ]
        }
      ],
      "source": [
        "! samtools view SRR5836473_1_bismark_bt2_pe.bam | head"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "P1dyhziI8bB0",
        "outputId": "d4e3d7ce-9737-4d94-d5ac-7845200e5cad"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "SRR5836475.1_1_length=140\t83\tchr11\t27560911\t0\t139M\t=\t27560897\t-153\tTTACAAAAAATTTAAAACAAATAAAATAAAAACAAAATAAACCTTTAATAACCTTAAAAAAAATAACTACCCCTCTTTTATTTTAAAAAACCAATTCTTTAAAATACAATAAACACATTCAACTATTCCTTATCCCATA\tFKKAKAKKKKKKKKKKKKKKKKKKFKKKKKKKKKFAKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKK\tNM:i:39\tMD:Z:0G1G3G7G0G8G0G1G2G0G3G0G1G0G0G5G0G1G5G0G0G0G0G1T3G2G10G1C5G1G12G0G1G1G2G1G3G14G6G0\tXM:Z:..h...h.......hh........hh.h..hh...hh.hhh.....hh.h.....hhhhh.....h..x..........h.......h.h............hh.h.z..h.h...z..............h......h\tXR:Z:CT\tXG:Z:GA\n",
            "SRR5836475.1_1_length=140\t163\tchr11\t27560897\t0\t140M\t=\t27560911\t153\tCCTTTAATATCAATTTACAAAAAATTTAAAACAAATAAAATAAAAACAAAATAAACCTTTAATAACCTTAAAAAAAATAACTACCCCTCTTTTATTTTAAAAAACCAATTCTTTAAAATACAATAAACACATTCAACTAT\tFFKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKFKKKKKKKKKKKFKKKKKKKKKKKKAKKKKFKKKFFKKKKKFKKAAKKKAFAAFKKKF<AKKK\tNM:i:41\tMD:Z:2C2G5G0G1G1G3G7G0G8G0G1G2G0G3G0G1G0G0G5G0G1G5G0G0G0G0G1T3G2G10G1C5G1G12G0G1G1G2G1G3G9\tXM:Z:.....h.....zx...h...h.......hh........hh.h..hh...hh.hhh.....hh.h.....hhhhh.....h..x..........h.......h.h............hh.h.z..h.h...z.........\tXR:Z:GA\tXG:Z:GA\n",
            "SRR5836475.121_121_length=140\t99\tchr11\t40076651\t0\t140M\t=\t40076665\t154\tAGAATTGAGTTGGAATTTTTTTGGGGATTGTATTGAATTTGTAGATTGTTTTTGGTAAGAGAATTATTTTTATTATATTGATTTTGTTAATTTATGAATATGGGAGATTTTTTTATTTTTTGAGATTTTTTTTAATTTTT\tKKKKFKKKKAAKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKFKKKFKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKFKKKKKKKKKKKKKKKKKKKKKKKKKKFFKF\tNM:i:26\tMD:Z:19G0A9C7C9C6C4T1G0C0C7C9C0C2C0C3C0C5C9C3C0C2C2C6C2C8C1\tXM:Z:..............................h.......x.........h......h.......hh.......h.........hx..hh...hh.....h.........h...hh..h..x......h..h........h.\tXR:Z:CT\tXG:Z:CT\n",
            "SRR5836475.121_121_length=140\t147\tchr11\t40076665\t0\t140M\t=\t40076651\t-154\tATTTTTTTGGGGATTGTATTGAATTTGTAGATTGTTTTTGGTAAGAGAATTATTTTTATTATATTGATTTTGTTAATTTATGAATATGGGAGATTTTTTTATTTTTTGAGATTTTTTTTAATTTTTTTTTTTAGAGATTT\t7KFKAAK<,FAA7A<,KKKAA<7<FFFFFFAA7AAFKKKKFFKKKKKFKKKKKKKKKFKKKKFKKKFKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKK\tNM:i:29\tMD:Z:5G0A9C7C9C6C4T1G0C0C7C9C0C2C0C3C0C5C9C3C0C2C2C6C2C8C3C2C5C2\tXM:Z:................h.......x.........h......h.......hh.......h.........hx..hh...hh.....h.........h...hh..h..x......h..h........h...h..x.....h..\tXR:Z:GA\tXG:Z:CT\n",
            "SRR5836475.141_141_length=140\t99\tchr11\t56070543\t15\t140M\t=\t56070604\t200\tGTTTGTTGTAGATTGTTTTTATTATGTTTAGTTTTGGGTTTTGAATTTTTGATTTTTTTAAAATTTTTATTATGAATGGGTGTTGGATTTTGTTAAATGTTTTTTTTGTATTTAACGAGATGATTATGTGGTTTTTGTTT\tKFFKKKKKKKKKKKKKKKKKKKFKFKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKFFFKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKFKKFKFKKKKKK\tNM:i:21\tMD:Z:5C9C6C8G6C0C7C0C4C3C0C4C6C17C4C5C5C2C2C12C13C1\tXM:Z:.....x.........h......h...............hh.......hx....h...hh....h......h.................h....h.....h.....x..h..h...Z........h.............h.\tXR:Z:CT\tXG:Z:CT\n",
            "SRR5836475.141_141_length=140\t147\tchr11\t56070604\t15\t139M\t=\t56070543\t-200\tAATTTTTATTATGAATGGGTGTTGGATTTTGTTAAATGTTTTTTTTGTATTTAACGAGATGATTATGTGGTTTTTGTTTTTGAGTTTGTTTATATAATGGATTATATTGATGGATTTTCGTATATTAAATTATTTTCTT\t<FAAAFKFAKKKKKFFFF<KAA7<,FKFKKKKKFFKKKKFFFFAKFFKFAAFKKKKKKFKKKFFKKKKFKKKAKKKKFFKKKKFAKFKKKKKKKKFFKKFK<AFKKKKKKKKKKKFFFFFFKKKKKKKKKKKKKKKFF,\tNM:i:19\tMD:Z:2C6C17C4C5C5C2C2C12C13C26C24C0C2C0C0C0T0G0C0\tXM:Z:..h......h.................h....h.....h.....x..h..h...Z........h.............h..........................h.............Z..........hh..hhx..h\tXR:Z:GA\tXG:Z:CT\n",
            "SRR5836475.2801_2801_length=140\t83\tchr11\t26344345\t8\t115M1D22M\t=\t26344320\t-163\tTTTATAATTAAATTACCTCACTCAAAATAATACCCTCCAAATCCATCCATTTACCTAAAAATTTCATAAATTCATTTTTTTTAATAACTAAATAATACTCCATTATATACATATACACATTTTCTATATCCATTCCT\tKKKFFKKKKFKKKKFFKKKKKKKKKKKKKKKFKKKKKFKKKKKKKKKFFKKKKFKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKK\tNM:i:24\tMD:Z:3G1G3G0G0G12G0G2G2G7G0G11G4G0G27G2G1G2G9G1G2A2G2^C10G11\tXM:Z:...h.h...hhh............xh..h..h.......xh...........h....hh...........................h..x.h..h.........h.h.....h............x...........\tXR:Z:CT\tXG:Z:GA\n",
            "SRR5836475.2801_2801_length=140\t163\tchr11\t26344320\t8\t139M\t=\t26344345\t163\tAATAAATACATATCATATAAATTCTTTTATAATTAAATTACCTCACTCAAAATAATACCCTCAAAATCCATCCATTTACCTAAAAATTTCATAAATTCATTTTTTTTAATAACTAAATAATACTCCATTATATACATAT\tFKKKKKKKKFK<<FKFFKKKKKKKKFKKKKKKKKKKKKKKAKK<A,FFFKFKKKKKKKKKKK7FFFKKKKKKKAKKKK,FK<FKFKF,,F<AAF7,,F,AFKKKA7A7<,FKA7,,,<7FKAFAFAFAA<,AKK7FFAF\tNM:i:31\tMD:Z:1G1G1G7T0G1G1G1G7G1G3G0G0G12G0G2G2G5C1G0G11G4G0G27G2G1G2G9G1G2A2G1\tXM:Z:.x.h.h........h.h.h.h.......h.h...hhh............xh..h..h.......xh...........h....hh...........................h..x.h..h.........h.h.....h.\tXR:Z:GA\tXG:Z:GA\n",
            "SRR5836475.3001_3001_length=140\t99\tchr11\t4393068\t3\t5M1I12M2I119M\t=\t4393122\t194\tTATATAATATATATATATGGTGTATTGGTTAGTTTTGTGTTAATTTGATATAGTTGGAGTTATTATAGAGAAAGGAGTTTTAGTTGAGGAAATGTTTTTATGAGATTTAGTTGTAAGGTATTTTTTTAATTAGTGATTA\tKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKFKFKKKKKKKKKKKKKKKFKKKKKKKKKKKKKKKKKKKKKKKKFKFKKKKKKKKKKKKKFKKKKKKKKKKKKKKKKKFKK\tNM:i:29\tMD:Z:0C7C3C3C4C3C11C2C4C1C2C9C1C11C2C13C0C1C0C7C0C2C7C5C1C10C1\tXM:Z:h........h...h...u......x...h...........h..h....h.x..x.........h.x...........h..x.............hh.hh.......hx..x.......h.....h.h..........h.\tXR:Z:CT\tXG:Z:CT\n",
            "SRR5836475.3001_3001_length=140\t147\tchr11\t4393122\t3\t140M\t=\t4393068\t-194\tAGTTATTATAGAGAAAGGAGTTTTAGTTGAGGAAATGTTTTTATGAGATTTAGTTGTAAGGTATTTTTTTAATTAGTGATTAAGGGGGGAGGTTTTTTTGTGGGTGGGATTATTTTTGGGTTGGTAGTTTTGGTTTTATA\tKKKFF<7KKKKKKKKKKKKKKFKKKKKKKKKKKKKKKKKKKKKKKKKKAAAKKKKKKKKKKKKKKKKKKKKKKKKKKKKKFKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKK\tNM:i:28\tMD:Z:6C1C11C2C13C0C1C0C7C0C2C7C5C1C10C12C0C0C0C10T0G0C0C2C1C4C7C4G6\tXM:Z:......h.x...........h..x.............hh.hh.......hx..x.......h.....h.h..........h............hhhh............hh..h.x....x.......h...........\tXR:Z:GA\tXG:Z:CT\n"
          ]
        }
      ],
      "source": [
        "! samtools view SRR5836475_1_bismark_bt2_pe.bam | head"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "ximsADi78gZg",
        "outputId": "d6169274-93c1-4c3f-d6a5-a4d49f89cf96"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "SRR3824222.171_171_length=140\t83\tchr11\t57255358\t3\t140M\t=\t57255170\t-328\tATTTACACAATAAAATACTACTCAACTATTAAAAAAAATAAATTTATAAAATTCTTAAACAAATAAATAAACCTAAAAAACATCATCCTAAGTAAAATAACCCAATCACAAAAAAACTCACATAATATATACTCACTAAT\tKKKKKKKKKKKKKKKKKKKAFF,FF<FA,FA,KKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKF7,KKKKKKKKKKKKKKKKKKKF<<<F7,KKKKKKKKFKKKKKKKKKKKKKFKKKKKKKKKKKKKKFKKKKKKKKK\tNM:i:29\tMD:Z:11G0G1G9G10G3G7G6C2G0G5G0G2G0G4T0G1G0G0G9G3G1G0G15G0G5G2C5G8G2\tXM:Z:...........hh.h.........x..........h...h.......h.........hh.....hh..hh.....h.hhh.........x.H.h.hh...............hh.....z........h........x..\tXR:Z:CT\tXG:Z:GA\n",
            "SRR3824222.171_171_length=140\t163\tchr11\t57255170\t3\t139M\t=\t57255358\t328\tAAGAAAGGAACATAATACTACCGAAAAATCCAACAATACCTCTCCTAAACATATATCCAAAAAATATTCCAACCGATAAAAAAAACACATACTCAACTATATTCATAACAACCTTATTTATAATAACCAAAATCTAAAA\tFFFFKKFKKKKKKKKKKKKKKKKFKKKKKKKKKKKKKKKKFKKKKKKKKKKAKKKKKKKKKKKKKKKKKKKKKKK<<FKKAKKKKKKKFKKKFKKKKKKKFKKKKKKKKKKKKKF<F<AFKKKFKKKKFK7FAFFKKKK\tNM:i:30\tMD:Z:0G1A2T0T1G5G7A0G1G0G5G13G1G10G5G9G3G2G0G0G5G9G6G2G8G5G3G2G2G0G2\tXM:Z:x......Hh.....h........x.hh.....x.............x.h..........x.....h........Zx...h..hhh.....h.........h......h..x........h.....h...x.....xh..\tXR:Z:GA\tXG:Z:GA\n",
            "SRR3824222.311_311_length=140\t83\tchr11\t103450159\t0\t50M1D90M\t=\t103450096\t-204\tAAAATTTAATAAAAATAAAACCACAAAATACCCAAACTTATAAAACACAAAAAAACATTTCTAAAAAAAAAACTCATAACTCTAAATACCTCCAAAAAAAAACTAAAAAAAACACACACTAACAACTTAACAACACACCT\tKKKKKKKKKAKKKFAAKKFFFF<AKKKAA<KF,,KAKKKFAAKKF7,KKKKFA<AKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKFKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKFKKKKK\tNM:i:28\tMD:Z:1G1G6G5G2G6C14G0G0G6^T0G3G9G1G0G10G0T3G1G1G10G6G1G1G1G9G2G3G11\tXM:Z:.h.h......h.....h..h.....................hhh......h...h.........h.hh..........h....x.h.h..........h......h.h.h.h.........h..x...h...........\tXR:Z:CT\tXG:Z:GA\n",
            "SRR3824222.311_311_length=140\t163\tchr11\t103450096\t0\t113M1D27M\t=\t103450159\t204\tCAACACTCTTCTCAATAATACCTTAATCAAAAAAAAAATAAAAAATAAAATTAAAAACTTTTTAAAATTTAATAAAAATAAAACCACAAAATACCCAAACTTATAAAACACAAAAAAACATTTCTAAAAAAAAAACTCAT\tKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKFKKKKKKKFKKKKKFKKFKAFKKKKKKKKKKKKKKKF<FK<KKKKKKKKKKKKKKKA<,A,,<AKKF<A7<<<AK7\tNM:i:26\tMD:Z:16G7G0G4G0G2G0G6G2A0G8G8G1G6G5G2G6C14G0G0G6^T0G3G9G1G0G9\tXM:Z:................h.......hh....hh..hh......h...h........h........h.h......h.....h..h.....................hhh......h...h.........h.hh.........\tXR:Z:GA\tXG:Z:GA\n",
            "SRR3824222.1271_1271_length=140\t99\tchr11\t12139514\t0\t140M\t=\t12139579\t204\tTTATTTTTTGGTTTTGATGTTTTTTTGTATTGGGGTATATAAAGTTTGTAAGATTAAGAGGTTTTTTTTTTTAATGATGGTCGATTAGGCTATTTTTTGTTATATATGTAGTTAGAGATATGAGTTTTGGAGGTATTGGT\tKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKFKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKAFKKKKFKKAFKKKKF\tNM:i:42\tMD:Z:0C0C1C2C0C3C0C0C2G0C0A2C0C0C0C4C5C12C4C0C6C0C1C1C2C0C0C8C0T2A5C2C2C2C2C5C2C6C1C3C1C3G9\tXM:Z:hh.h..hx...hhx...h...hhhx....x.....h............h....hh......hh.h.h..hhh........x........Hh..h..x..h..h.....x..h......h.z...h.x.............\tXR:Z:CT\tXG:Z:CT\n",
            "SRR3824222.1271_1271_length=140\t147\tchr11\t12139579\t0\t139M\t=\t12139514\t-204\tTTTTTTTAATGATGGTCGATTAGGCTATTTTTTGTTATATATGTAGTTAGAGATATGAGTTTTGGAGGTATTGGTTAGTTTATATTGTTGTTTCATTTATAGGGTTGTAGATTTTTTTAGTTTTTTAGGTATTCCTCTC\tA7,AK<<AKKKFKAFFKFKKKFFFKKKKKKKKKFAFKKAFFFKKKKKAKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKFKKKKKKKKKKKKKKKKKKKKKKKKKKFKKKKKKKKKKKKKKKKKKKK\tNM:i:42\tMD:Z:1C2C0C0C8C0T2A5C2C2C2C2C5C2C6C1C3C1C3G14C11C2C0C7C2C3C0C0C0C2C2C1C0C2G3G0C1T0T0C0T0C0T0\tXM:Z:.h..hhh........x........Hh..h..x..h..h.....x..h......h.z...h.x..................h...........hH.hh.......x..x...hhhh..x..h.hh.......h...h.h.\tXR:Z:GA\tXG:Z:CT\n",
            "SRR3824222.5081_5081_length=140\t99\tchr11\t22485726\t0\t140M\t=\t22485741\t154\tAAAATAATGTATGATTTTTTTAGTTTTGGGTTATTAATTGTAATTGAGGTTGTATTTTTATTAATGGTTTTTTATGGTTTTTTATAGTGTTAAGTTTTAGTTGTTTTGCGTGATTTTTTTATATTTTTAAAATTAGTATT\tKKKFFA<FKFFKFFKKKKKKKKKKK,FFKKKKKKKKKKFFKKK,7FKKFFKKKKKKKKKFKKKFKFKFFA7FKKKKKAKFAKKAKK,<FFFKAAFFKKKAKKFAFKKKKF7AFAAKKKKKAKFKK7FKKFF,<AA7AK<A\tNM:i:53\tMD:Z:4C1C2C4C0C0C0C2C3C0C4C0C0T1C5C2C5C2C1C0C2C1C0C5C0C2C0C4C0C1C1C1C4C0C3C0C1C2C2C4T4C0C0C0C2C2G0C0C2C4C0C4C0C0\tXM:Z:....h....h....hhhh..x...hx....hh..h.....h..x.....x..h.hh..h.hh.....hh..hh....hh.h.h.x....hh...hh.x..x..h.........hhhh..h...hh..h....hx....hh\tXR:Z:CT\tXG:Z:CT\n",
            "SRR3824222.5081_5081_length=140\t147\tchr11\t22485741\t0\t139M\t=\t22485726\t-154\tTTTTTTAGTTTTGTGTTATTAATTGTAATTGAGGTTGTATTTTTATTAATGGTTTTTTATGGTTTTTTATAGTGTTAAGTTTTAGTTGTTTTGCGTGATTTTTTTATATTTTTAAAATTAGTATTATTTGGATGACTTT\t<7,KAFF7<FA,<,KAKAFFAKK<<A7<,F<A,<F7,KFAKKKAFKFKKKKFKKKKKFFF7FAFFKAKKKKKKKKK,KFKFFFAKKKKKKKKKKKKKKKKFKKKKFKKKKFKKKKKKKKKF<KKFKFKKKKKKKKKKKK\tNM:i:55\tMD:Z:0C0C0C2C3C0C2G1C0C0T1C5C2C5C2C1C0C2C1C0C5C0C2C0C4C0C1C1C1C4C0C3C0C1C2C2C4T4C0C0C0C2C2G0C0C2C4C0C4C0C1C0C3G4C0C1\tXM:Z:hhh..x...hx....hh..h.....h..x.....x..h.hh..h.hh.....hh..hh....hh.h.h.x....hh...hh.x..x..h.........hhhh..h...hh..h....hx....hh.hx.......Hhh.\tXR:Z:GA\tXG:Z:CT\n",
            "SRR3824222.5241_5241_length=140\t83\tchr11\t16526254\t3\t140M\t=\t16526231\t-163\tAAAATCCTCCATCATCATCATAAAAAATAACTTTAAATCCATACCTTACTTTTCTAATATAATAATATATCCAAAACTTACTATAATAAAAAAATTTAATTCTAATAATACCAAATAACCTTAATTTCTATTACTTCTAT\tFF7FFKKKFKKKKKKFKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKK\tNM:i:35\tMD:Z:3G17G4G1G6G6G0T11G0G1G1G2G0G1G6G0G4G4G2G0G0G1G1G3G0G4G2G2G1A2G7G0G5G2G5G1\tXM:Z:...h.................h....h.h......h......h............xh.h.h..hh.h......xh....h....h..hhh.h.h...hh....x..h..h....h.......hh.....x..h.....x.\tXR:Z:CT\tXG:Z:GA\n",
            "SRR3824222.5241_5241_length=140\t163\tchr11\t16526231\t3\t139M\t=\t16526254\t163\tATACTTTCCTATATTTCCTACTTAAAATCCTCCATCATCATCATAAAAAATAACTTTAAATCCATACCTTACTTTTCTAATATAATAATATATCCAAAACTTACTATAATAAAAAAATTTAATTCTAATAATACCAAAT\tKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKFKKKKKKKKKKKKKKKKFKKAFKKKKFKAKKKFKKFKFFAAK<FFA<AK\tNM:i:35\tMD:Z:0G1G0T6G8T6G17G4G1G6G6G0T11G0G1G1G2G0G1G6G0G4G4G2G0G0G1G1G3G0G4G2G2G1A2G1\tXM:Z:x.h.......x...............h.................h....h.h......h......h............xh.h.h..hh.h......xh....h....h..hhh.h.h...hh....x..h..h....h.\tXR:Z:GA\tXG:Z:GA\n"
          ]
        }
      ],
      "source": [
        "! samtools view SRR3824222_1_bismark_bt2_pe.bam | head"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "SrGGCxZQ2Ao3",
        "outputId": "98bac7b5-4650-460d-e5cc-3d632185ae48"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "[bam_sort_core] merging from 4 files and 1 in-memory blocks...\n"
          ]
        }
      ],
      "source": [
        "! samtools sort SRR5836473_1_bismark_bt2_pe.bam -o sorted_8cell.bam"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "GErvoOM-3Axj"
      },
      "outputs": [],
      "source": [
        "! samtools index  sorted_8cell.bam"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "FRnsbShJ9Uhh",
        "outputId": "f254e1f5-7081-439a-d8df-2186bd2fc739"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "[E::hts_open_format] Failed to open file \"sorted_icm.bam.tmp.0000.bam\" : File exists\n",
            "samtools sort: failed to create \"sorted_icm.bam.tmp.0000.bam\": File exists\n",
            "[E::hts_open_format] Failed to open file \"sorted_icm.bam.tmp.0000-001.bam\" : File exists\n",
            "samtools sort: failed to create \"sorted_icm.bam.tmp.0000-001.bam\": File exists\n",
            "[E::hts_open_format] Failed to open file \"sorted_icm.bam.tmp.0001.bam\" : File exists\n",
            "samtools sort: failed to create \"sorted_icm.bam.tmp.0001.bam\": File exists\n",
            "[E::hts_open_format] Failed to open file \"sorted_icm.bam.tmp.0002.bam\" : File exists\n",
            "samtools sort: failed to create \"sorted_icm.bam.tmp.0002.bam\": File exists\n",
            "[E::hts_open_format] Failed to open file \"sorted_icm.bam.tmp.0003.bam\" : File exists\n",
            "samtools sort: failed to create \"sorted_icm.bam.tmp.0003.bam\": File exists\n",
            "[bam_sort_core] merging from 5 files and 1 in-memory blocks...\n"
          ]
        }
      ],
      "source": [
        "! samtools sort SRR5836475_1_bismark_bt2_pe.bam -o sorted_icm.bam\n",
        "! samtools index  sorted_icm.bam"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "JdyKEkg3-Lmo",
        "outputId": "4df87e71-5a92-47fc-ebaf-eaaff78a0059"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "[bam_sort_core] merging from 9 files and 1 in-memory blocks...\n"
          ]
        }
      ],
      "source": [
        "! samtools sort SRR3824222_1_bismark_bt2_pe.bam -o sorted_epiblast.bam\n",
        "! samtools index  sorted_epiblast.bam"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "Lw--6KavBoXp"
      },
      "source": [
        "–°–∫–æ–ª—å–∫–æ —Ä–∏–¥–æ–≤ –∑–∞–∫–∞—Ä—Ç–∏—Ä–æ–≤–∞–ª–æ—Å—å –Ω–∞ —Ä–µ–≥–∏–æ–Ω 58717111-58917111 –≤ –æ–±—Ä–∞–∑—Ü–µ 8-cell?"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "b-ZhJMlax75S",
        "outputId": "0c138be0-1861-46a7-c109-2b2959d73b1e"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "551\n"
          ]
        }
      ],
      "source": [
        "! samtools view sorted_8cell.bam chr11:11347700-11367700 | cut -f1 | sort -u | wc -l"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "wv_dQzPB4-sI",
        "outputId": "5f782d18-7963-4d67-eecf-8c709644399b"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "194\n"
          ]
        }
      ],
      "source": [
        "! samtools view sorted_8cell.bam chr11:40185800-40195800 | cut -f1 | sort -u | wc -l"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "Pa19aQlQ8Pnl",
        "outputId": "e56ca2b4-d45f-40c4-8975-f7fe3f8568d5"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "797\n",
            "274\n"
          ]
        }
      ],
      "source": [
        "! samtools view sorted_icm.bam chr11:11347700-11367700 | cut -f1 | sort -u | wc -l\n",
        "! samtools view sorted_icm.bam chr11:40185800-40195800 | cut -f1 | sort -u | wc -l"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "FvxL0MkE_o5I",
        "outputId": "3b07f508-cfa8-4896-e4fd-37c255f3f262"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "1344\n",
            "565\n"
          ]
        }
      ],
      "source": [
        "! samtools view sorted_epiblast.bam chr11:11347700-11367700 | cut -f1 | sort -u | wc -l\n",
        "! samtools view sorted_epiblast.bam chr11:40185800-40195800 | cut -f1 | sort -u | wc -l"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 143
        },
        "id": "TeJCNuZVBHXz",
        "outputId": "c3e10eef-a2f6-441d-c023-ed4ce21704ab"
      },
      "outputs": [
        {
          "data": {
            "text/html": [
              "\n",
              "  <div id=\"df-796c0188-c2d9-4efd-bdc5-621c75fd11bd\" class=\"colab-df-container\">\n",
              "    <div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>–û–±—Ä–∞–∑—Ü—ã</th>\n",
              "      <th>chr11:11347700-11367700</th>\n",
              "      <th>chr11:40185800-40195800</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>0</th>\n",
              "      <td>8 Cell</td>\n",
              "      <td>551</td>\n",
              "      <td>194</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1</th>\n",
              "      <td>ICM</td>\n",
              "      <td>797</td>\n",
              "      <td>274</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2</th>\n",
              "      <td>Epiblast</td>\n",
              "      <td>1344</td>\n",
              "      <td>565</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "</div>\n",
              "    <div class=\"colab-df-buttons\">\n",
              "\n",
              "  <div class=\"colab-df-container\">\n",
              "    <button class=\"colab-df-convert\" onclick=\"convertToInteractive('df-796c0188-c2d9-4efd-bdc5-621c75fd11bd')\"\n",
              "            title=\"Convert this dataframe to an interactive table.\"\n",
              "            style=\"display:none;\">\n",
              "\n",
              "  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\" viewBox=\"0 -960 960 960\">\n",
              "    <path d=\"M120-120v-720h720v720H120Zm60-500h600v-160H180v160Zm220 220h160v-160H400v160Zm0 220h160v-160H400v160ZM180-400h160v-160H180v160Zm440 0h160v-160H620v160ZM180-180h160v-160H180v160Zm440 0h160v-160H620v160Z\"/>\n",
              "  </svg>\n",
              "    </button>\n",
              "\n",
              "  <style>\n",
              "    .colab-df-container {\n",
              "      display:flex;\n",
              "      gap: 12px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert {\n",
              "      background-color: #E8F0FE;\n",
              "      border: none;\n",
              "      border-radius: 50%;\n",
              "      cursor: pointer;\n",
              "      display: none;\n",
              "      fill: #1967D2;\n",
              "      height: 32px;\n",
              "      padding: 0 0 0 0;\n",
              "      width: 32px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert:hover {\n",
              "      background-color: #E2EBFA;\n",
              "      box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "      fill: #174EA6;\n",
              "    }\n",
              "\n",
              "    .colab-df-buttons div {\n",
              "      margin-bottom: 4px;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert {\n",
              "      background-color: #3B4455;\n",
              "      fill: #D2E3FC;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert:hover {\n",
              "      background-color: #434B5C;\n",
              "      box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n",
              "      filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n",
              "      fill: #FFFFFF;\n",
              "    }\n",
              "  </style>\n",
              "\n",
              "    <script>\n",
              "      const buttonEl =\n",
              "        document.querySelector('#df-796c0188-c2d9-4efd-bdc5-621c75fd11bd button.colab-df-convert');\n",
              "      buttonEl.style.display =\n",
              "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "\n",
              "      async function convertToInteractive(key) {\n",
              "        const element = document.querySelector('#df-796c0188-c2d9-4efd-bdc5-621c75fd11bd');\n",
              "        const dataTable =\n",
              "          await google.colab.kernel.invokeFunction('convertToInteractive',\n",
              "                                                    [key], {});\n",
              "        if (!dataTable) return;\n",
              "\n",
              "        const docLinkHtml = 'Like what you see? Visit the ' +\n",
              "          '<a target=\"_blank\" href=https://colab.research.google.com/notebooks/data_table.ipynb>data table notebook</a>'\n",
              "          + ' to learn more about interactive tables.';\n",
              "        element.innerHTML = '';\n",
              "        dataTable['output_type'] = 'display_data';\n",
              "        await google.colab.output.renderOutput(dataTable, element);\n",
              "        const docLink = document.createElement('div');\n",
              "        docLink.innerHTML = docLinkHtml;\n",
              "        element.appendChild(docLink);\n",
              "      }\n",
              "    </script>\n",
              "  </div>\n",
              "\n",
              "\n",
              "<div id=\"df-2938618b-03a6-4544-a890-156a94a5f667\">\n",
              "  <button class=\"colab-df-quickchart\" onclick=\"quickchart('df-2938618b-03a6-4544-a890-156a94a5f667')\"\n",
              "            title=\"Suggest charts\"\n",
              "            style=\"display:none;\">\n",
              "\n",
              "<svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n",
              "     width=\"24px\">\n",
              "    <g>\n",
              "        <path d=\"M19 3H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zM9 17H7v-7h2v7zm4 0h-2V7h2v10zm4 0h-2v-4h2v4z\"/>\n",
              "    </g>\n",
              "</svg>\n",
              "  </button>\n",
              "\n",
              "<style>\n",
              "  .colab-df-quickchart {\n",
              "      --bg-color: #E8F0FE;\n",
              "      --fill-color: #1967D2;\n",
              "      --hover-bg-color: #E2EBFA;\n",
              "      --hover-fill-color: #174EA6;\n",
              "      --disabled-fill-color: #AAA;\n",
              "      --disabled-bg-color: #DDD;\n",
              "  }\n",
              "\n",
              "  [theme=dark] .colab-df-quickchart {\n",
              "      --bg-color: #3B4455;\n",
              "      --fill-color: #D2E3FC;\n",
              "      --hover-bg-color: #434B5C;\n",
              "      --hover-fill-color: #FFFFFF;\n",
              "      --disabled-bg-color: #3B4455;\n",
              "      --disabled-fill-color: #666;\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart {\n",
              "    background-color: var(--bg-color);\n",
              "    border: none;\n",
              "    border-radius: 50%;\n",
              "    cursor: pointer;\n",
              "    display: none;\n",
              "    fill: var(--fill-color);\n",
              "    height: 32px;\n",
              "    padding: 0;\n",
              "    width: 32px;\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart:hover {\n",
              "    background-color: var(--hover-bg-color);\n",
              "    box-shadow: 0 1px 2px rgba(60, 64, 67, 0.3), 0 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "    fill: var(--button-hover-fill-color);\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart-complete:disabled,\n",
              "  .colab-df-quickchart-complete:disabled:hover {\n",
              "    background-color: var(--disabled-bg-color);\n",
              "    fill: var(--disabled-fill-color);\n",
              "    box-shadow: none;\n",
              "  }\n",
              "\n",
              "  .colab-df-spinner {\n",
              "    border: 2px solid var(--fill-color);\n",
              "    border-color: transparent;\n",
              "    border-bottom-color: var(--fill-color);\n",
              "    animation:\n",
              "      spin 1s steps(1) infinite;\n",
              "  }\n",
              "\n",
              "  @keyframes spin {\n",
              "    0% {\n",
              "      border-color: transparent;\n",
              "      border-bottom-color: var(--fill-color);\n",
              "      border-left-color: var(--fill-color);\n",
              "    }\n",
              "    20% {\n",
              "      border-color: transparent;\n",
              "      border-left-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "    }\n",
              "    30% {\n",
              "      border-color: transparent;\n",
              "      border-left-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "      border-right-color: var(--fill-color);\n",
              "    }\n",
              "    40% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "    }\n",
              "    60% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "    }\n",
              "    80% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "      border-bottom-color: var(--fill-color);\n",
              "    }\n",
              "    90% {\n",
              "      border-color: transparent;\n",
              "      border-bottom-color: var(--fill-color);\n",
              "    }\n",
              "  }\n",
              "</style>\n",
              "\n",
              "  <script>\n",
              "    async function quickchart(key) {\n",
              "      const quickchartButtonEl =\n",
              "        document.querySelector('#' + key + ' button');\n",
              "      quickchartButtonEl.disabled = true;  // To prevent multiple clicks.\n",
              "      quickchartButtonEl.classList.add('colab-df-spinner');\n",
              "      try {\n",
              "        const charts = await google.colab.kernel.invokeFunction(\n",
              "            'suggestCharts', [key], {});\n",
              "      } catch (error) {\n",
              "        console.error('Error during call to suggestCharts:', error);\n",
              "      }\n",
              "      quickchartButtonEl.classList.remove('colab-df-spinner');\n",
              "      quickchartButtonEl.classList.add('colab-df-quickchart-complete');\n",
              "    }\n",
              "    (() => {\n",
              "      let quickchartButtonEl =\n",
              "        document.querySelector('#df-2938618b-03a6-4544-a890-156a94a5f667 button');\n",
              "      quickchartButtonEl.style.display =\n",
              "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "    })();\n",
              "  </script>\n",
              "</div>\n",
              "    </div>\n",
              "  </div>\n"
            ],
            "text/plain": [
              "    –û–±—Ä–∞–∑—Ü—ã  chr11:11347700-11367700  chr11:40185800-40195800\n",
              "0    8 Cell                      551                      194\n",
              "1       ICM                      797                      274\n",
              "2  Epiblast                     1344                      565"
            ]
          },
          "execution_count": 78,
          "metadata": {},
          "output_type": "execute_result"
        }
      ],
      "source": [
        "import pandas as pd\n",
        "df = pd.DataFrame({ '–û–±—Ä–∞–∑—Ü—ã': ['8 Cell', 'ICM', 'Epiblast'],\n",
        "                   'chr11:11347700-11367700': [551, 797, 1344],\n",
        "                    'chr11:40185800-40195800': [194, 274, 565]})\n",
        "df"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "CliRCSUQTmV-"
      },
      "source": [
        "–î–µ–¥—É–ø–ª–∏–∫–∞—Ü–∏—è"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "ShW-Fhzq19kE",
        "outputId": "03f65afb-c43f-41b7-d439-f15b0b19d29c"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "Output filename was given as: s_8_cell\n",
            "Processing paired-end Bismark output file(s) (SAM format):\n",
            "SRR5836473_1_bismark_bt2_pe.bam\n",
            "\n",
            "\n",
            "If there are several alignments to a single position in the genome the first alignment will be chosen. Since the input files are not in any way sorted this is a near-enough random selection of reads.\n",
            "\n",
            "Checking file >>SRR5836473_1_bismark_bt2_pe.bam<< for signs of file truncation...\n",
            "\n",
            "\n",
            "\n",
            "Now testing Bismark result file SRR5836473_1_bismark_bt2_pe.bam for positional sorting (which would be bad...)\t...passed!\n",
            "Output file is: s_8_cell.deduplicated.bam\n",
            "\n",
            "skipping header line:\t@HD\tVN:1.0\tSO:unsorted\n",
            "skipping header line:\t@SQ\tSN:chr11\tLN:121973369\n",
            "skipping header line:\t@PG\tID:Bismark\tVN:v0.24.2\tCL:\"bismark --genome_folder /mnt/cache-1/for_minor/mm39_11/ -q --non_directional -parallel 10 -1 SRR3824222_1.fq,SRR5836473_1.fq,SRR5836473_1.fq -2 SRR3824222_2.fq,SRR5836473_2.fq,SRR5836473_2.fq\"\n",
            "skipping header line:\t@PG\tID:samtools\tPN:samtools\tPP:Bismark\tVN:1.19.2\tCL:/usr/local/bin/samtools view -h --threads 1 SRR5836473_1_bismark_bt2_pe.bam\n",
            "\n",
            "Total number of alignments analysed in SRR5836473_1_bismark_bt2_pe.bam:\t2858170\n",
            "Total number duplicated alignments removed:\t522512 (18.28%)\n",
            "Duplicated alignments were found at:\t230677 different position(s)\n",
            "\n",
            "Total count of deduplicated leftover sequences: 2335658 (81.72% of total)\n",
            "\n"
          ]
        }
      ],
      "source": [
        "! deduplicate_bismark  --bam  --paired  -o s_8_cell  SRR5836473_1_bismark_bt2_pe.bam"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "AK7gyG3oB4Gp",
        "outputId": "bdad8ad3-fbe9-4765-f922-cecfc1f0efa1"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "Output filename was given as: icm_cell\n",
            "Processing paired-end Bismark output file(s) (SAM format):\n",
            "SRR5836475_1_bismark_bt2_pe.bam\n",
            "\n",
            "\n",
            "If there are several alignments to a single position in the genome the first alignment will be chosen. Since the input files are not in any way sorted this is a near-enough random selection of reads.\n",
            "\n",
            "Checking file >>SRR5836475_1_bismark_bt2_pe.bam<< for signs of file truncation...\n",
            "\n",
            "\n",
            "\n",
            "Now testing Bismark result file SRR5836475_1_bismark_bt2_pe.bam for positional sorting (which would be bad...)\t...passed!\n",
            "Output file is: icm_cell.deduplicated.bam\n",
            "\n",
            "skipping header line:\t@HD\tVN:1.0\tSO:unsorted\n",
            "skipping header line:\t@SQ\tSN:chr11\tLN:121973369\n",
            "skipping header line:\t@PG\tID:Bismark\tVN:v0.24.2\tCL:\"bismark --genome_folder /mnt/cache-1/for_minor/mm39_11/ -q --non_directional -parallel 10 -1 SRR5836475_1.fq -2 SRR5836475_2.fq\"\n",
            "skipping header line:\t@PG\tID:samtools\tPN:samtools\tPP:Bismark\tVN:1.19.2\tCL:/usr/local/bin/samtools view -h --threads 1 SRR5836475_1_bismark_bt2_pe.bam\n",
            "\n",
            "Total number of alignments analysed in SRR5836475_1_bismark_bt2_pe.bam:\t4170098\n",
            "Total number duplicated alignments removed:\t378125 (9.07%)\n",
            "Duplicated alignments were found at:\t229940 different position(s)\n",
            "\n",
            "Total count of deduplicated leftover sequences: 3791973 (90.93% of total)\n",
            "\n"
          ]
        }
      ],
      "source": [
        "! deduplicate_bismark  --bam  --paired  -o icm_cell  SRR5836475_1_bismark_bt2_pe.bam"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "eJpkDeI5B4vs",
        "outputId": "502dbbd1-61f5-42f9-ed84-2f5e9939b13f"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "Output filename was given as: epiblast_cell\n",
            "Processing paired-end Bismark output file(s) (SAM format):\n",
            "SRR3824222_1_bismark_bt2_pe.bam\n",
            "\n",
            "\n",
            "If there are several alignments to a single position in the genome the first alignment will be chosen. Since the input files are not in any way sorted this is a near-enough random selection of reads.\n",
            "\n",
            "Checking file >>SRR3824222_1_bismark_bt2_pe.bam<< for signs of file truncation...\n",
            "\n",
            "\n",
            "\n",
            "Now testing Bismark result file SRR3824222_1_bismark_bt2_pe.bam for positional sorting (which would be bad...)\t...passed!\n",
            "Output file is: epiblast_cell.deduplicated.bam\n",
            "\n",
            "skipping header line:\t@HD\tVN:1.0\tSO:unsorted\n",
            "skipping header line:\t@SQ\tSN:chr11\tLN:121973369\n",
            "skipping header line:\t@PG\tID:Bismark\tVN:v0.24.2\tCL:\"bismark --genome_folder /mnt/cache-1/for_minor/mm39_11/ -q --non_directional -parallel 10 -1 SRR3824222_1.fq,SRR5836473_1.fq,SRR5836473_1.fq -2 SRR3824222_2.fq,SRR5836473_2.fq,SRR5836473_2.fq\"\n",
            "skipping header line:\t@PG\tID:samtools\tPN:samtools\tPP:Bismark\tVN:1.19.2\tCL:/usr/local/bin/samtools view -h --threads 1 SRR3824222_1_bismark_bt2_pe.bam\n",
            "\n",
            "Total number of alignments analysed in SRR3824222_1_bismark_bt2_pe.bam:\t7051731\n",
            "Total number duplicated alignments removed:\t205287 (2.91%)\n",
            "Duplicated alignments were found at:\t164998 different position(s)\n",
            "\n",
            "Total count of deduplicated leftover sequences: 6846444 (97.09% of total)\n",
            "\n"
          ]
        }
      ],
      "source": [
        "! deduplicate_bismark  --bam  --paired  -o epiblast_cell  SRR3824222_1_bismark_bt2_pe.bam"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "OkrrqUQPXCCI",
        "outputId": "d8eb39b5-35a0-4299-fffd-c8a0c795c18b"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "checksum.md5\t\t\t\tsorted_epiblast.bam.bai\n",
            "condacolab_install.log\t\t\tsorted_icm.bam\n",
            "epiblast_cell.deduplicated.bam\t\tsorted_icm.bam.bai\n",
            "epiblast_cell.deduplication_report.txt\tsorted_icm.bam.tmp.0000-001.bam\n",
            "FastQC\t\t\t\t\tsorted_icm.bam.tmp.0000.bam\n",
            "fastqc_v0.11.9.zip\t\t\tsorted_icm.bam.tmp.0001.bam\n",
            "fastqc_v0.11.9.zip.1\t\t\tsorted_icm.bam.tmp.0002.bam\n",
            "fastqc_v0.11.9.zip.2\t\t\tsorted_icm.bam.tmp.0003.bam\n",
            "icm_cell.deduplicated.bam\t\tSRR3824222_1_bismark_bt2_pe.bam\n",
            "icm_cell.deduplication_report.txt\tSRR5836473_1_bismark_bt2_pe.bam\n",
            "s_8_cell.deduplicated.bam\t\tSRR5836473_1.fastq.gz\n",
            "s_8_cell.deduplication_report.txt\tSRR5836473_1.fastq.gz.1\n",
            "sample_data\t\t\t\tSRR5836473_1.fastq.gz.2\n",
            "sorted_8cell.bam\t\t\tSRR5836473_2.fastq.gz\n",
            "sorted_8cell.bam.bai\t\t\tSRR5836473_2.fastq.gz.1\n",
            "sorted_epiblast.bam\t\t\tSRR5836475_1_bismark_bt2_pe.bam\n"
          ]
        }
      ],
      "source": [
        "! ls"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "969PlQMGwBO3"
      },
      "source": [
        "11-—é —Ö—Ä–æ–º–æ—Å–æ–º—É –º—ã—à–∏ –º–æ–∂–Ω–æ —Å–∫–∞—á–∞—Ç—å —Å UCSC (—Å–±–æ—Ä–∫–∞ mm39)"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "4NZ1s3i5bUUq",
        "outputId": "04efa33e-8bc4-4827-d52d-8777fb5b4774"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "unzip:  cannot find or open mouse_chr.zip, mouse_chr.zip.zip or mouse_chr.zip.ZIP.\n"
          ]
        }
      ],
      "source": [
        "# ! unzip mouse_chr.zip"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "sf_yqXE71QjG"
      },
      "source": [
        "–í—Ä–µ–º—è —Ä–∞–±–æ—Ç—ã: ¬±30 –º–∏–Ω"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "buKTYu3PBuA3",
        "outputId": "c955c043-7922-4a4e-8448-ee1bf10b006a"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "head: cannot open 's_8_cell.deduplicated.bedGraph' for reading: No such file or directory\n"
          ]
        }
      ],
      "source": [
        "# ! head s_8_cell.deduplicated.bedGraph"
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "–î–æ–±–∞–≤–∏–ª –ø–∞–ø–∫—É —Å —Ñ–∞–π–ª–æ–º –≤—Ä—É—á–Ω—É—é"
      ],
      "metadata": {
        "id": "O9KCU99PgQtZ"
      }
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "LKBD3cwo1wLx"
      },
      "source": [
        "–ö–æ–ª–ª–∏–Ω–≥ –º–µ—Ç–∏–ª–∏—Ä–æ–≤–∞–Ω–∏—è"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "4c5BEvKpDw5K",
        "outputId": "433c81d6-e3bb-45e6-d4fb-30b3f0077dab"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "\n",
            " *** Bismark methylation extractor version v0.24.2 ***\n",
            "\n",
            "Trying to determine the type of mapping from the SAM header line of file s_8_cell.deduplicated.bam\n",
            "Treating file(s) as paired-end data (as extracted from @PG line)\n",
            "\n",
            "Setting option '--no_overlap' since this is (normally) the right thing to do for paired-end data\n",
            "\n",
            "\n",
            "Setting core usage to single-threaded (default). Consider using --multicore <int> to speed up the extraction process.\n",
            "\n",
            "Summarising Bismark methylation extractor parameters:\n",
            "===============================================================\n",
            "Bismark paired-end SAM format specified (default)\n",
            "Number of cores to be used: 1\n",
            "Output will be written to the current directory ('/content')\n",
            "\n",
            "\n",
            "Summarising bedGraph parameters:\n",
            "===============================================================\n",
            "Generating additional output in bedGraph and coverage format\n",
            "bedGraph format:\t<Chromosome> <Start Position> <End Position> <Methylation Percentage>\n",
            "coverage format:\t<Chromosome> <Start Position> <End Position> <Methylation Percentage> <count methylated> <count non-methylated>\n",
            "\n",
            "Using a cutoff of 1 read(s) to report cytosine positions\n",
            "Reporting and sorting cytosine methylation information in CpG context only (default)\n",
            "The bedGraph UNIX sort command will use the following memory setting:\t'2G'. Temporary directory used for sorting is the output directory\n",
            "\n",
            "\n",
            "Summarising genome-wide cytosine methylation report parameters:\n",
            "===============================================================\n",
            "Generating comprehensive genome-wide cytosine report\n",
            "(output format: <Chromosome> <Position> <Strand> <count methylated> <count non-methylated>  <C-context>  <trinucleotide context> )\n",
            "Reporting cytosine methylation in CpG context only (default)\n",
            "Using 1-based genomic coordinates (default)\n",
            "Genome folder was specified as ./mouse_chr/\n",
            "\n",
            "Checking file >>s_8_cell.deduplicated.bam<< for signs of file truncation...\n",
            "\n",
            "\n",
            "Now testing Bismark result file >s_8_cell.deduplicated.bam< for positional sorting (which would be bad...)\t...passed!\n",
            "Writing result file containing methylation information for C in CpG context from the original top strand to CpG_OT_s_8_cell.deduplicated.txt.gz\n",
            "Writing result file containing methylation information for C in CpG context from the complementary to original top strand to CpG_CTOT_s_8_cell.deduplicated.txt.gz\n",
            "Writing result file containing methylation information for C in CpG context from the complementary to original bottom strand to CpG_CTOB_s_8_cell.deduplicated.txt.gz\n",
            "Writing result file containing methylation information for C in CpG context from the original bottom strand to CpG_OB_s_8_cell.deduplicated.txt.gz\n",
            "\n",
            "Writing result file containing methylation information for C in CHG context from the original top strand to CHG_OT_s_8_cell.deduplicated.txt.gz\n",
            "Writing result file containing methylation information for C in CHG context from the complementary to original top strand to CHG_CTOT_s_8_cell.deduplicated.txt.gz\n",
            "Writing result file containing methylation information for C in CHG context from the complementary to original bottom strand to CHG_CTOB_s_8_cell.deduplicated.txt.gz\n",
            "Writing result file containing methylation information for C in CHG context from the original bottom strand to CHG_OB_s_8_cell.deduplicated.txt.gz\n",
            "\n",
            "Writing result file containing methylation information for C in CHH context from the original top strand to CHH_OT_s_8_cell.deduplicated.txt.gz\n",
            "Writing result file containing methylation information for C in CHH context from the complementary to original top strand to CHH_CTOT_s_8_cell.deduplicated.txt.gz\n",
            "Writing result file containing methylation information for C in CHH context from the complementary to original bottom strand to CHH_CTOB_s_8_cell.deduplicated.txt.gz\n",
            "Writing result file containing methylation information for C in CHH context from the original bottom strand to CHH_OB_s_8_cell.deduplicated.txt.gz\n",
            "\n",
            "Now reading in Bismark result file s_8_cell.deduplicated.bam\n",
            "skipping SAM header line:\t@HD\tVN:1.0\tSO:unsorted\n",
            "skipping SAM header line:\t@SQ\tSN:chr11\tLN:121973369\n",
            "skipping SAM header line:\t@PG\tID:Bismark\tVN:v0.24.2\tCL:\"bismark --genome_folder /mnt/cache-1/for_minor/mm39_11/ -q --non_directional -parallel 10 -1 SRR3824222_1.fq,SRR5836473_1.fq,SRR5836473_1.fq -2 SRR3824222_2.fq,SRR5836473_2.fq,SRR5836473_2.fq\"\n",
            "skipping SAM header line:\t@PG\tID:samtools\tPN:samtools\tPP:Bismark\tVN:1.19.2\tCL:/usr/local/bin/samtools view -h --threads 1 SRR5836473_1_bismark_bt2_pe.bam\n",
            "skipping SAM header line:\t@PG\tID:samtools.1\tPN:samtools\tPP:samtools\tVN:1.19.2\tCL:/usr/local/bin/samtools view -bS --threads 1 -\n",
            "skipping SAM header line:\t@PG\tID:samtools.2\tPN:samtools\tPP:samtools.1\tVN:1.19.2\tCL:/usr/local/bin/samtools view -h s_8_cell.deduplicated.bam\n",
            "Processed lines: 500000\n",
            "Processed lines: 1000000\n",
            "Processed lines: 1500000\n",
            "Processed lines: 2000000\n",
            "\n",
            "Processed 2335658 lines in total\n",
            "Total number of methylation call strings processed: 4671316\n",
            "\n",
            "Final Cytosine Methylation Report\n",
            "=================================\n",
            "Total number of C's analysed:\t104748620\n",
            "\n",
            "Total methylated C's in CpG context:\t2103916\n",
            "Total methylated C's in CHG context:\t270195\n",
            "Total methylated C's in CHH context:\t1236285\n",
            "\n",
            "Total C to T conversions in CpG context:\t2729982\n",
            "Total C to T conversions in CHG context:\t22951565\n",
            "Total C to T conversions in CHH context:\t75456677\n",
            "\n",
            "C methylated in CpG context:\t43.5%\n",
            "C methylated in CHG context:\t1.2%\n",
            "C methylated in CHH context:\t1.6%\n",
            "\n",
            "\n",
            "Determining maximum read lengths for M-Bias plots\n",
            "Maximum read length of Read 1: 140\n",
            "Maximum read length of Read 2: 140\n",
            "\n",
            "Perl module GD::Graph::lines is not installed, skipping drawing M-bias plots (only writing out M-bias plot table)\n",
            "Determining maximum read lengths for M-Bias plots\n",
            "Maximum read length of Read 1: 140\n",
            "Maximum read length of Read 2: 140\n",
            "\n",
            "Perl module GD::Graph::lines is not installed, skipping drawing M-bias plots (only writing out M-bias plot table)\n",
            "Deleting unused files ...\n",
            "\n",
            "CpG_OT_s_8_cell.deduplicated.txt.gz contains data ->\tkept\n",
            "CpG_CTOT_s_8_cell.deduplicated.txt.gz contains data ->\tkept\n",
            "CpG_CTOB_s_8_cell.deduplicated.txt.gz contains data ->\tkept\n",
            "CpG_OB_s_8_cell.deduplicated.txt.gz contains data ->\tkept\n",
            "CHG_OT_s_8_cell.deduplicated.txt.gz contains data ->\tkept\n",
            "CHG_CTOT_s_8_cell.deduplicated.txt.gz contains data ->\tkept\n",
            "CHG_CTOB_s_8_cell.deduplicated.txt.gz contains data ->\tkept\n",
            "CHG_OB_s_8_cell.deduplicated.txt.gz contains data ->\tkept\n",
            "CHH_OT_s_8_cell.deduplicated.txt.gz contains data ->\tkept\n",
            "CHH_CTOT_s_8_cell.deduplicated.txt.gz contains data ->\tkept\n",
            "CHH_CTOB_s_8_cell.deduplicated.txt.gz contains data ->\tkept\n",
            "CHH_OB_s_8_cell.deduplicated.txt.gz contains data ->\tkept\n",
            "\n",
            "\n",
            "Using these input files: CpG_OT_s_8_cell.deduplicated.txt.gz CpG_CTOT_s_8_cell.deduplicated.txt.gz CpG_CTOB_s_8_cell.deduplicated.txt.gz CpG_OB_s_8_cell.deduplicated.txt.gz CHG_OT_s_8_cell.deduplicated.txt.gz CHG_CTOT_s_8_cell.deduplicated.txt.gz CHG_CTOB_s_8_cell.deduplicated.txt.gz CHG_OB_s_8_cell.deduplicated.txt.gz CHH_OT_s_8_cell.deduplicated.txt.gz CHH_CTOT_s_8_cell.deduplicated.txt.gz CHH_CTOB_s_8_cell.deduplicated.txt.gz CHH_OB_s_8_cell.deduplicated.txt.gz\n",
            "\n",
            "Summary of parameters for bismark2bedGraph conversion:\n",
            "======================================================\n",
            "bedGraph output:\t\ts_8_cell.deduplicated.bedGraph.gz\n",
            "output directory:\t\t><\n",
            "remove whitespaces:\t\tno\n",
            "CX context:\t\t\tno (CpG context only, default)\n",
            "No-header selected:\t\tno\n",
            "Sorting method:\t\t\tUnix sort-based (smaller memory footprint, but slower)\n",
            "Sort buffer size:\t\t2G\n",
            "Coverage threshold:\t\t1\n",
            "=============================================================================\n",
            "Methylation information will now be written into a bedGraph and coverage file\n",
            "=============================================================================\n",
            "\n",
            "Using the following files as Input:\n",
            "/content/CpG_OT_s_8_cell.deduplicated.txt.gz\t/content/CpG_CTOT_s_8_cell.deduplicated.txt.gz\t/content/CpG_CTOB_s_8_cell.deduplicated.txt.gz\t/content/CpG_OB_s_8_cell.deduplicated.txt.gz\n",
            "\n",
            "Writing bedGraph to file: s_8_cell.deduplicated.bedGraph.gz\n",
            "Also writing out a coverage file including counts methylated and unmethylated residues to file: s_8_cell.deduplicated.bismark.cov.gz\n",
            "\n",
            "Now writing methylation information for file >>CpG_OT_s_8_cell.deduplicated.txt.gz<< to individual files for each chromosome\n",
            "Finished writing out individual chromosome files for CpG_OT_s_8_cell.deduplicated.txt.gz\n",
            "Now writing methylation information for file >>CpG_CTOT_s_8_cell.deduplicated.txt.gz<< to individual files for each chromosome\n",
            "Finished writing out individual chromosome files for CpG_CTOT_s_8_cell.deduplicated.txt.gz\n",
            "Now writing methylation information for file >>CpG_CTOB_s_8_cell.deduplicated.txt.gz<< to individual files for each chromosome\n",
            "Finished writing out individual chromosome files for CpG_CTOB_s_8_cell.deduplicated.txt.gz\n",
            "Now writing methylation information for file >>CpG_OB_s_8_cell.deduplicated.txt.gz<< to individual files for each chromosome\n",
            "Finished writing out individual chromosome files for CpG_OB_s_8_cell.deduplicated.txt.gz\n",
            "\n",
            "Collecting temporary chromosome file information... Processing the following input file(s):\n",
            "CpG_OT_s_8_cell.deduplicated.txt.gz.chrchr11.methXtractor.temp\n",
            "\n",
            "Sorting input file CpG_OT_s_8_cell.deduplicated.txt.gz.chrchr11.methXtractor.temp by positions (using -S of 2G)\n",
            "Successfully deleted the temporary input file CpG_OT_s_8_cell.deduplicated.txt.gz.chrchr11.methXtractor.temp\n",
            "\n",
            "Finished BedGraph conversion ...\n",
            "\n",
            "Summary of parameters for genome-wide cytosine report:\n",
            "==============================================================================\n",
            "Coverage infile:\t\t\t\ts_8_cell.deduplicated.bismark.cov.gz\n",
            "Output directory:\t\t\t\t><\n",
            "Parent directory:\t\t\t\t>/content/<\n",
            "Genome directory:\t\t\t\t>/content/mouse_chr/<\n",
            "Coverage threshold\t\t\t\t0 (all positions will be reported, default)\n",
            "CX context:\t\t\t\t\tno (CpG context only, default)\n",
            "Genome coordinates used:\t\t\t1-based (default)\n",
            "GZIP compression:\t\t\t\tyes\n",
            "Split by chromosome:\t\t\t\tno\n",
            "\n",
            "\n",
            "Now reading in and storing sequence information of the genome specified in: /content/mouse_chr/\n",
            "\n",
            "chr chr11 (121973369 bp)\n",
            "\n",
            "Stored sequence information of 1 chromosomes/scaffolds in total\n",
            "\n",
            "==============================================================================\n",
            "Methylation information will now be written into a genome-wide cytosine report\n",
            "==============================================================================\n",
            "\n",
            "Adding context-specific methylation summaries\n",
            "\n",
            ">>> Writing genome-wide cytosine report to: s_8_cell.deduplicated.CpG_report.txt.gz <<<\n",
            "\n",
            ">>> Writing all cytosine context summary file to: s_8_cell.deduplicated.cytosine_context_summary.txt <<<\n",
            "\n",
            "Storing all covered cytosine positions for chromosome: chr11\n",
            "Writing cytosine report for last chromosome chr11 (stored 1737533 different covered positions)\n",
            "Finished writing out cytosine report for covered chromosomes (processed 1 chromosomes/scaffolds in total)\n",
            "\n",
            "Now processing chromosomes that were not covered by any methylation calls in the coverage file...\n",
            "All chromosomes in the genome were covered by at least some reads. coverage2cytosine processing complete.\n",
            "\n",
            "\n",
            "\n",
            "Finished generating genome-wide cytosine report\n",
            "\n"
          ]
        }
      ],
      "source": [
        "! bismark_methylation_extractor --gzip --bedGraph --cytosine_report --genome_folder ./mouse_chr s_8_cell.deduplicated.bam"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "bSMmzCEmO4Y3",
        "outputId": "57f76c39-aab9-46de-f0a1-95bc59995293"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "\n",
            " *** Bismark methylation extractor version v0.24.2 ***\n",
            "\n",
            "Trying to determine the type of mapping from the SAM header line of file icm_cell.deduplicated.bam\n",
            "Treating file(s) as paired-end data (as extracted from @PG line)\n",
            "\n",
            "Setting option '--no_overlap' since this is (normally) the right thing to do for paired-end data\n",
            "\n",
            "\n",
            "Setting core usage to single-threaded (default). Consider using --multicore <int> to speed up the extraction process.\n",
            "\n",
            "Summarising Bismark methylation extractor parameters:\n",
            "===============================================================\n",
            "Bismark paired-end SAM format specified (default)\n",
            "Number of cores to be used: 1\n",
            "Output will be written to the current directory ('/content')\n",
            "\n",
            "\n",
            "Summarising bedGraph parameters:\n",
            "===============================================================\n",
            "Generating additional output in bedGraph and coverage format\n",
            "bedGraph format:\t<Chromosome> <Start Position> <End Position> <Methylation Percentage>\n",
            "coverage format:\t<Chromosome> <Start Position> <End Position> <Methylation Percentage> <count methylated> <count non-methylated>\n",
            "\n",
            "Using a cutoff of 1 read(s) to report cytosine positions\n",
            "Reporting and sorting cytosine methylation information in CpG context only (default)\n",
            "The bedGraph UNIX sort command will use the following memory setting:\t'2G'. Temporary directory used for sorting is the output directory\n",
            "\n",
            "\n",
            "Summarising genome-wide cytosine methylation report parameters:\n",
            "===============================================================\n",
            "Generating comprehensive genome-wide cytosine report\n",
            "(output format: <Chromosome> <Position> <Strand> <count methylated> <count non-methylated>  <C-context>  <trinucleotide context> )\n",
            "Reporting cytosine methylation in CpG context only (default)\n",
            "Using 1-based genomic coordinates (default)\n",
            "Genome folder was specified as ./mouse_chr/\n",
            "\n",
            "Checking file >>icm_cell.deduplicated.bam<< for signs of file truncation...\n",
            "\n",
            "\n",
            "Now testing Bismark result file >icm_cell.deduplicated.bam< for positional sorting (which would be bad...)\t...passed!\n",
            "Writing result file containing methylation information for C in CpG context from the original top strand to CpG_OT_icm_cell.deduplicated.txt.gz\n",
            "Writing result file containing methylation information for C in CpG context from the complementary to original top strand to CpG_CTOT_icm_cell.deduplicated.txt.gz\n",
            "Writing result file containing methylation information for C in CpG context from the complementary to original bottom strand to CpG_CTOB_icm_cell.deduplicated.txt.gz\n",
            "Writing result file containing methylation information for C in CpG context from the original bottom strand to CpG_OB_icm_cell.deduplicated.txt.gz\n",
            "\n",
            "Writing result file containing methylation information for C in CHG context from the original top strand to CHG_OT_icm_cell.deduplicated.txt.gz\n",
            "Writing result file containing methylation information for C in CHG context from the complementary to original top strand to CHG_CTOT_icm_cell.deduplicated.txt.gz\n",
            "Writing result file containing methylation information for C in CHG context from the complementary to original bottom strand to CHG_CTOB_icm_cell.deduplicated.txt.gz\n",
            "Writing result file containing methylation information for C in CHG context from the original bottom strand to CHG_OB_icm_cell.deduplicated.txt.gz\n",
            "\n",
            "Writing result file containing methylation information for C in CHH context from the original top strand to CHH_OT_icm_cell.deduplicated.txt.gz\n",
            "Writing result file containing methylation information for C in CHH context from the complementary to original top strand to CHH_CTOT_icm_cell.deduplicated.txt.gz\n",
            "Writing result file containing methylation information for C in CHH context from the complementary to original bottom strand to CHH_CTOB_icm_cell.deduplicated.txt.gz\n",
            "Writing result file containing methylation information for C in CHH context from the original bottom strand to CHH_OB_icm_cell.deduplicated.txt.gz\n",
            "\n",
            "Now reading in Bismark result file icm_cell.deduplicated.bam\n",
            "skipping SAM header line:\t@HD\tVN:1.0\tSO:unsorted\n",
            "skipping SAM header line:\t@SQ\tSN:chr11\tLN:121973369\n",
            "skipping SAM header line:\t@PG\tID:Bismark\tVN:v0.24.2\tCL:\"bismark --genome_folder /mnt/cache-1/for_minor/mm39_11/ -q --non_directional -parallel 10 -1 SRR5836475_1.fq -2 SRR5836475_2.fq\"\n",
            "skipping SAM header line:\t@PG\tID:samtools\tPN:samtools\tPP:Bismark\tVN:1.19.2\tCL:/usr/local/bin/samtools view -h --threads 1 SRR5836475_1_bismark_bt2_pe.bam\n",
            "skipping SAM header line:\t@PG\tID:samtools.1\tPN:samtools\tPP:samtools\tVN:1.19.2\tCL:/usr/local/bin/samtools view -bS --threads 1 -\n",
            "skipping SAM header line:\t@PG\tID:samtools.2\tPN:samtools\tPP:samtools.1\tVN:1.19.2\tCL:/usr/local/bin/samtools view -h icm_cell.deduplicated.bam\n",
            "Processed lines: 500000\n",
            "Processed lines: 1000000\n",
            "Processed lines: 1500000\n",
            "Processed lines: 2000000\n",
            "Processed lines: 2500000\n",
            "Processed lines: 3000000\n",
            "Processed lines: 3500000\n",
            "\n",
            "Processed 3791973 lines in total\n",
            "Total number of methylation call strings processed: 7583946\n",
            "\n",
            "Final Cytosine Methylation Report\n",
            "=================================\n",
            "Total number of C's analysed:\t177777254\n",
            "\n",
            "Total methylated C's in CpG context:\t1915876\n",
            "Total methylated C's in CHG context:\t298499\n",
            "Total methylated C's in CHH context:\t1434775\n",
            "\n",
            "Total C to T conversions in CpG context:\t6256205\n",
            "Total C to T conversions in CHG context:\t39157632\n",
            "Total C to T conversions in CHH context:\t128714267\n",
            "\n",
            "C methylated in CpG context:\t23.4%\n",
            "C methylated in CHG context:\t0.8%\n",
            "C methylated in CHH context:\t1.1%\n",
            "\n",
            "\n",
            "Determining maximum read lengths for M-Bias plots\n",
            "Maximum read length of Read 1: 140\n",
            "Maximum read length of Read 2: 140\n",
            "\n",
            "Perl module GD::Graph::lines is not installed, skipping drawing M-bias plots (only writing out M-bias plot table)\n",
            "Determining maximum read lengths for M-Bias plots\n",
            "Maximum read length of Read 1: 140\n",
            "Maximum read length of Read 2: 140\n",
            "\n",
            "Perl module GD::Graph::lines is not installed, skipping drawing M-bias plots (only writing out M-bias plot table)\n",
            "Deleting unused files ...\n",
            "\n",
            "CpG_OT_icm_cell.deduplicated.txt.gz contains data ->\tkept\n",
            "CpG_CTOT_icm_cell.deduplicated.txt.gz contains data ->\tkept\n",
            "CpG_CTOB_icm_cell.deduplicated.txt.gz contains data ->\tkept\n",
            "CpG_OB_icm_cell.deduplicated.txt.gz contains data ->\tkept\n",
            "CHG_OT_icm_cell.deduplicated.txt.gz contains data ->\tkept\n",
            "CHG_CTOT_icm_cell.deduplicated.txt.gz contains data ->\tkept\n",
            "CHG_CTOB_icm_cell.deduplicated.txt.gz contains data ->\tkept\n",
            "CHG_OB_icm_cell.deduplicated.txt.gz contains data ->\tkept\n",
            "CHH_OT_icm_cell.deduplicated.txt.gz contains data ->\tkept\n",
            "CHH_CTOT_icm_cell.deduplicated.txt.gz contains data ->\tkept\n",
            "CHH_CTOB_icm_cell.deduplicated.txt.gz contains data ->\tkept\n",
            "CHH_OB_icm_cell.deduplicated.txt.gz contains data ->\tkept\n",
            "\n",
            "\n",
            "Using these input files: CpG_OT_icm_cell.deduplicated.txt.gz CpG_CTOT_icm_cell.deduplicated.txt.gz CpG_CTOB_icm_cell.deduplicated.txt.gz CpG_OB_icm_cell.deduplicated.txt.gz CHG_OT_icm_cell.deduplicated.txt.gz CHG_CTOT_icm_cell.deduplicated.txt.gz CHG_CTOB_icm_cell.deduplicated.txt.gz CHG_OB_icm_cell.deduplicated.txt.gz CHH_OT_icm_cell.deduplicated.txt.gz CHH_CTOT_icm_cell.deduplicated.txt.gz CHH_CTOB_icm_cell.deduplicated.txt.gz CHH_OB_icm_cell.deduplicated.txt.gz\n",
            "\n",
            "Summary of parameters for bismark2bedGraph conversion:\n",
            "======================================================\n",
            "bedGraph output:\t\ticm_cell.deduplicated.bedGraph.gz\n",
            "output directory:\t\t><\n",
            "remove whitespaces:\t\tno\n",
            "CX context:\t\t\tno (CpG context only, default)\n",
            "No-header selected:\t\tno\n",
            "Sorting method:\t\t\tUnix sort-based (smaller memory footprint, but slower)\n",
            "Sort buffer size:\t\t2G\n",
            "Coverage threshold:\t\t1\n",
            "=============================================================================\n",
            "Methylation information will now be written into a bedGraph and coverage file\n",
            "=============================================================================\n",
            "\n",
            "Using the following files as Input:\n",
            "/content/CpG_OT_icm_cell.deduplicated.txt.gz\t/content/CpG_CTOT_icm_cell.deduplicated.txt.gz\t/content/CpG_CTOB_icm_cell.deduplicated.txt.gz\t/content/CpG_OB_icm_cell.deduplicated.txt.gz\n",
            "\n",
            "Writing bedGraph to file: icm_cell.deduplicated.bedGraph.gz\n",
            "Also writing out a coverage file including counts methylated and unmethylated residues to file: icm_cell.deduplicated.bismark.cov.gz\n",
            "\n",
            "Now writing methylation information for file >>CpG_OT_icm_cell.deduplicated.txt.gz<< to individual files for each chromosome\n",
            "Finished writing out individual chromosome files for CpG_OT_icm_cell.deduplicated.txt.gz\n",
            "Now writing methylation information for file >>CpG_CTOT_icm_cell.deduplicated.txt.gz<< to individual files for each chromosome\n",
            "Finished writing out individual chromosome files for CpG_CTOT_icm_cell.deduplicated.txt.gz\n",
            "Now writing methylation information for file >>CpG_CTOB_icm_cell.deduplicated.txt.gz<< to individual files for each chromosome\n",
            "Finished writing out individual chromosome files for CpG_CTOB_icm_cell.deduplicated.txt.gz\n",
            "Now writing methylation information for file >>CpG_OB_icm_cell.deduplicated.txt.gz<< to individual files for each chromosome\n",
            "Finished writing out individual chromosome files for CpG_OB_icm_cell.deduplicated.txt.gz\n",
            "\n",
            "Collecting temporary chromosome file information... Processing the following input file(s):\n",
            "CpG_OT_icm_cell.deduplicated.txt.gz.chrchr11.methXtractor.temp\n",
            "\n",
            "Sorting input file CpG_OT_icm_cell.deduplicated.txt.gz.chrchr11.methXtractor.temp by positions (using -S of 2G)\n",
            "Successfully deleted the temporary input file CpG_OT_icm_cell.deduplicated.txt.gz.chrchr11.methXtractor.temp\n",
            "\n",
            "Finished BedGraph conversion ...\n",
            "\n",
            "Summary of parameters for genome-wide cytosine report:\n",
            "==============================================================================\n",
            "Coverage infile:\t\t\t\ticm_cell.deduplicated.bismark.cov.gz\n",
            "Output directory:\t\t\t\t><\n",
            "Parent directory:\t\t\t\t>/content/<\n",
            "Genome directory:\t\t\t\t>/content/mouse_chr/<\n",
            "Coverage threshold\t\t\t\t0 (all positions will be reported, default)\n",
            "CX context:\t\t\t\t\tno (CpG context only, default)\n",
            "Genome coordinates used:\t\t\t1-based (default)\n",
            "GZIP compression:\t\t\t\tyes\n",
            "Split by chromosome:\t\t\t\tno\n",
            "\n",
            "\n",
            "Now reading in and storing sequence information of the genome specified in: /content/mouse_chr/\n",
            "\n",
            "chr chr11 (121973369 bp)\n",
            "\n",
            "Stored sequence information of 1 chromosomes/scaffolds in total\n",
            "\n",
            "==============================================================================\n",
            "Methylation information will now be written into a genome-wide cytosine report\n",
            "==============================================================================\n",
            "\n",
            "Adding context-specific methylation summaries\n",
            "\n",
            ">>> Writing genome-wide cytosine report to: icm_cell.deduplicated.CpG_report.txt.gz <<<\n",
            "\n",
            ">>> Writing all cytosine context summary file to: icm_cell.deduplicated.cytosine_context_summary.txt <<<\n",
            "\n",
            "Storing all covered cytosine positions for chromosome: chr11\n",
            "Writing cytosine report for last chromosome chr11 (stored 2054296 different covered positions)\n",
            "Finished writing out cytosine report for covered chromosomes (processed 1 chromosomes/scaffolds in total)\n",
            "\n",
            "Now processing chromosomes that were not covered by any methylation calls in the coverage file...\n",
            "All chromosomes in the genome were covered by at least some reads. coverage2cytosine processing complete.\n",
            "\n",
            "\n",
            "\n",
            "Finished generating genome-wide cytosine report\n",
            "\n"
          ]
        }
      ],
      "source": [
        "! bismark_methylation_extractor --gzip --bedGraph --cytosine_report --genome_folder ./mouse_chr icm_cell.deduplicated.bam"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "noxWppgvO_Nv",
        "outputId": "a4c59fcd-960a-40b0-cd50-104129031eae"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "\n",
            " *** Bismark methylation extractor version v0.24.2 ***\n",
            "\n",
            "Trying to determine the type of mapping from the SAM header line of file epiblast_cell.deduplicated.bam\n",
            "Treating file(s) as paired-end data (as extracted from @PG line)\n",
            "\n",
            "Setting option '--no_overlap' since this is (normally) the right thing to do for paired-end data\n",
            "\n",
            "\n",
            "Setting core usage to single-threaded (default). Consider using --multicore <int> to speed up the extraction process.\n",
            "\n",
            "Summarising Bismark methylation extractor parameters:\n",
            "===============================================================\n",
            "Bismark paired-end SAM format specified (default)\n",
            "Number of cores to be used: 1\n",
            "Output will be written to the current directory ('/content')\n",
            "\n",
            "\n",
            "Summarising bedGraph parameters:\n",
            "===============================================================\n",
            "Generating additional output in bedGraph and coverage format\n",
            "bedGraph format:\t<Chromosome> <Start Position> <End Position> <Methylation Percentage>\n",
            "coverage format:\t<Chromosome> <Start Position> <End Position> <Methylation Percentage> <count methylated> <count non-methylated>\n",
            "\n",
            "Using a cutoff of 1 read(s) to report cytosine positions\n",
            "Reporting and sorting cytosine methylation information in CpG context only (default)\n",
            "The bedGraph UNIX sort command will use the following memory setting:\t'2G'. Temporary directory used for sorting is the output directory\n",
            "\n",
            "\n",
            "Summarising genome-wide cytosine methylation report parameters:\n",
            "===============================================================\n",
            "Generating comprehensive genome-wide cytosine report\n",
            "(output format: <Chromosome> <Position> <Strand> <count methylated> <count non-methylated>  <C-context>  <trinucleotide context> )\n",
            "Reporting cytosine methylation in CpG context only (default)\n",
            "Using 1-based genomic coordinates (default)\n",
            "Genome folder was specified as ./mouse_chr/\n",
            "\n",
            "Checking file >>epiblast_cell.deduplicated.bam<< for signs of file truncation...\n",
            "\n",
            "\n",
            "Now testing Bismark result file >epiblast_cell.deduplicated.bam< for positional sorting (which would be bad...)\t...passed!\n",
            "Writing result file containing methylation information for C in CpG context from the original top strand to CpG_OT_epiblast_cell.deduplicated.txt.gz\n",
            "Writing result file containing methylation information for C in CpG context from the complementary to original top strand to CpG_CTOT_epiblast_cell.deduplicated.txt.gz\n",
            "Writing result file containing methylation information for C in CpG context from the complementary to original bottom strand to CpG_CTOB_epiblast_cell.deduplicated.txt.gz\n",
            "Writing result file containing methylation information for C in CpG context from the original bottom strand to CpG_OB_epiblast_cell.deduplicated.txt.gz\n",
            "\n",
            "Writing result file containing methylation information for C in CHG context from the original top strand to CHG_OT_epiblast_cell.deduplicated.txt.gz\n",
            "Writing result file containing methylation information for C in CHG context from the complementary to original top strand to CHG_CTOT_epiblast_cell.deduplicated.txt.gz\n",
            "Writing result file containing methylation information for C in CHG context from the complementary to original bottom strand to CHG_CTOB_epiblast_cell.deduplicated.txt.gz\n",
            "Writing result file containing methylation information for C in CHG context from the original bottom strand to CHG_OB_epiblast_cell.deduplicated.txt.gz\n",
            "\n",
            "Writing result file containing methylation information for C in CHH context from the original top strand to CHH_OT_epiblast_cell.deduplicated.txt.gz\n",
            "Writing result file containing methylation information for C in CHH context from the complementary to original top strand to CHH_CTOT_epiblast_cell.deduplicated.txt.gz\n",
            "Writing result file containing methylation information for C in CHH context from the complementary to original bottom strand to CHH_CTOB_epiblast_cell.deduplicated.txt.gz\n",
            "Writing result file containing methylation information for C in CHH context from the original bottom strand to CHH_OB_epiblast_cell.deduplicated.txt.gz\n",
            "\n",
            "Now reading in Bismark result file epiblast_cell.deduplicated.bam\n",
            "skipping SAM header line:\t@HD\tVN:1.0\tSO:unsorted\n",
            "skipping SAM header line:\t@SQ\tSN:chr11\tLN:121973369\n",
            "skipping SAM header line:\t@PG\tID:Bismark\tVN:v0.24.2\tCL:\"bismark --genome_folder /mnt/cache-1/for_minor/mm39_11/ -q --non_directional -parallel 10 -1 SRR3824222_1.fq,SRR5836473_1.fq,SRR5836473_1.fq -2 SRR3824222_2.fq,SRR5836473_2.fq,SRR5836473_2.fq\"\n",
            "skipping SAM header line:\t@PG\tID:samtools\tPN:samtools\tPP:Bismark\tVN:1.19.2\tCL:/usr/local/bin/samtools view -h --threads 1 SRR3824222_1_bismark_bt2_pe.bam\n",
            "skipping SAM header line:\t@PG\tID:samtools.1\tPN:samtools\tPP:samtools\tVN:1.19.2\tCL:/usr/local/bin/samtools view -bS --threads 1 -\n",
            "skipping SAM header line:\t@PG\tID:samtools.2\tPN:samtools\tPP:samtools.1\tVN:1.19.2\tCL:/usr/local/bin/samtools view -h epiblast_cell.deduplicated.bam\n",
            "Processed lines: 500000\n",
            "Processed lines: 1000000\n",
            "Processed lines: 1500000\n",
            "Processed lines: 2000000\n",
            "Processed lines: 2500000\n",
            "Processed lines: 3000000\n",
            "Processed lines: 3500000\n",
            "Processed lines: 4000000\n",
            "Processed lines: 4500000\n",
            "Processed lines: 5000000\n",
            "Processed lines: 5500000\n",
            "Processed lines: 6000000\n",
            "Processed lines: 6500000\n",
            "\n",
            "Processed 6846444 lines in total\n",
            "Total number of methylation call strings processed: 13692888\n",
            "\n",
            "Final Cytosine Methylation Report\n",
            "=================================\n",
            "Total number of C's analysed:\t331429490\n",
            "\n",
            "Total methylated C's in CpG context:\t11550199\n",
            "Total methylated C's in CHG context:\t1574513\n",
            "Total methylated C's in CHH context:\t4133143\n",
            "\n",
            "Total C to T conversions in CpG context:\t3463972\n",
            "Total C to T conversions in CHG context:\t71924830\n",
            "Total C to T conversions in CHH context:\t238782833\n",
            "\n",
            "C methylated in CpG context:\t76.9%\n",
            "C methylated in CHG context:\t2.1%\n",
            "C methylated in CHH context:\t1.7%\n",
            "\n",
            "\n",
            "Determining maximum read lengths for M-Bias plots\n",
            "Maximum read length of Read 1: 140\n",
            "Maximum read length of Read 2: 140\n",
            "\n",
            "Perl module GD::Graph::lines is not installed, skipping drawing M-bias plots (only writing out M-bias plot table)\n",
            "Determining maximum read lengths for M-Bias plots\n",
            "Maximum read length of Read 1: 140\n",
            "Maximum read length of Read 2: 140\n",
            "\n",
            "Perl module GD::Graph::lines is not installed, skipping drawing M-bias plots (only writing out M-bias plot table)\n",
            "Deleting unused files ...\n",
            "\n",
            "CpG_OT_epiblast_cell.deduplicated.txt.gz contains data ->\tkept\n",
            "CpG_CTOT_epiblast_cell.deduplicated.txt.gz contains data ->\tkept\n",
            "CpG_CTOB_epiblast_cell.deduplicated.txt.gz contains data ->\tkept\n",
            "CpG_OB_epiblast_cell.deduplicated.txt.gz contains data ->\tkept\n",
            "CHG_OT_epiblast_cell.deduplicated.txt.gz contains data ->\tkept\n",
            "CHG_CTOT_epiblast_cell.deduplicated.txt.gz contains data ->\tkept\n",
            "CHG_CTOB_epiblast_cell.deduplicated.txt.gz contains data ->\tkept\n",
            "CHG_OB_epiblast_cell.deduplicated.txt.gz contains data ->\tkept\n",
            "CHH_OT_epiblast_cell.deduplicated.txt.gz contains data ->\tkept\n",
            "CHH_CTOT_epiblast_cell.deduplicated.txt.gz contains data ->\tkept\n",
            "CHH_CTOB_epiblast_cell.deduplicated.txt.gz contains data ->\tkept\n",
            "CHH_OB_epiblast_cell.deduplicated.txt.gz contains data ->\tkept\n",
            "\n",
            "\n",
            "Using these input files: CpG_OT_epiblast_cell.deduplicated.txt.gz CpG_CTOT_epiblast_cell.deduplicated.txt.gz CpG_CTOB_epiblast_cell.deduplicated.txt.gz CpG_OB_epiblast_cell.deduplicated.txt.gz CHG_OT_epiblast_cell.deduplicated.txt.gz CHG_CTOT_epiblast_cell.deduplicated.txt.gz CHG_CTOB_epiblast_cell.deduplicated.txt.gz CHG_OB_epiblast_cell.deduplicated.txt.gz CHH_OT_epiblast_cell.deduplicated.txt.gz CHH_CTOT_epiblast_cell.deduplicated.txt.gz CHH_CTOB_epiblast_cell.deduplicated.txt.gz CHH_OB_epiblast_cell.deduplicated.txt.gz\n",
            "\n",
            "Summary of parameters for bismark2bedGraph conversion:\n",
            "======================================================\n",
            "bedGraph output:\t\tepiblast_cell.deduplicated.bedGraph.gz\n",
            "output directory:\t\t><\n",
            "remove whitespaces:\t\tno\n",
            "CX context:\t\t\tno (CpG context only, default)\n",
            "No-header selected:\t\tno\n",
            "Sorting method:\t\t\tUnix sort-based (smaller memory footprint, but slower)\n",
            "Sort buffer size:\t\t2G\n",
            "Coverage threshold:\t\t1\n",
            "=============================================================================\n",
            "Methylation information will now be written into a bedGraph and coverage file\n",
            "=============================================================================\n",
            "\n",
            "Using the following files as Input:\n",
            "/content/CpG_OT_epiblast_cell.deduplicated.txt.gz\t/content/CpG_CTOT_epiblast_cell.deduplicated.txt.gz\t/content/CpG_CTOB_epiblast_cell.deduplicated.txt.gz\t/content/CpG_OB_epiblast_cell.deduplicated.txt.gz\n",
            "\n",
            "Writing bedGraph to file: epiblast_cell.deduplicated.bedGraph.gz\n",
            "Also writing out a coverage file including counts methylated and unmethylated residues to file: epiblast_cell.deduplicated.bismark.cov.gz\n",
            "\n",
            "Now writing methylation information for file >>CpG_OT_epiblast_cell.deduplicated.txt.gz<< to individual files for each chromosome\n",
            "Finished writing out individual chromosome files for CpG_OT_epiblast_cell.deduplicated.txt.gz\n",
            "Now writing methylation information for file >>CpG_CTOT_epiblast_cell.deduplicated.txt.gz<< to individual files for each chromosome\n",
            "Finished writing out individual chromosome files for CpG_CTOT_epiblast_cell.deduplicated.txt.gz\n",
            "Now writing methylation information for file >>CpG_CTOB_epiblast_cell.deduplicated.txt.gz<< to individual files for each chromosome\n",
            "Finished writing out individual chromosome files for CpG_CTOB_epiblast_cell.deduplicated.txt.gz\n",
            "Now writing methylation information for file >>CpG_OB_epiblast_cell.deduplicated.txt.gz<< to individual files for each chromosome\n",
            "Finished writing out individual chromosome files for CpG_OB_epiblast_cell.deduplicated.txt.gz\n",
            "\n",
            "Collecting temporary chromosome file information... Processing the following input file(s):\n",
            "CpG_OT_epiblast_cell.deduplicated.txt.gz.chrchr11.methXtractor.temp\n",
            "\n",
            "Sorting input file CpG_OT_epiblast_cell.deduplicated.txt.gz.chrchr11.methXtractor.temp by positions (using -S of 2G)\n",
            "Successfully deleted the temporary input file CpG_OT_epiblast_cell.deduplicated.txt.gz.chrchr11.methXtractor.temp\n",
            "\n",
            "Finished BedGraph conversion ...\n",
            "\n",
            "Summary of parameters for genome-wide cytosine report:\n",
            "==============================================================================\n",
            "Coverage infile:\t\t\t\tepiblast_cell.deduplicated.bismark.cov.gz\n",
            "Output directory:\t\t\t\t><\n",
            "Parent directory:\t\t\t\t>/content/<\n",
            "Genome directory:\t\t\t\t>/content/mouse_chr/<\n",
            "Coverage threshold\t\t\t\t0 (all positions will be reported, default)\n",
            "CX context:\t\t\t\t\tno (CpG context only, default)\n",
            "Genome coordinates used:\t\t\t1-based (default)\n",
            "GZIP compression:\t\t\t\tyes\n",
            "Split by chromosome:\t\t\t\tno\n",
            "\n",
            "\n",
            "Now reading in and storing sequence information of the genome specified in: /content/mouse_chr/\n",
            "\n",
            "chr chr11 (121973369 bp)\n",
            "\n",
            "Stored sequence information of 1 chromosomes/scaffolds in total\n",
            "\n",
            "==============================================================================\n",
            "Methylation information will now be written into a genome-wide cytosine report\n",
            "==============================================================================\n",
            "\n",
            "Adding context-specific methylation summaries\n",
            "\n",
            ">>> Writing genome-wide cytosine report to: epiblast_cell.deduplicated.CpG_report.txt.gz <<<\n",
            "\n",
            ">>> Writing all cytosine context summary file to: epiblast_cell.deduplicated.cytosine_context_summary.txt <<<\n",
            "\n",
            "Storing all covered cytosine positions for chromosome: chr11\n",
            "Writing cytosine report for last chromosome chr11 (stored 2250540 different covered positions)\n",
            "Finished writing out cytosine report for covered chromosomes (processed 1 chromosomes/scaffolds in total)\n",
            "\n",
            "Now processing chromosomes that were not covered by any methylation calls in the coverage file...\n",
            "All chromosomes in the genome were covered by at least some reads. coverage2cytosine processing complete.\n",
            "\n",
            "\n",
            "\n",
            "Finished generating genome-wide cytosine report\n",
            "\n"
          ]
        }
      ],
      "source": [
        "! bismark_methylation_extractor --gzip --bedGraph --cytosine_report --genome_folder ./mouse_chr epiblast_cell.deduplicated.bam"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "iNwzMUeAkQ4f",
        "outputId": "b0e66947-36a0-474d-f0fe-9303ab41c5b2"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "total 14G\n",
            "-rw-r--r-- 1 root root   198 Feb 10 15:29 checksum.md5\n",
            "-rw-r--r-- 1 root root   13K Feb 10 20:42 CHG_CTOB_epiblast_cell.deduplicated.txt.gz\n",
            "-rw-r--r-- 1 root root  9.4K Feb 10 19:23 CHG_CTOB_icm_cell.deduplicated.txt.gz\n",
            "-rw-r--r-- 1 root root  7.7K Feb 10 18:39 CHG_CTOB_s_8_cell.deduplicated.txt.gz\n",
            "-rw-r--r-- 1 root root   14K Feb 10 20:42 CHG_CTOT_epiblast_cell.deduplicated.txt.gz\n",
            "-rw-r--r-- 1 root root   10K Feb 10 19:23 CHG_CTOT_icm_cell.deduplicated.txt.gz\n",
            "-rw-r--r-- 1 root root  8.2K Feb 10 18:39 CHG_CTOT_s_8_cell.deduplicated.txt.gz\n",
            "-rw-r--r-- 1 root root  157M Feb 10 20:42 CHG_OB_epiblast_cell.deduplicated.txt.gz\n",
            "-rw-r--r-- 1 root root   84M Feb 10 19:23 CHG_OB_icm_cell.deduplicated.txt.gz\n",
            "-rw-r--r-- 1 root root   51M Feb 10 18:39 CHG_OB_s_8_cell.deduplicated.txt.gz\n",
            "-rw-r--r-- 1 root root  157M Feb 10 20:42 CHG_OT_epiblast_cell.deduplicated.txt.gz\n",
            "-rw-r--r-- 1 root root   85M Feb 10 19:23 CHG_OT_icm_cell.deduplicated.txt.gz\n",
            "-rw-r--r-- 1 root root   51M Feb 10 18:39 CHG_OT_s_8_cell.deduplicated.txt.gz\n",
            "-rw-r--r-- 1 root root  1.9M Feb 10 20:42 CHH_CTOB_epiblast_cell.deduplicated.txt.gz\n",
            "-rw-r--r-- 1 root root  1.6M Feb 10 19:23 CHH_CTOB_icm_cell.deduplicated.txt.gz\n",
            "-rw-r--r-- 1 root root  1.1M Feb 10 18:39 CHH_CTOB_s_8_cell.deduplicated.txt.gz\n",
            "-rw-r--r-- 1 root root  1.7M Feb 10 20:42 CHH_CTOT_epiblast_cell.deduplicated.txt.gz\n",
            "-rw-r--r-- 1 root root  1.4M Feb 10 19:23 CHH_CTOT_icm_cell.deduplicated.txt.gz\n",
            "-rw-r--r-- 1 root root 1014K Feb 10 18:39 CHH_CTOT_s_8_cell.deduplicated.txt.gz\n",
            "-rw-r--r-- 1 root root  419M Feb 10 20:42 CHH_OB_epiblast_cell.deduplicated.txt.gz\n",
            "-rw-r--r-- 1 root root  223M Feb 10 19:23 CHH_OB_icm_cell.deduplicated.txt.gz\n",
            "-rw-r--r-- 1 root root  133M Feb 10 18:39 CHH_OB_s_8_cell.deduplicated.txt.gz\n",
            "-rw-r--r-- 1 root root  419M Feb 10 20:42 CHH_OT_epiblast_cell.deduplicated.txt.gz\n",
            "-rw-r--r-- 1 root root  223M Feb 10 19:23 CHH_OT_icm_cell.deduplicated.txt.gz\n",
            "-rw-r--r-- 1 root root  133M Feb 10 18:39 CHH_OT_s_8_cell.deduplicated.txt.gz\n",
            "-rw-r--r-- 1 root root   23K Feb 10 15:13 condacolab_install.log\n",
            "-rw-r--r-- 1 root root  7.5K Feb 10 20:42 CpG_CTOB_epiblast_cell.deduplicated.txt.gz\n",
            "-rw-r--r-- 1 root root  4.7K Feb 10 19:23 CpG_CTOB_icm_cell.deduplicated.txt.gz\n",
            "-rw-r--r-- 1 root root  4.2K Feb 10 18:39 CpG_CTOB_s_8_cell.deduplicated.txt.gz\n",
            "-rw-r--r-- 1 root root  8.0K Feb 10 20:42 CpG_CTOT_epiblast_cell.deduplicated.txt.gz\n",
            "-rw-r--r-- 1 root root  4.6K Feb 10 19:23 CpG_CTOT_icm_cell.deduplicated.txt.gz\n",
            "-rw-r--r-- 1 root root  3.8K Feb 10 18:39 CpG_CTOT_s_8_cell.deduplicated.txt.gz\n",
            "-rw-r--r-- 1 root root   49M Feb 10 20:42 CpG_OB_epiblast_cell.deduplicated.txt.gz\n",
            "-rw-r--r-- 1 root root   28M Feb 10 19:23 CpG_OB_icm_cell.deduplicated.txt.gz\n",
            "-rw-r--r-- 1 root root   17M Feb 10 18:39 CpG_OB_s_8_cell.deduplicated.txt.gz\n",
            "-rw-r--r-- 1 root root   49M Feb 10 20:42 CpG_OT_epiblast_cell.deduplicated.txt.gz\n",
            "-rw-r--r-- 1 root root   28M Feb 10 19:23 CpG_OT_icm_cell.deduplicated.txt.gz\n",
            "-rw-r--r-- 1 root root   17M Feb 10 18:39 CpG_OT_s_8_cell.deduplicated.txt.gz\n",
            "-rw-r--r-- 1 root root  1.9G Feb 10 17:32 epiblast_cell.deduplicated.bam\n",
            "-rw-r--r-- 1 root root   14M Feb 10 20:44 epiblast_cell.deduplicated.bedGraph.gz\n",
            "-rw-r--r-- 1 root root   14M Feb 10 20:44 epiblast_cell.deduplicated.bismark.cov.gz\n",
            "-rw-r--r-- 1 root root   11M Feb 10 20:48 epiblast_cell.deduplicated.CpG_report.txt.gz\n",
            "-rw-r--r-- 1 root root  1.7K Feb 10 20:48 epiblast_cell.deduplicated.cytosine_context_summary.txt\n",
            "-rw-r--r-- 1 root root   24K Feb 10 20:42 epiblast_cell.deduplicated.M-bias.txt\n",
            "-rw-r--r-- 1 root root   850 Feb 10 20:42 epiblast_cell.deduplicated_splitting_report.txt\n",
            "-rw-r--r-- 1 root root   282 Feb 10 17:32 epiblast_cell.deduplication_report.txt\n",
            "drwxr-xr-x 8 root root  4.0K Feb 10 15:04 FastQC\n",
            "-rw-r--r-- 1 root root  9.8M Jan 15  2020 fastqc_v0.11.9.zip\n",
            "-rw-r--r-- 1 root root  9.8M Jan 15  2020 fastqc_v0.11.9.zip.1\n",
            "-rw-r--r-- 1 root root  9.8M Jan 15  2020 fastqc_v0.11.9.zip.2\n",
            "-rw-r--r-- 1 root root  1.1G Feb 10 17:26 icm_cell.deduplicated.bam\n",
            "-rw-r--r-- 1 root root   12M Feb 10 19:25 icm_cell.deduplicated.bedGraph.gz\n",
            "-rw-r--r-- 1 root root   12M Feb 10 19:25 icm_cell.deduplicated.bismark.cov.gz\n",
            "-rw-r--r-- 1 root root   11M Feb 10 19:29 icm_cell.deduplicated.CpG_report.txt.gz\n",
            "-rw-r--r-- 1 root root  1.6K Feb 10 19:29 icm_cell.deduplicated.cytosine_context_summary.txt\n",
            "-rw-r--r-- 1 root root   23K Feb 10 19:23 icm_cell.deduplicated.M-bias.txt\n",
            "-rw-r--r-- 1 root root   842 Feb 10 19:23 icm_cell.deduplicated_splitting_report.txt\n",
            "-rw-r--r-- 1 root root   282 Feb 10 17:26 icm_cell.deduplication_report.txt\n",
            "drwxr-xr-x 2 root root  4.0K Feb 10 18:14 mouse_chr\n",
            "-rw-r--r-- 1 root root  620M Feb 10 16:43 s_8_cell.deduplicated.bam\n",
            "-rw-r--r-- 1 root root  9.8M Feb 10 18:40 s_8_cell.deduplicated.bedGraph.gz\n",
            "-rw-r--r-- 1 root root  9.4M Feb 10 18:40 s_8_cell.deduplicated.bismark.cov.gz\n",
            "-rw-r--r-- 1 root root  9.9M Feb 10 18:43 s_8_cell.deduplicated.CpG_report.txt.gz\n",
            "-rw-r--r-- 1 root root  1.6K Feb 10 18:43 s_8_cell.deduplicated.cytosine_context_summary.txt\n",
            "-rw-r--r-- 1 root root   22K Feb 10 18:39 s_8_cell.deduplicated.M-bias.txt\n",
            "-rw-r--r-- 1 root root   841 Feb 10 18:39 s_8_cell.deduplicated_splitting_report.txt\n",
            "-rw-r--r-- 1 root root   283 Feb 10 16:43 s_8_cell.deduplication_report.txt\n",
            "drwxr-xr-x 1 root root  4.0K Feb  8 14:21 sample_data\n",
            "-rw-r--r-- 1 root root  528M Feb 10 16:25 sorted_8cell.bam\n",
            "-rw-r--r-- 1 root root  203K Feb 10 16:26 sorted_8cell.bam.bai\n",
            "-rw-r--r-- 1 root root  1.3G Feb 10 17:10 sorted_epiblast.bam\n",
            "-rw-r--r-- 1 root root  349K Feb 10 17:10 sorted_epiblast.bam.bai\n",
            "-rw-r--r-- 1 root root  777M Feb 10 17:13 sorted_icm.bam\n",
            "-rw-r--r-- 1 root root  341K Feb 10 17:13 sorted_icm.bam.bai\n",
            "-rw-r--r-- 1 root root  156M Feb 10 17:05 sorted_icm.bam.tmp.0000-001.bam\n",
            "-rw-r--r-- 1 root root  175M Feb 10 17:02 sorted_icm.bam.tmp.0000.bam\n",
            "-rw-r--r-- 1 root root  168M Feb 10 17:02 sorted_icm.bam.tmp.0001.bam\n",
            "-rw-r--r-- 1 root root  175M Feb 10 17:03 sorted_icm.bam.tmp.0002.bam\n",
            "-rw-r--r-- 1 root root  168M Feb 10 17:03 sorted_icm.bam.tmp.0003.bam\n",
            "-rw-r--r-- 1 root root  1.9G Feb 10 16:02 SRR3824222_1_bismark_bt2_pe.bam\n",
            "-rw-r--r-- 1 root root  769M Feb 10 16:03 SRR5836473_1_bismark_bt2_pe.bam\n",
            "-rw-r--r-- 1 root root  245M Feb 10 15:13 SRR5836473_1.fastq.gz\n",
            "-rw-r--r-- 1 root root   16M Feb 10 16:01 SRR5836473_1.fastq.gz.1\n",
            "-rw-r--r-- 1 root root   21M Feb 10 16:02 SRR5836473_1.fastq.gz.2\n",
            "-rw-r--r-- 1 root root  896K Feb 10 15:13 SRR5836473_2.fastq.gz\n",
            "-rw-r--r-- 1 root root  3.1M Feb 10 16:01 SRR5836473_2.fastq.gz.1\n",
            "-rw-r--r-- 1 root root  1.2G Feb 10 16:22 SRR5836475_1_bismark_bt2_pe.bam\n"
          ]
        }
      ],
      "source": [
        "! ls -lh"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "1yuRuud-mMK-"
      },
      "source": []
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "6RQ7FwM-UsSi",
        "outputId": "35da1d1c-f80a-4f1d-b3ae-ded382fd3ce0"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "\n",
            "Writing Bismark HTML report to >> SRR5836473_1_bismark_bt2_PE_report.html <<\n",
            "\n",
            "==============================================================================================================\n",
            "Using the following alignment report:\t\t> SRR5836473_1_bismark_bt2_PE_report.txt <\n",
            "Processing alignment report SRR5836473_1_bismark_bt2_PE_report.txt ...\n",
            "Complete\n",
            "\n",
            "No deduplication report file specified, skipping this step\n",
            "No splitting report file specified, skipping this step\n",
            "Using the following M-bias report:\t\t> s_8_cell.deduplicated.M-bias.txt <\n",
            "Processing M-bias report s_8_cell.deduplicated.M-bias.txt ...\n",
            "Complete\n",
            "\n",
            "No nucleotide coverage report file specified, skipping this step\n",
            "==============================================================================================================\n",
            "\n",
            "\n"
          ]
        }
      ],
      "source": [
        "! bismark2report -alignment_report SRR5836473_1_bismark_bt2_PE_report.txt -mbias_report s_8_cell.deduplicated.M-bias.txt"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "8u_wrHEL2f9F",
        "outputId": "48c80ede-2e49-412a-f058-3cdb049d9aa2"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "\n",
            "Writing Bismark HTML report to >> SRR5836475_1_bismark_bt2_PE_report.html <<\n",
            "\n",
            "==============================================================================================================\n",
            "Using the following alignment report:\t\t> SRR5836475_1_bismark_bt2_PE_report.txt <\n",
            "Processing alignment report SRR5836475_1_bismark_bt2_PE_report.txt ...\n",
            "Complete\n",
            "\n",
            "No deduplication report file specified, skipping this step\n",
            "No splitting report file specified, skipping this step\n",
            "Using the following M-bias report:\t\t> icm_cell.deduplicated.M-bias.txt <\n",
            "Processing M-bias report icm_cell.deduplicated.M-bias.txt ...\n",
            "Complete\n",
            "\n",
            "No nucleotide coverage report file specified, skipping this step\n",
            "==============================================================================================================\n",
            "\n",
            "\n"
          ]
        }
      ],
      "source": [
        "! bismark2report -alignment_report SRR5836475_1_bismark_bt2_PE_report.txt -mbias_report icm_cell.deduplicated.M-bias.txt"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "bF6MOpWW2qc2",
        "outputId": "70ef9976-ae82-4da7-90f4-4ee2eb679ca8"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "\n",
            "Writing Bismark HTML report to >> SRR3824222_1_bismark_bt2_PE_report.html <<\n",
            "\n",
            "==============================================================================================================\n",
            "Using the following alignment report:\t\t> SRR3824222_1_bismark_bt2_PE_report.txt <\n",
            "Processing alignment report SRR3824222_1_bismark_bt2_PE_report.txt ...\n",
            "Complete\n",
            "\n",
            "No deduplication report file specified, skipping this step\n",
            "No splitting report file specified, skipping this step\n",
            "Using the following M-bias report:\t\t> epiblast_cell.deduplicated.M-bias.txt <\n",
            "Processing M-bias report epiblast_cell.deduplicated.M-bias.txt ...\n",
            "Complete\n",
            "\n",
            "No nucleotide coverage report file specified, skipping this step\n",
            "==============================================================================================================\n",
            "\n",
            "\n"
          ]
        }
      ],
      "source": [
        "! bismark2report -alignment_report SRR3824222_1_bismark_bt2_PE_report.txt -mbias_report epiblast_cell.deduplicated.M-bias.txt"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "5qtv-9jI4C4R"
      },
      "outputs": [],
      "source": [
        "! gzip -d s_8_cell.deduplicated.bedGraph.gz"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "tI5AUYTPliqA",
        "outputId": "3730094a-947a-4e55-8569-a08a3f56a2a1"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "track type=bedGraph\n",
            "chr11\t3050030\t3050031\t50\n",
            "chr11\t3050047\t3050048\t66.6666666666667\n",
            "chr11\t3050048\t3050049\t50\n",
            "chr11\t3050223\t3050224\t87.5\n",
            "chr11\t3050224\t3050225\t53.3333333333333\n",
            "chr11\t3050285\t3050286\t66.6666666666667\n",
            "chr11\t3050286\t3050287\t33.3333333333333\n",
            "chr11\t3050309\t3050310\t62.5\n",
            "chr11\t3050310\t3050311\t50\n"
          ]
        }
      ],
      "source": [
        "! head s_8_cell.deduplicated.bedGraph"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "Io6TTpKS4OJt",
        "outputId": "0c06bb04-d8ed-4ddb-a6ab-bf7482e28017"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "track type=bedGraph\n",
            "chr11\t3050030\t3050031\t33.3333333333333\n",
            "chr11\t3050031\t3050032\t50\n",
            "chr11\t3050047\t3050048\t50\n",
            "chr11\t3050048\t3050049\t57.1428571428571\n",
            "chr11\t3050223\t3050224\t36.3636363636364\n",
            "chr11\t3050224\t3050225\t22.7272727272727\n",
            "chr11\t3050285\t3050286\t25\n",
            "chr11\t3050286\t3050287\t28.5714285714286\n",
            "chr11\t3050309\t3050310\t22.2222222222222\n"
          ]
        }
      ],
      "source": [
        "! gzip -d icm_cell.deduplicated.bedGraph.gz\n",
        "! head icm_cell.deduplicated.bedGraph"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "DAn5M1rP4WDk",
        "outputId": "36328289-942d-49e0-f7d2-687b1e87ca87"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "track type=bedGraph\n",
            "chr11\t3050030\t3050031\t75\n",
            "chr11\t3050031\t3050032\t100\n",
            "chr11\t3050047\t3050048\t100\n",
            "chr11\t3050048\t3050049\t100\n",
            "chr11\t3050223\t3050224\t96.875\n",
            "chr11\t3050224\t3050225\t93.75\n",
            "chr11\t3050285\t3050286\t93.3333333333333\n",
            "chr11\t3050286\t3050287\t90\n",
            "chr11\t3050309\t3050310\t96.7741935483871\n"
          ]
        }
      ],
      "source": [
        "! gzip -d epiblast_cell.deduplicated.bedGraph.gz\n",
        "! head epiblast_cell.deduplicated.bedGraph"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "eWLyRpDo6xTn"
      },
      "outputs": [],
      "source": [
        "import pandas as pd\n",
        "import seaborn as sns\n",
        "import matplotlib.pyplot as plt"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 206
        },
        "id": "EzaU9JEb6tjj",
        "outputId": "b4dfcb8e-352d-4b43-d792-be9d7420f7bf"
      },
      "outputs": [
        {
          "data": {
            "text/html": [
              "\n",
              "  <div id=\"df-091c4bf9-3150-4baa-bd3c-8082a29e67ac\" class=\"colab-df-container\">\n",
              "    <div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>0</th>\n",
              "      <th>1</th>\n",
              "      <th>2</th>\n",
              "      <th>3</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>0</th>\n",
              "      <td>chr11</td>\n",
              "      <td>3050030</td>\n",
              "      <td>3050031</td>\n",
              "      <td>50.000000</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1</th>\n",
              "      <td>chr11</td>\n",
              "      <td>3050047</td>\n",
              "      <td>3050048</td>\n",
              "      <td>66.666667</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2</th>\n",
              "      <td>chr11</td>\n",
              "      <td>3050048</td>\n",
              "      <td>3050049</td>\n",
              "      <td>50.000000</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3</th>\n",
              "      <td>chr11</td>\n",
              "      <td>3050223</td>\n",
              "      <td>3050224</td>\n",
              "      <td>87.500000</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>4</th>\n",
              "      <td>chr11</td>\n",
              "      <td>3050224</td>\n",
              "      <td>3050225</td>\n",
              "      <td>53.333333</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "</div>\n",
              "    <div class=\"colab-df-buttons\">\n",
              "\n",
              "  <div class=\"colab-df-container\">\n",
              "    <button class=\"colab-df-convert\" onclick=\"convertToInteractive('df-091c4bf9-3150-4baa-bd3c-8082a29e67ac')\"\n",
              "            title=\"Convert this dataframe to an interactive table.\"\n",
              "            style=\"display:none;\">\n",
              "\n",
              "  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\" viewBox=\"0 -960 960 960\">\n",
              "    <path d=\"M120-120v-720h720v720H120Zm60-500h600v-160H180v160Zm220 220h160v-160H400v160Zm0 220h160v-160H400v160ZM180-400h160v-160H180v160Zm440 0h160v-160H620v160ZM180-180h160v-160H180v160Zm440 0h160v-160H620v160Z\"/>\n",
              "  </svg>\n",
              "    </button>\n",
              "\n",
              "  <style>\n",
              "    .colab-df-container {\n",
              "      display:flex;\n",
              "      gap: 12px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert {\n",
              "      background-color: #E8F0FE;\n",
              "      border: none;\n",
              "      border-radius: 50%;\n",
              "      cursor: pointer;\n",
              "      display: none;\n",
              "      fill: #1967D2;\n",
              "      height: 32px;\n",
              "      padding: 0 0 0 0;\n",
              "      width: 32px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert:hover {\n",
              "      background-color: #E2EBFA;\n",
              "      box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "      fill: #174EA6;\n",
              "    }\n",
              "\n",
              "    .colab-df-buttons div {\n",
              "      margin-bottom: 4px;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert {\n",
              "      background-color: #3B4455;\n",
              "      fill: #D2E3FC;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert:hover {\n",
              "      background-color: #434B5C;\n",
              "      box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n",
              "      filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n",
              "      fill: #FFFFFF;\n",
              "    }\n",
              "  </style>\n",
              "\n",
              "    <script>\n",
              "      const buttonEl =\n",
              "        document.querySelector('#df-091c4bf9-3150-4baa-bd3c-8082a29e67ac button.colab-df-convert');\n",
              "      buttonEl.style.display =\n",
              "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "\n",
              "      async function convertToInteractive(key) {\n",
              "        const element = document.querySelector('#df-091c4bf9-3150-4baa-bd3c-8082a29e67ac');\n",
              "        const dataTable =\n",
              "          await google.colab.kernel.invokeFunction('convertToInteractive',\n",
              "                                                    [key], {});\n",
              "        if (!dataTable) return;\n",
              "\n",
              "        const docLinkHtml = 'Like what you see? Visit the ' +\n",
              "          '<a target=\"_blank\" href=https://colab.research.google.com/notebooks/data_table.ipynb>data table notebook</a>'\n",
              "          + ' to learn more about interactive tables.';\n",
              "        element.innerHTML = '';\n",
              "        dataTable['output_type'] = 'display_data';\n",
              "        await google.colab.output.renderOutput(dataTable, element);\n",
              "        const docLink = document.createElement('div');\n",
              "        docLink.innerHTML = docLinkHtml;\n",
              "        element.appendChild(docLink);\n",
              "      }\n",
              "    </script>\n",
              "  </div>\n",
              "\n",
              "\n",
              "<div id=\"df-9ccf1d84-4268-4b65-a63a-14100eaec2d1\">\n",
              "  <button class=\"colab-df-quickchart\" onclick=\"quickchart('df-9ccf1d84-4268-4b65-a63a-14100eaec2d1')\"\n",
              "            title=\"Suggest charts\"\n",
              "            style=\"display:none;\">\n",
              "\n",
              "<svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n",
              "     width=\"24px\">\n",
              "    <g>\n",
              "        <path d=\"M19 3H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zM9 17H7v-7h2v7zm4 0h-2V7h2v10zm4 0h-2v-4h2v4z\"/>\n",
              "    </g>\n",
              "</svg>\n",
              "  </button>\n",
              "\n",
              "<style>\n",
              "  .colab-df-quickchart {\n",
              "      --bg-color: #E8F0FE;\n",
              "      --fill-color: #1967D2;\n",
              "      --hover-bg-color: #E2EBFA;\n",
              "      --hover-fill-color: #174EA6;\n",
              "      --disabled-fill-color: #AAA;\n",
              "      --disabled-bg-color: #DDD;\n",
              "  }\n",
              "\n",
              "  [theme=dark] .colab-df-quickchart {\n",
              "      --bg-color: #3B4455;\n",
              "      --fill-color: #D2E3FC;\n",
              "      --hover-bg-color: #434B5C;\n",
              "      --hover-fill-color: #FFFFFF;\n",
              "      --disabled-bg-color: #3B4455;\n",
              "      --disabled-fill-color: #666;\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart {\n",
              "    background-color: var(--bg-color);\n",
              "    border: none;\n",
              "    border-radius: 50%;\n",
              "    cursor: pointer;\n",
              "    display: none;\n",
              "    fill: var(--fill-color);\n",
              "    height: 32px;\n",
              "    padding: 0;\n",
              "    width: 32px;\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart:hover {\n",
              "    background-color: var(--hover-bg-color);\n",
              "    box-shadow: 0 1px 2px rgba(60, 64, 67, 0.3), 0 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "    fill: var(--button-hover-fill-color);\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart-complete:disabled,\n",
              "  .colab-df-quickchart-complete:disabled:hover {\n",
              "    background-color: var(--disabled-bg-color);\n",
              "    fill: var(--disabled-fill-color);\n",
              "    box-shadow: none;\n",
              "  }\n",
              "\n",
              "  .colab-df-spinner {\n",
              "    border: 2px solid var(--fill-color);\n",
              "    border-color: transparent;\n",
              "    border-bottom-color: var(--fill-color);\n",
              "    animation:\n",
              "      spin 1s steps(1) infinite;\n",
              "  }\n",
              "\n",
              "  @keyframes spin {\n",
              "    0% {\n",
              "      border-color: transparent;\n",
              "      border-bottom-color: var(--fill-color);\n",
              "      border-left-color: var(--fill-color);\n",
              "    }\n",
              "    20% {\n",
              "      border-color: transparent;\n",
              "      border-left-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "    }\n",
              "    30% {\n",
              "      border-color: transparent;\n",
              "      border-left-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "      border-right-color: var(--fill-color);\n",
              "    }\n",
              "    40% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "    }\n",
              "    60% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "    }\n",
              "    80% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "      border-bottom-color: var(--fill-color);\n",
              "    }\n",
              "    90% {\n",
              "      border-color: transparent;\n",
              "      border-bottom-color: var(--fill-color);\n",
              "    }\n",
              "  }\n",
              "</style>\n",
              "\n",
              "  <script>\n",
              "    async function quickchart(key) {\n",
              "      const quickchartButtonEl =\n",
              "        document.querySelector('#' + key + ' button');\n",
              "      quickchartButtonEl.disabled = true;  // To prevent multiple clicks.\n",
              "      quickchartButtonEl.classList.add('colab-df-spinner');\n",
              "      try {\n",
              "        const charts = await google.colab.kernel.invokeFunction(\n",
              "            'suggestCharts', [key], {});\n",
              "      } catch (error) {\n",
              "        console.error('Error during call to suggestCharts:', error);\n",
              "      }\n",
              "      quickchartButtonEl.classList.remove('colab-df-spinner');\n",
              "      quickchartButtonEl.classList.add('colab-df-quickchart-complete');\n",
              "    }\n",
              "    (() => {\n",
              "      let quickchartButtonEl =\n",
              "        document.querySelector('#df-9ccf1d84-4268-4b65-a63a-14100eaec2d1 button');\n",
              "      quickchartButtonEl.style.display =\n",
              "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "    })();\n",
              "  </script>\n",
              "</div>\n",
              "    </div>\n",
              "  </div>\n"
            ],
            "text/plain": [
              "       0        1        2          3\n",
              "0  chr11  3050030  3050031  50.000000\n",
              "1  chr11  3050047  3050048  66.666667\n",
              "2  chr11  3050048  3050049  50.000000\n",
              "3  chr11  3050223  3050224  87.500000\n",
              "4  chr11  3050224  3050225  53.333333"
            ]
          },
          "execution_count": 99,
          "metadata": {},
          "output_type": "execute_result"
        }
      ],
      "source": [
        "df = pd.read_csv(\"s_8_cell.deduplicated.bedGraph\", delimiter='\\t', skiprows=1, header=None)\n",
        "df.head()"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 547
        },
        "id": "okmXhIZ68L3_",
        "outputId": "60f7b230-7eba-4ba1-ceb5-fadf04eca214"
      },
      "outputs": [
        {
          "data": {
            "image/png": "iVBORw0KGgoAAAANSUhEUgAAA9UAAAISCAYAAADC0OynAAAAOXRFWHRTb2Z0d2FyZQBNYXRwbG90bGliIHZlcnNpb24zLjcuMSwgaHR0cHM6Ly9tYXRwbG90bGliLm9yZy/bCgiHAAAACXBIWXMAAA9hAAAPYQGoP6dpAAA/50lEQVR4nO3de5hVZd0/4M8MiMZBpAHxNZRCE5CDgmfUPISWmmmYJkWa4oulWSkkHlLDNCjPmOYBNVKTSNHyLOYxRbHEVFJT0RIlwAFFhFcOM78/+M3kOAPMLJAZ4b6vy+ty1nrWXt+997P25rPXs55VUllZWRkAAACgwUobuwAAAAD4pBKqAQAAoCChGgAAAAoSqgEAAKAgoRoAAAAKEqoBAACgIKEaAAAAChKqAQAAoCChGgAAAAoSqoFPvK5du9b4r1u3btl+++3zzW9+M3/4wx9SWVnZ2CUCrBNmzpyZs88+O/vss0969uyZbbbZJgceeGBGjx6d+fPnr/F6nnzyyXTt2jWnnHJKjeUTJkxI165dc+mll67xmoC1T/PGLgBgdfna176WJFm6dGneeOONPP300/nb3/6WSZMm5cILL2zk6gDWbq+//noGDhyYOXPm5DOf+Uz22muvfPDBB5kyZUouu+yy3HvvvRk3blzatGnT2KUCrFZCNbDWGDVqVI2/H3vssQwZMiR33nlnDjzwwOy1116NVBnA2u/888/PnDlz8s1vfjM/+clP0qxZsyTJe++9l2OOOSbPPPNMrrvuuvzgBz9o5EoBVi/Dv4G11q677pqvfvWrSZL777+/kasBWLs99dRTSZLjjjuuOlAnSZs2bXLMMcckSZ577rlGqQ3g4yRUA2u1rbfeOknyn//8p3rZrFmzcvXVV2fQoEHZfffd07Nnz+y66675/ve/n2effXa5j7VgwYJcddVVGTBgQPr27Zttt902X/7ylzNixIi89tpr1e2qrtVb2X8fvZbv29/+drp27Zrp06fnj3/8YwYMGJBtttkmu+yyS4YPH56ZM2cut7ZHHnkkQ4YMyc4775yePXvmi1/8YkaOHJm5c+cud5uV1TlhwoQ6t5sxY0bOPvvs9O/fP7169cqOO+6YY489Nk8//fRy91V1XWN9X4skWbhwYa688socfPDB6dOnT/r06ZPDDjsst956a5376Nq1a/bee+/l1lDX+hVdV3n33Xev8LVoaH3L8+H34ec//3mdbR588MHqNh+9NjRJKisrc8cdd+SII47IDjvskF69emW//fbLpZdemoULF9ZoW5++WfVflUsvvXS5r8OYMWOq2z/55JN11r+ifXz729+u0faUU05Z7mONGDGiervp06dXL58+fXr1Y82fPz/nnHNO9thjj+rX4Te/+U0qKirqrG3GjBk588wzs9dee6Vnz57ZZZddlvtZULWfD/9Xdbydeuqpef3112tt88ILL+SXv/xlBgwYUOP4/OlPf1rnMf3h51KX5a1flffotttuy8CBA7PddtulW7duNZ7fio6pj2rRosVK22y00Ua1li1evDg33XRTBg4cmO233z69e/fOPvvsk1NPPTXPP/98rfavvvpqTjnllOyxxx7p2bNn+vXrlxNPPDEvv/xyvWsFWJ0M/wbWau+//36SZL311qte9uc//znnn39+Pve5z6Vr165p3bp1/vWvf2XixIl56KGHcsUVV2S33Xar8TizZs3K0UcfnZdffjlt27bNjjvumBYtWmT69OkZN25cOnfunM997nM1tunWrVu6d+9eq6Z//etfKwyg1157bX73u99l++23zxe/+MU888wzue222/LEE0/k97//fTbZZJMa7c8///xcffXVWW+99dKrV6906NAhL730Un7zm9/kgQceyE033ZT27dsvd38frfOFF17Iiy++WGfbKVOm5Nhjj827776bz33uc9lzzz0zZ86c/OUvf8mjjz6a888/P/vvv/9y97X55ptnu+22W+lrUV5enqOOOiovvfRSOnTokB122CGVlZWZMmVKTjnllDz//PM544wzlrufVbVo0aKcf/75y13/cdV32223ZejQoVl//fVrLL/pppuWu01FRUV+/OMf54477kjLli3Ts2fPtG3bNs8//3x+9atf5ZFHHsn111+fDTbYIMl/5x6oUvV+9+3bN507d25QvXPmzMkVV1xRr7YtW7bMl770peq/FyxYkHvvvbfe+3rllVcyfvz4FbZZtGhRjjjiiLzxxhvZeeeds3jx4kyaNCkjR47Miy++WOsSkZdeeilHHnlk5s6dm8997nPZd99989Zbb2XixIl58MEHc/7552e//fartZ/27dtn9913T5J88MEHee655zJhwoTcf//9ueOOO9KxY8fqtldddVXuu+++dO3atbrvv/DCC7npppty//3355ZbbqnRfnVb2Xs0duzY/PznP09JSUn69OmTTp06VZ9lbugPRLvttlsmTJiQyy+/vNbw7zFjxiRJDjnkkBrbLFiwIEOGDMlTTz2Vli1bpm/fvtlwww3z5ptv5vbbb0/r1q3Ts2fP6vb3339/TjzxxCxatCjdu3fPNttsk//85z+5++678+CDD+bqq6/ODjvs0KC6AVaVUA2stSorK/PQQw8lSY2zbn379s0dd9yRz3/+8zXaP/roo/ne976XESNG5L777ktJSUn1upNPPjkvv/xy9ttvv5x77rlp1apV9brp06fXOatt//79c8IJJ9RaPmHChBWG6t///ve58sors8ceeyRZdhbn1FNPze23356zzz47l19+eXXbu+++O1dffXW22mqr/OpXv6oORZWVlbn00ktz2WWX5dxzz81FF11Uaz9VZ+4+Wuell15aZ6ieP39+TjjhhMyfPz/nnXde9dD6ZNmQzsGDB+f000/PzjvvnE9/+tN17mv77bfPyJEjV/panHrqqXnppZdyxBFH5Mc//nH1GbC33347xx57bG644Ybsscce+cIXvrDc13FV/OY3v8n06dOzySab1Bjl8HHWt8MOO+Spp57KXXfdVSP4vvnmm3n00Uez4447ZvLkybW2u/baa3PHHXdkxx13zIUXXpgOHTokWRYwR4wYkZtvvjm/+tWvMmzYsCS15x6oer8PPfTQDBgwoN71Jskll1yS9957b7mvU7Js4sAkadeuXY19T58+vUGhetSoUamoqEiHDh0ye/bsOts888wz6dq1a+69997qPvjvf/873/rWt3Lrrbemf//+6d+/f5Jlx8iwYcMyd+7cHHPMMRk2bFj1MX/vvffmRz/6UU477bRst9122XjjjWvsp0uXLjWeS0VFRX7wgx9k4sSJue+++2qcRf7GN76R008/vcYPWxUVFbn88stz6aWX5uKLL65xTKxuK3uPxo4dmyS58MILa/0g1tBQfdJJJ+X555/P7373uzz88MPp2bNnPvjggzz99NNZf/31c95552XnnXeusc25556bp556KjvssENGjx5d47Pj7bffzptvvln99/Tp0/PjH/84zZs3z5VXXpl+/fpVr3vkkUdy3HHH5cc//nHuu+++ep01B1hdDP8G1jpLly7N66+/ntNOOy1TpkxJixYtapwd6dq1a61AnSS77757vvzlL+ff//53/vnPf1Yvf/bZZzNp0qSUlZXlnHPOqRGok6RTp07p1q3baqv/y1/+cnWgTpadZT/99NPzqU99Kg888EBmzJhRva7qDNQFF1xQ4yxjSUlJTjjhhHTv3j333ntv5syZU2s/H3zwQZKkefP6/b568803Z/bs2TnyyCNrBOok6dWrV4477rgsWLAgf/rTn2ptu3jx4nrv64UXXsjDDz+cXr165dRTT63xj+P27dvnZz/7WZIVn71dFeXl5bnyyivTpUuXHHzwwWusvp122ildunTJuHHjaiz//e9/n5KSkhx66KG1tlmyZEnGjBmTli1b5qKLLqoO1MmyobhnnHFGOnTokPHjxy93+HNR//znP/OHP/wh/fr1yy677LLcdkuWLElSc7RIQz388MN59NFHM2DAgFojQj5q+PDhNYLZ5ptvnuOOOy5JcsMNN1Qvf/LJJ/PPf/4zm266aX70ox/V+BHtS1/6Uvr3758FCxbklltuWWl9paWl2X777ZP890eEKjvvvHOtkSKlpaX5/ve/n44dO+aBBx5Y6eMXVZ/3qOoHioYM816eDh065Prrr89uu+2WN998M/fee28eeuihzJs3L3369EmPHj1qtJ85c2ZuvfXWtGjRIr/4xS9q/RjXvn37bLPNNtV/jx07NgsWLMjQoUNrBOok+cIXvpDDDz88M2bMyMMPP7zKzwWgIYRqYK1RdQ3g1ltvnS996UuZMGFCWrVqlQsvvDCbb755jbaLFi3K/fffn4suuihnnHFGTjnllJxyyinVYfpf//pXddvHH388SXLAAQekdevWH/vzOOCAA2ota9euXXbddddUVlbmb3/7W5Jl4e/FF1/MZz/72Wy11Va1tikpKUnfvn2zdOnSTJ06tdb6//u//0uS6mHBK/PYY48lSfbZZ58611cNba3rWtRFixYlqd81l3/5y1+SLDuDXlpa+2tq6623TsuWLT+2CY8uvvjizJ8/P6ecckqdPwJ8nPV94xvfyDPPPFM9UmDx4sW55ZZbsueee9Y5RPgf//hH5s6dmz59+tQ5xH+DDTZIjx498u6779Z5ve+qqDpTW9c13h9W9eNN0TOHS5YsyS9+8Yu0atUqJ5544grbbrTRRtl1111rLf/KV76SZNnlC1U/Lvz1r39NsuxHrLoC/0EHHVSj3fJ88MEHefLJJzN27Nist956NX4QqzJ37tzccsstGTVqVE477bTqz5slS5bknXfeyTvvvLPCfRRVn/foM5/5TJJk4sSJq7y/F198MQcddFBee+21XH755XnqqafyyCOP5LTTTssjjzySb37zm5k2bVp1+8mTJ2fp0qXZfffdq+tYkZV9BlX9sLGiuTEAPg6GfwNrjaohsyUlJWndunW22mqr7Lvvvmnbtm2Ndi+99FK+973v1RhW+FFV12InqT4z/NFg/nHZdNNN61xe9Y/OWbNmJUl1/a+//nqN4e11qWvCsqozVB99fZanan8DBw5s8L6qQkN97k9btZ+LLrqozmHrVaqC+ke3XdlrsSIvvvhibr755uy2227ZY4896vzH+arUtzJf+9rXctFFF+Wmm27KiBEjMnHixLz99ts5/PDD62xfNVnXY489VqgPFPXQQw/lscceyze+8Y2V7vfdd99NUr/3vi6///3v8+qrr+akk05a4dwAyfKPnTZt2mTDDTfMvHnz8u6776Zdu3bVx1GnTp3q3Oajx9uHTZ48udbz/uxnP5sbbrih1pn0O+64I2eccUYWLFiw3Lrff//9WhN41bWPhqjve/Td7343p5xySoYNG5Ybb7wxm222WY2Zu+tr8eLF+eEPf5hZs2bl5ptvrj4rveGGG+bII49MRUVFRo0aldGjR+fiiy9O0vDP1qpjb2WXVazOvg5QH0I1sNb46LWidamsrMyPfvSjvPnmmzn88MMzcODAdOrUKa1atUpJSUkuvPDCXHnllamsrFwDFa+aqjNuHTp0qDWx2kfVFTaq/oG6vCCyvP196UtfSsuWLZfbrkuXLrWWVQX4lYWiD+9nu+22a/APGR+dDOvD6nN96KhRo1JSUrLCM3urUt/KtG3bNvvvv39uv/32nHzyybnpppvSqVOn7L777nVeT13VTzt37py+ffuu8LHrmnW5iKozx61bt84Pf/jDlbavCqX1ee8/at68ebn00kvzmc98JkcddVSDt18VHx4O/lEfnqhs8eLFmT59ep555pkMHz48V111VfWlGG+++WZ1XzrttNOqRxxUjQ45/PDDM2XKlDo/bz68jw+rzwRvDXmPDj744GyyySb59a9/nSeeeCJTpkxZYfvl+fvf/57XX389m2++ea1h3smyEQGjRo1a6Zn/Fak69j462d5HfXjIOMCaIFQD65Rp06Zl2rRp6dmzZ0aMGFFr/RtvvFFr2f/8z/8kWTbh0Zrw1ltv1XmN9ltvvZUk1ZMmVc0C/tEJoOrr+eefT0lJSb2vB99kk03y2muvZciQITVm462PqluO1RW469pPsmx49dFHH92g/azotVhZqP7zn/+cSZMmZeDAgXVec7866quPgQMHZsKECbnkkksyefLkDB06dLkBr2pI+Ecnzvo4/e53v8u0adMybNiwlJWVrbR91Xu/smuh63LZZZdl7ty5OfPMM+s1fLzqGPmo+fPnZ968edlggw2y4YYbJvnvcbS8EStVowA+OklZUvfrfdddd+XEE0/M2WefnWuuuSbJsmvBFy9enKOPPjpHHnlkrcep6/NmRfuoqmtlobqh79HOO+9cfb35RRddVD1hWUPOlFdNgra8EQlVy6tGLiQN/2zdZJNN8u9//zvDhw9Pu3bt6l0bwMfNNdXAOqXqH3QfvS1V1bqq66c/rGpCnDvvvLPGsPCPy913311r2TvvvJPHHnus+jrpZNlz6NKlS1555ZUa98mujxdffDFvvfVWtt5663qfwax6HRp67WVFRUUmT56c5s2b1yuMV10Tuzqu8ayvJUuW5Je//GXatGmTH/zgByts+3HX17t37/To0aP6Gt2P3oLoo23btGmTyZMnf2zX5X7YvHnzctlll2WzzTarMyTW5YknnkiS9OnTp0H7+te//pUbb7wx22233Qpv0/Zh77zzTiZNmlRr+Z133pkk2XbbbauHNlddf3vPPffUmlwsSfWEe1XtVqZ///4pKSmpMZv9vHnzktT9efPUU0/l7bffrtdjN0SR9+iPf/xjJk6cmC9/+cv1fq0/qmokwmuvvVbn3RCq5hj48LXTO+64Y5o1a5a//OUvNSZgXJ6qz6D777+/UI0AHxehGlindO7cOaWlpXniiSdqTNz0wQcf5KyzzqozmPTu3Ts77bRTysvLc+aZZ9a6NnL69Ol56aWXVluNd999dx599NHqv5csWZKRI0dmwYIF2XPPPWsM1z7uuOOqb+fzwgsv1HqsuXPn1rq3b2VlZS677LIkte8ZuyKHH354ysrKcs011+T3v/99rdmklyxZkkcffbTGzOkVFRW5+OKLM2PGjOyxxx7VZwlXZJtttsmuu+6ap59+OiNGjKjzH+gvvvhiHnnkkXrXvjITJkzI66+/nuOOO67WDMSNUd93v/vd7LnnnhkyZMgKzzS2aNEixxxzTN5///2ccMIJdZ75nDlzZm677bbCtXzYNddck3feeafGbcRW5IEHHsjdd9+d9u3b15qteWUuvfTSLFmyJKeeemqDtvvFL35R45raN954o7q/f+tb36pevtNOO2WrrbbKm2++mdGjR9cYgj1x4sRMnDgxLVu2rPcxMnHixFRWVtaYo+Czn/1skmUB/cOfGzNnzsxZZ53VoOdVXw19j2bOnJlzzz03ZWVlq1RTnz59UlZWlgULFuRnP/tZjTkFZs6cmZ///OdJUuPyjI4dO+aggw7KBx98kOHDh9e6Frq8vDx///vfq/8++uijs8EGG+QXv/hF7rvvvlo1LFq0KPfcc89yb+8G8HEx/BtYp5SVleXrX/96xo8fn4MOOig777xz1l9//fztb3/L0qVLM2DAgEyYMKHWduedd16OPPLI3HHHHfnLX/6Svn37pkWLFnnjjTfywgsvZPjw4as0qdCHHXbYYfnf//3f7LDDDunQoUP+/ve/Z/r06dl4441z5pln1mh74IEH5pVXXskVV1yRAQMGpHv37tlss81SWVmZN954Iy+99FJatmyZww47LMmymYzPP//86usmJ0+eXGuW6qpw/oc//CEvv/xyhg8fnmTZhEOXX355vvvd7+bMM8/Mr3/963z+85/PhhtumLfffjv/+Mc/qs+SbbXVVnnyySdz2mmnZfr06enYsWODwtF5552XY445Jr/73e9yxx13pFu3btl4440zf/78vPTSS5kxY0aOOOKI1Xaf6lmzZqVz584ZNGhQk6hv3333zb777luvtkOGDMm0adPyxz/+Mfvtt1+23nrrdOrUKYsXL85rr72WV155JV27dq3z9mANNWvWrOywww7LvW69ypw5czJkyJA899xzadGiRc4555wGz/49a9asHHzwwenVq1e9t9l2222zePHi7Lvvvtl5552zePHiPPHEE1m4cGG++tWv1nhNS0pKcv755+eII47IFVdckYkTJ6Z79+5566238vTTT6d58+Y599xz6xz+PW3atOprpZcsWZI33ngjzzzzTJLkf//3f6vb7b333vn85z+f559/Pvvuu2/69u1bPVt4t27d0qdPn8LXMC9Pfd+jKqeffnrefffdXHrppSv9QWlF1l9//Zx99tn54Q9/mNtuuy2TJk2qvk/1lClT8v7776dHjx4ZMmRIrf2/9tprefLJJ7P33ntn++23T+vWrfPWW29l6tSpGThwYPU10p07d84FF1yQYcOG5YQTTkjnzp3TpUuXtGzZMjNnzsw//vGPLFiwILfddludowMAPi5CNbDO+elPf5ouXbrk5ptvzqRJk9KmTZvssssuOfHEE+sM1MmyMyo333xzxo4dm3vvvTePP/54SktLs8kmm+Sb3/xm9tprr9VW39FHH52ePXvmt7/9bf7+97/nU5/6VA466KCcdNJJdf5D8cQTT8xuu+2WG264IU8//XT++c9/plWrVunYsWMGDhyYL3/5y9Vt//3vf9f4R/w999yz3DqefvrpzJw5szpUJ8tCy+23357f/OY3efjhh/PUU08lWTZZ2g477JB99tmn+n64//nPf9KiRYsMHjw4Rx99dIMmqiorK8u4ceMyfvz43HnnnXnhhRcyZcqUtG/fPptttlm+/e1v13nrsVVx8skn1zv4NUZ9y1NaWppf/vKX+dKXvpTx48fnueeeyz/+8Y9suOGG2WSTTTJ48ODCQ3o/qqSkpF4/jixcuDCzZ8/OV7/61QwePLjQfdw/9alP5aSTTmrQNi1atMiYMWNy4YUX5v7778/cuXPTqVOnHHbYYXUOhe7atWtuvfXW/PrXv86jjz6ae++9N61bt07//v1z7LHHpnfv3nXu5+23366+Rr9Zs2bp0KFDdt9991o/pLRo0SI33nhjLrroojzyyCN58MEH07FjxwwaNCjHH398rYC5OtT3PUqS8ePH59FHH81XvvKVev+IsyL9+/fPH/7wh1xzzTX561//mkceeSTrrbdeOnfunP322y9HHnlkrVv4tW7dOr/97W8zbty43H777fnrX/+aioqKbLzxxjnwwANr/RjUv3///OlPf8p1112Xxx9/PI8//niaN2+ejTfeOHvttVf22WefbLHFFqv8XAAaoqTykzDFLcA64Nvf/nYmT56cP//5z8u9zc+qmjBhQk499dR67eOUU07J5MmT88ADD3wstcDqMn369Hzxi1/MjjvumOuvv76xywFgHeOaagAAACjI8G+Adch2222XkSNH1ut2NF//+tez5557fvxFAQB8ggnVAOuQzp07p3PnzvVqW99bCQEArMtcUw0AAAAFuaYaAAAAChKqAQAAoCChGgAAAAoSqgEAAKCgT9Ts3+Xl76UpT6tWUpKUlbVp8nWy7tJHaer0UZo6fZSmTh+lqfsk9dGqWlfmExWqKyvT5F/45JNTJ+sufZSmTh+lqdNHaer0UZq6tamPGv4NAAAABQnVAAAAUJBQDQAAAAUJ1QAAAFCQUA0AAAAFCdUAAABQkFANAAAABQnVAAAAUJBQDQAAAAUJ1QAAAFCQUA0AAAAFCdUAAABQkFANAAAABQnVAAAAUJBQDQAAAAUJ1QAAAFCQUA0AAAAFCdUAAABQUPPGLmBt1KzZ8n+rqKioTEVF5RqsBgAAgI+LUL0alZaWZGlFZdq1a7XcNkuWVuTddxYI1gAAAGsBoXo1KikpSbPSkvxw3JS8Mmt+rfVbbtw6lxzeJ6WlJUI1AADAWkCo/hi8Mmt+pr41r7HLAAAA4GNmojIAAAAoSKgGAACAgoRqAAAAKEioBgAAgIKEagAAAChIqAYAAICChGoAAAAoSKgGAACAgoRqAAAAKEioBgAAgIKEagAAAChIqAYAAICChGoAAAAoSKgGAACAgoRqAAAAKEioBgAAgIKEagAAAChIqAYAAICChGoAAAAoSKgGAACAgoRqAAAAKEioBgAAgIKEagAAAChIqAYAAICChGoAAAAoSKgGAACAgoRqAAAAKEioBgAAgIKEagAAAChIqAYAAICChGoAAAAoSKgGAACAgoRqAAAAKEioBgAAgIKEagAAAChIqAYAAICChGoAAAAoSKgGAACAgoRqAAAAKEioBgAAgIKEagAAACioUKi+8cYbs/fee6dXr1459NBD8+yzz9ZruzvvvDNdu3bNcccdV2S3AAAA0KQ0OFTfddddGTlyZI4//vjceuut6datWwYPHpzy8vIVbjd9+vT84he/yPbbb1+4WAAAAGhKGhyqr7vuuhx22GE55JBDsuWWW2bEiBHZYIMNcssttyx3m6VLl2bYsGE54YQTstlmm61SwQAAANBUNChUL1q0KFOnTk2/fv3++wClpenXr1+mTJmy3O0uu+yylJWV5dBDDy1eKQAAADQxzRvSeO7cuVm6dGnKyspqLC8rK8u0adPq3Oavf/1rbr755tx2222Fi6xSUrLKD/Gxakh9Tf25sHaq6nf6H02VPkpTp4/S1OmjNHWfpD5a3xobFKobav78+Tn55JPzs5/9LJ/+9KdX+fHKytqshqoaX7t2rRq7BNZxa8uxxNpLH6Wp00dp6vRRmrq1qY82KFS3a9cuzZo1qzUpWXl5edq3b1+r/RtvvJE333wz3/ve96qXVVRUJEm23nrr3HPPPdl8883rvf/y8vdSWdmQites5s1Ls9FGKw/Mc+e+n6VLK9ZARVBTScmyD7Cmfiyx7tJHaer0UZo6fZSm7pPUR6tqXZkGheoWLVqkR48emTRpUvr3759kWUieNGlSBg0aVKt9ly5dcvvtt9dYdvHFF+f999/P6aefnk022aQhu09lZZr0C9+Q2pry82Dt19SPJdBHaer0UZo6fZSmbm3qow0e/n3UUUdl+PDh6dmzZ3r37p2xY8dm4cKFGTBgQJLk5JNPTseOHTN06NCsv/762WqrrWpsv+GGGyZJreUAAADwSdPgUL3//vtnzpw5GT16dGbPnp3u3btnzJgx1cO/Z8yYkdLSBt+pCwAAAD5xCk1UNmjQoDqHeyfJ9ddfv8JtR40aVWSXAAAA0OQ4pQwAAAAFCdUAAABQkFANAAAABQnVAAAAUJBQDQAAAAUJ1QAAAFCQUA0AAAAFCdUAAABQkFANAAAABQnVAAAAUJBQDQAAAAUJ1QAAAFCQUA0AAAAFCdUAAABQkFANAAAABQnVAAAAUJBQDQAAAAUJ1QAAAFCQUA0AAAAFCdUAAABQkFANAAAABQnVAAAAUJBQDQAAAAUJ1QAAAFCQUA0AAAAFCdUAAABQkFANAAAABQnVAAAAUJBQDQAAAAUJ1QAAAFCQUA0AAAAFCdUAAABQkFANAAAABQnVAAAAUJBQDQAAAAUJ1QAAAFCQUA0AAAAFCdUAAABQkFANAAAABQnVAAAAUJBQDQAAAAUJ1QAAAFCQUA0AAAAFCdUAAABQkFANAAAABQnVAAAAUJBQDQAAAAUJ1QAAAFCQUA0AAAAFCdUAAABQkFANAAAABQnVAAAAUJBQDQAAAAUJ1QAAAFCQUA0AAAAFCdUAAABQkFANAAAABQnVAAAAUJBQDQAAAAUJ1QAAAFCQUA0AAAAFCdUAAABQkFANAAAABQnVAAAAUJBQDQAAAAUJ1QAAAFCQUA0AAAAFCdUAAABQkFANAAAABQnVAAAAUJBQDQAAAAUJ1QAAAFCQUA0AAAAFCdUAAABQkFANAAAABQnVAAAAUJBQDQAAAAUJ1QAAAFCQUA0AAAAFCdUAAABQkFANAAAABQnVAAAAUJBQDQAAAAUJ1QAAAFCQUA0AAAAFCdUAAABQkFANAAAABRUK1TfeeGP23nvv9OrVK4ceemieffbZ5ba97777MmDAgGy//fbZdtttc9BBB+W2224rWi8AAAA0Gc0busFdd92VkSNHZsSIEdlmm20yduzYDB48OPfcc0/KyspqtW/btm2+973vpUuXLllvvfXy4IMP5rTTTktZWVl233331fIkAAAAoDE0+Ez1ddddl8MOOyyHHHJIttxyy4wYMSIbbLBBbrnlljrb77TTTtlnn32yxRZbZPPNN8+RRx6Zrl275m9/+9sqFw8AAACNqUFnqhctWpSpU6fm2GOPrV5WWlqafv36ZcqUKSvdvrKyMk888URee+21DBs2rMHFlpQ0eJM1qiH1NfXnwtqpqt/pfzRV+ihNnT5KU6eP0tR9kvpofWtsUKieO3duli5dWmuYd1lZWaZNm7bc7d5777184QtfyKJFi1JaWpqzzjoru+66a0N2/f/306bB2zRF7dq1auwSWMetLccSay99lKZOH6Wp00dp6tamPtrga6qLaNWqVW677bYsWLAgkyZNyqhRo7LZZptlp512atDjlJe/l8rKj6nI1aB589JstNHKA/Pcue9n6dKKNVAR1FRSsuwDrKkfS6y79FGaOn2Upk4fpan7JPXRqlpXpkGhul27dmnWrFnKy8trLC8vL0/79u2Xu11paWk6d+6cJOnevXteffXVXHXVVQ0O1ZWVadIvfENqa8rPg7VfUz+WQB+lqdNHaer0UZq6tamPNmiishYtWqRHjx6ZNGlS9bKKiopMmjQpffr0qffjVFRUZNGiRQ3ZNQAAADQ5DR7+fdRRR2X48OHp2bNnevfunbFjx2bhwoUZMGBAkuTkk09Ox44dM3To0CTJlVdemZ49e2bzzTfPokWL8vDDD+dPf/pTfvrTn67WJwIAAABrWoND9f777585c+Zk9OjRmT17drp3754xY8ZUD/+eMWNGSkv/ewJ8wYIFGTFiRP7zn/9kgw02SJcuXXLeeedl//33X33PAgAAABpBSWXlJ2ck+9tvN+2L2Zs3L027dq1ywOhHM/WtebXW99h0w9z5g90zd+77WbLERGWseSUlSfv2bZr8scS6Sx+lqdNHaer0UZq6T1Ifrap1ZRp0TTUAAADwX0I1AAAAFCRUAwAAQEFCNQAAABQkVAMAAEBBQjUAAAAUJFQDAABAQUI1AAAAFCRUAwAAQEFCNQAAABQkVAMAAEBBQjUAAAAUJFQDAABAQUI1AAAAFCRUAwAAQEFCNQAAABQkVAMAAEBBQjUAAAAUJFQDAABAQUI1AAAAFCRUAwAAQEFCNQAAABQkVAMAAEBBQjUAAAAUJFQDAABAQUI1AAAAFCRUAwAAQEFCNQAAABQkVAMAAEBBQjUAAAAUJFQDAABAQUI1AAAAFCRUAwAAQEFCNQAAABQkVAMAAEBBQjUAAAAUJFQDAABAQc0buwAAAADWHqWlJSktLWnsMtYYoRoAAIDVorS0JG03apnmzZY/KHppRWVKS0uydGnlGqzs4yNUAwAAsFqUlpakebPS/HDclLwya36t9Vtu3DqXHN4nJSUlSYRqAAAAqOWVWfMz9a15jV3GGmGiMgAAAChIqAYAAICChGoAAAAoSKgGAACAgoRqAAAAKEioBgAAgIKEagAAAChIqAYAAICChGoAAAAoSKgGAACAgoRqAAAAKEioBgAAgIKEagAAAChIqAYAAICChGoAAAAoSKgGAACAgoRqAAAAKEioBgAAgIKEagAAAChIqAYAAICChGoAAAAoSKgGAACAgoRqAAAAKEioBgAAgIKEagAAAChIqAYAAICChGoAAAAoSKgGAACAgoRqAAAAKEioBgAAgIKEagAAAChIqAYAAICChGoAAAAoSKgGAACAgoRqAAAAKEioBgAAgIKEagAAAChIqAYAAICChGoAAAAoSKgGAACAgoRqAAAAKEioBgAAgIKEagAAAChIqAYAAICChGoAAAAoSKgGAACAggqF6htvvDF77713evXqlUMPPTTPPvvsctuOHz8+3/zmN7PDDjtkhx12yHe+850VtgcAAIBPigaH6rvuuisjR47M8ccfn1tvvTXdunXL4MGDU15eXmf7J598MgcccEB++9vfZty4cfmf//mfHH300Zk5c+YqFw8AAACNqcGh+rrrrsthhx2WQw45JFtuuWVGjBiRDTbYILfcckud7S+44IJ861vfSvfu3bPFFlvknHPOSUVFRSZNmrTKxQMAAEBjat6QxosWLcrUqVNz7LHHVi8rLS1Nv379MmXKlHo9xsKFC7NkyZK0bdu2YZUmKSlp8CZrVEPqa+rPhbVTVb/T/2iq9FGaOn2Upk4f5ZOipKTp99P61tegUD137twsXbo0ZWVlNZaXlZVl2rRp9XqM888/PxtvvHH69evXkF3///20afA2TVG7dq0auwTWcWvLscTaSx+lqdNHaer0UZq6jTZaezJRg0L1qrrqqqty11135be//W3WX3/9Bm9fXv5eKis/hsJWk+bNS+vVOebOfT9Ll1asgYqgppKSZV+yTf1YYt2lj9LU6aM0dfooja1Zs9J6nUR85533s2RJ085EVcfTyjQoVLdr1y7NmjWrNSlZeXl52rdvv8Jtr7nmmlx11VW57rrr0q1bt4bstlplZZr0h0NDamvKz4O1X1M/lkAfpanTR2nq9FGaurWpjzZoorIWLVqkR48eNSYZq5p0rE+fPsvd7uqrr87ll1+eMWPGpFevXsWrBQAAgCakwcO/jzrqqAwfPjw9e/ZM7969M3bs2CxcuDADBgxIkpx88snp2LFjhg4dmmTZkO/Ro0fnggsuyGc+85nMnj07SdKyZcu0arX2jKMHAABg3dPgUL3//vtnzpw5GT16dGbPnp3u3btnzJgx1cO/Z8yYkdLS/54AHzduXBYvXpwf/OAHNR7n+9//fk444YRVLB8AAAAaT6GJygYNGpRBgwbVue7666+v8fcDDzxQZBcAAADQ5DXommoAAADgv4RqAAAAKEioBgAAgIKEagAAAChIqAYAAICChGoAAAAoSKgGAACAgoRqAAAAKEioBgAAgIKEagAAAChIqAYAAICChGoAAAAoSKgGAACAgoRqAAAAKEioBgAAgIKEagAAAChIqAYAAICChGoAAAAoSKgGAACAgoRqAAAAKEioBgAAgIKEagAAAChIqAYAAICChGoAAAAoSKgGAACAgoRqAAAAKEioBgAAgIKEagAAAChIqAYAAICChGoAAAAoSKgGAACAgoRqAAAAKEioBgAAgIKEagAAAChIqAYAAICChGoAAAAoSKgGAACAgoRqAAAAKEioBgAAgIKEagAAAChIqAYAAICChGoAAAAoSKgGAACAgoRqAAAAKEioBgAAgIKEagAAAChIqAYAAICChGoAAAAoSKgGAACAgoRqAAAAKEioBgAAgIKEagAAAChIqAYAAICChGoAAAAoSKgGAACAgoRqAAAAKEioBgAAgIKEagAAAChIqAYAAICChGoAAAAoSKgGAACAgoRqAAAAKEioBgAAgIKEagAAAChIqAYAAICChGoAAAAoSKgGAACAgoRqAAAAKEioBgAAgIKEagAAACioeWMXAAB8vEpLS1JaWrLc9RUVlamoqFyDFQHA2kOoBoC1WGlpSdpu1DLNmy1/cNqSpRV5950FgjUAFCBUA8BarLS0JM2bleaH46bklVnza63fcuPWueTwPiktLRGqAaAAoRoA1gGvzJqfqW/Na+wyAGCtY6IyAAAAKEioBgAAgIKEagAAAChIqAYAAICChGoAAAAoSKgGAACAgoRqAAAAKEioBgAAgIKEagAAAChIqAYAAICChGoAAAAoSKgGAACAgoRqAAAAKKhQqL7xxhuz9957p1evXjn00EPz7LPPLrftyy+/nBNOOCF77713unbtmt/85jdFawUAAIAmpcGh+q677srIkSNz/PHH59Zbb023bt0yePDglJeX19l+4cKF6dSpU4YOHZoOHTqscsEAAADQVDQ4VF933XU57LDDcsghh2TLLbfMiBEjssEGG+SWW26ps33v3r0zfPjwHHDAAWnRosUqFwwAAABNRfOGNF60aFGmTp2aY489tnpZaWlp+vXrlylTpqz24j6qpORj38UqaUh9Tf25sHaq6nf6H02VPtq4vO4rp4/S1OmjfFKUlDT9flrf+hoUqufOnZulS5emrKysxvKysrJMmzatIQ9VSFlZm499H2tCu3atGrsE1nFry7HE2ksfXfN8NzWMPkpTp4/S1G200drzvdOgUN3YysvfS2VlY1exfM2bl9arc8yd+36WLq1YAxVBTSUly75km/qxxLpLH139mjUrrVdg9t1UP/ooTZ0+SmOr7/fOO++8nyVLmvb3TtXxtDINCtXt2rVLs2bNak1KVl5envbt2zeswgIqK9OkPxwaUltTfh6s/Zr6sQT6aOPwmtefPkpTp4/S1K1NfbRBE5W1aNEiPXr0yKRJk6qXVVRUZNKkSenTp89qLw4AAACasgYP/z7qqKMyfPjw9OzZM717987YsWOzcOHCDBgwIEly8sknp2PHjhk6dGiSZZObvfrqq9X/P3PmzLzwwgtp2bJlOnfuvBqfCgAAAKxZDQ7V+++/f+bMmZPRo0dn9uzZ6d69e8aMGVM9/HvGjBkpLf3vCfBZs2bl4IMPrv772muvzbXXXpsdd9wx119//ao/AwAAAGgkhSYqGzRoUAYNGlTnuo8G5U6dOuWll14qshsAAABo0hp0TTUAAADwX0I1AAAAFCRUAwAAQEFCNQAAABQkVAMAAEBBQjUAAAAUJFQDAABAQUI1AAAAFCRUAwAAQEFCNQAAABQkVAMAAEBBQjUAAAAUJFQDAABAQUI1AAAAFCRUAwAAQEFCNQAAABQkVAMAAEBBQjUAAAAUJFQDAABAQUI1AAAAFCRUAwAAQEFCNQAAABQkVAMAAEBBQjUAAAAUJFQDAABAQUI1AAAAFCRUAwAAQEFCNQAAABQkVAMAAEBBQjUAAAAUJFQDAABAQUI1AAAAFCRUAwAAQEFCNQAAABQkVAMAAEBBQjUAAAAUJFQDAABAQUI1AAAAFCRUAwAAQEFCNQAAABQkVAMAAEBBQjUAAAAUJFQDAABAQc0buwAAVq60tCSlpSXLXV9RUZmKiso1WBEAAIlQDdDklZaWpO1GLdO82fIHFy1ZWpF331kgWAMArGFCNUATV1pakubNSvPDcVPyyqz5tdZvuXHrXHJ4n5SWlgjVAABrmFAN8Anxyqz5mfrWvMYuAwCADzFRGQAAABQkVAMAAEBBQjUAAAAU5JpqAIB1lNv1Aaw6oRoAYB3kdn0Aq4dQDQCwDnK7PoDVQ6gGAFiHuV0fwKoxURkAAAAUJFQDAABAQUI1AAAAFCRUAwAAQEFCNQAAABQkVAMAAEBBQjUAAAAUJFQDAABAQUI1AAAAFCRUAwAAQEFCNQAAABQkVAMAAEBBQjUAAAAUJFQDAABAQUI1AAAAFCRUAwAAQEFCNQAAABQkVAMAAEBBQjUAAAAU1LyxCwCoS2lpSUpLS1bYpqKiMhUVlWuoIgAAqE2oBpqc0tKStN2oZZo3W/FgmiVLK/LuOwsEawAAGo1QDTQ5paUlad6sND8cNyWvzJpfZ5stN26dSw7vk9LSEqEaAIBGI1QDTdYrs+Zn6lvzGrsMAABYLqEaAADqwXwfQF2EagAAWAnzfQDLI1QDAMBKmO8DWB6hGgAA6sl8H8BHrXj8CgAAALBcQjUAAAAUJFQDAABAQa6phnVQs5XMXOp2IDQ2fZSmTh8FoIpQDeuQ0tKSLK2oTLt2rVbYzu1AaCz6KE2dPgrARwnVsA4pKSlJs9IStwOhydJHaer0UT4pjKaANadQqL7xxhtzzTXXZPbs2enWrVvOOOOM9O7de7nt77777lxyySV5880389nPfjbDhg3LHnvsUbhoYNW4HQhNnT5KU6eP0lQZTQFrXoND9V133ZWRI0dmxIgR2WabbTJ27NgMHjw499xzT8rKymq1f/rppzN06NCcdNJJ2WuvvXL77bfn+OOPz4QJE7LVVlutlicBnwSlpSUpLS1ZYRu/GgMAq8JoCljzGhyqr7vuuhx22GE55JBDkiQjRozIQw89lFtuuSVDhgyp1f63v/1tdt999xxzzDFJkh/96Ed5/PHHc8MNN+Tss89exfLhk6G0tCRtN2qZ5isZiuVX4zVvZT92+KGjpk/K6+VHrHXXJ6WP0jCO6YZbE6MpVsfx5phlbdCgUL1o0aJMnTo1xx57bPWy0tLS9OvXL1OmTKlzm2eeeSbf+c53aizbbbfdcv/99ze42NLSpLIJH1Ml///zoMemG+ZTLZrVWt+l/bJhOCu7xqWy8r+PVWS9x2h6dTZrVprmzUpzxUOv5q13F9bZZtO2n8p399wi663XLEuXVnwsdTRvvqzvLa+PJvXrp2vi9VoTdZaUlKR1mw1W+GPHkqUVeX/+/63wC72xX4819dmyJl6v1dFH61NnfWr9JHy21Gd9U+k/q+Mx1pY+urJ91LfN2vLZsrI2a+qYXlmbNfXd9En5rl8dx1tT+R5uCsfj2vQY9f1sKSlZlu+aspW9DtXtKivrH1NnzpyZL3zhCxk3blz69OlTvfyXv/xlnnrqqfzhD3+otU3Pnj0zatSofOUrX6leduONN+ayyy7L448/Xt9dAwAAQJPTxH8bAAAAgKarQaG6Xbt2adasWcrLy2ssLy8vT/v27evcpn379nn77bfr3R4AAAA+KRoUqlu0aJEePXpk0qRJ1csqKioyadKkGsPBP2zbbbfNE088UWPZ448/nm233bbh1QIAAEAT0uDh30cddVTGjx+fW2+9Na+++mp++tOfZuHChRkwYECS5OSTT84FF1xQ3f6II47Io48+mmuvvTavvvpqLr300jz//PMZNGjQ6nsWAAAA0AgafEut/fffP3PmzMno0aMze/bsdO/ePWPGjKkezj1jxoyUfmgat759++b888/PxRdfnAsvvDCf/exnc9lll7lHNQAAAJ94DZr9GwAAAPgvs38DAABAQUI1AAAAFCRUAwAAQEFCNQAAABQkVK8mN954Y/bee+/06tUrhx56aJ599tnGLol11JVXXplDDjkkffr0yS677JLjjjsu06ZNq9Hmgw8+yIgRI7LTTjulT58+OeGEE/L22283UsWs66666qp07do15557bvUyfZTGNnPmzAwbNiw77bRTevfunQMPPDDPPfdc9frKyspccskl2W233dK7d+985zvfyeuvv954BbPOWbp0aS6++OLsvffe6d27d/r375/LLrssH56DWD9lTXrqqafy3e9+N7vttlu6du2a+++/v8b6+vTHd955J0OHDk3fvn2z/fbb57TTTsv777+/Bp9FMUL1anDXXXdl5MiROf7443PrrbemW7duGTx4cMrLyxu7NNZBkydPzre+9a2MHz8+1113XZYsWZLBgwdnwYIF1W1+/vOf58EHH8zFF1+c66+/PrNmzcr3v//9RqyaddWzzz6bcePGpWvXrjWW66M0pnfffTcDBw7Meuutl6uvvjp33nlnhg8fnrZt21a3ufrqq3P99dfnpz/9acaPH59PfepTGTx4cD744INGrJx1ydVXX52bbropZ555Zu66664MGzYsY8aMyfXXX1+jjX7KmrJgwYJ07do1Z511Vp3r69Mfhw0blldeeSXXXXddrrjiivz1r3/NmWeeuaaeQnGVrLKvf/3rlSNGjKj+e+nSpZW77bZb5ZVXXtmIVcEy5eXllVtttVXl5MmTKysrKyvnzZtX2aNHj8q77767us0rr7xSudVWW1VOmTKlkapkXTR//vzKfffdt/Kxxx6rHDRoUOU555xTWVmpj9L4zjvvvMqBAwcud31FRUXlrrvuWjlmzJjqZfPmzavs2bNn5R133LEmSoTKIUOGVJ566qk1ln3/+9+vHDp0aGVlpX5K49pqq60qJ06cWP13ffpj1Xf9s88+W93m4YcfruzatWvlf/7znzVXfAHOVK+iRYsWZerUqenXr1/1stLS0vTr1y9TpkxpxMpgmffeey9Jqs+wPP/881m8eHGNPrvFFltk0003zTPPPNMYJbKOOvvss7PHHnvU6IuJPkrje+CBB9KzZ8/84Ac/yC677JKDDz4448ePr14/ffr0zJ49u0YfbdOmTbbZZhvf/awxffr0yRNPPJHXXnstSfLiiy/mb3/7W77whS8k0U9pWurTH6dMmZINN9wwvXr1qm7Tr1+/lJaWNvlLa5s3dgGfdHPnzs3SpUtTVlZWY3lZWVmt61hhTauoqMjPf/7z9O3bN1tttVWS5O233856662XDTfcsEbbsrKyzJ49uzHKZB1055135h//+EduvvnmWuv0URrbG2+8kZtuuilHHXVUvvvd7+a5557LOeeck/XWWy9f+9rXqvthXd/9rv1nTRkyZEjmz5+f/fbbL82aNcvSpUtz4okn5qtf/WqS6Kc0KfXpj2+//XY+/elP11jfvHnztG3btsl//wvVsBYbMWJEXn755fzud79r7FKg2owZM3Luuefm2muvzfrrr9/Y5UAtlZWV6dmzZ0466aQkydZbb52XX34548aNy9e+9rVGrg6Wufvuu3P77bfnggsuyJZbbpkXXnghI0eOzMYbb6yfwhpm+PcqateuXZo1a1ZrUrLy8vK0b9++kaqCZUNrH3rooYwdOzabbLJJ9fL27dtn8eLFmTdvXo325eXl6dChw5ouk3XQ1KlTU15engEDBmTrrbfO1ltvncmTJ+f666/P1ltvrY/S6Dp06JAtttiixrIuXbrkrbfeql6fxHc/jeqXv/xlhgwZkgMOOCBdu3bNwQcfnCOPPDJXXnllEv2UpqU+/bF9+/aZM2dOjfVLlizJu+++2+S//4XqVdSiRYv06NEjkyZNql5WUVGRSZMmpU+fPo1YGeuqysrKnH322Zk4cWLGjh2bzTbbrMb6nj17Zr311qvRZ6dNm5a33nor22677RqulnXRzjvvnNtvvz233XZb9X89e/bMgQceWP3/+iiNqW/fvtXXqVZ5/fXX85nPfCZJ0qlTp3To0KFGH50/f37+/ve/++5njfm///u/lJSU1FjWrFmz6ltq6ac0JfXpj3369Mm8efPy/PPPV7d54oknUlFRkd69e6/xmhvC8O/V4Kijjsrw4cPTs2fP9O7dO2PHjs3ChQszYMCAxi6NddCIESNyxx135PLLL0+rVq2qr0Fp06ZNNthgg7Rp0yaHHHJIRo0albZt26Z169Y555xz0qdPH4GFNaJ169bV1/hXadmyZTbaaKPq5foojenII4/MwIEDc8UVV2S//fbLs88+m/Hjx+fss89OkpSUlOSII47Ir3/963Tu3DmdOnXKJZdcko033jj9+/dv5OpZV+y111654oorsummm1YP/77uuutyyCGHJNFPWfPef//9/Pvf/67+e/r06XnhhRfStm3bbLrppivtj1tssUV23333nHHGGRkxYkQWL16cn/3sZznggAPSsWPHxnpa9VJSWfmhO8RT2A033JBrrrkms2fPTvfu3fOTn/wk22yzTWOXxTroo/f7rTJy5MjqH3o++OCDjBo1KnfeeWcWLVqU3XbbLWeddVaTH1rD2uvb3/52unXrltNPPz2JPkrje/DBB3PhhRfm9ddfT6dOnXLUUUflsMMOq15fWVmZ0aNHZ/z48Zk3b1622267nHXWWfnc5z7XiFWzLpk/f34uueSS3H///SkvL8/GG2+cAw44IMcff3xatGiRRD9lzXryySdzxBFH1Fr+ta99LaNGjapXf3znnXfys5/9LA888EBKS0uz77775ic/+UlatWq1Jp9KgwnVAAAAUJBrqgEAAKAgoRoAAAAKEqoBAACgIKEaAAAAChKqAQAAoCChGgAAAAoSqgEAAKAgoRoAAAAKEqoBAACgIKEaAAAAChKqAQAAoCChGgAAAAr6f095peHLanHWAAAAAElFTkSuQmCC\n",
            "text/plain": [
              "<Figure size 1200x600 with 1 Axes>"
            ]
          },
          "metadata": {},
          "output_type": "display_data"
        }
      ],
      "source": [
        "sns.set_style(\"darkgrid\")\n",
        "plt.figure(figsize=[12, 6])\n",
        "plt.title(\"–†–∞—Å–ø—Ä–µ–¥–µ–ª–µ–Ω–∏–µ –º–µ—Ç–∏–ª–∏—Ä–æ–≤–∞–Ω–∏—è 8cell\", fontsize=15)\n",
        "plt.hist(df[3], bins=100, density=True)\n",
        "plt.show()"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 206
        },
        "id": "j7Pcas9P7ee_",
        "outputId": "9e93c5f4-b458-445a-e33f-bf7ea7d3d942"
      },
      "outputs": [
        {
          "data": {
            "text/html": [
              "\n",
              "  <div id=\"df-c7d882e8-078c-4c10-83c6-b2a8e3ef70c7\" class=\"colab-df-container\">\n",
              "    <div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>0</th>\n",
              "      <th>1</th>\n",
              "      <th>2</th>\n",
              "      <th>3</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>0</th>\n",
              "      <td>chr11</td>\n",
              "      <td>3050030</td>\n",
              "      <td>3050031</td>\n",
              "      <td>33.333333</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1</th>\n",
              "      <td>chr11</td>\n",
              "      <td>3050031</td>\n",
              "      <td>3050032</td>\n",
              "      <td>50.000000</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2</th>\n",
              "      <td>chr11</td>\n",
              "      <td>3050047</td>\n",
              "      <td>3050048</td>\n",
              "      <td>50.000000</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3</th>\n",
              "      <td>chr11</td>\n",
              "      <td>3050048</td>\n",
              "      <td>3050049</td>\n",
              "      <td>57.142857</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>4</th>\n",
              "      <td>chr11</td>\n",
              "      <td>3050223</td>\n",
              "      <td>3050224</td>\n",
              "      <td>36.363636</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "</div>\n",
              "    <div class=\"colab-df-buttons\">\n",
              "\n",
              "  <div class=\"colab-df-container\">\n",
              "    <button class=\"colab-df-convert\" onclick=\"convertToInteractive('df-c7d882e8-078c-4c10-83c6-b2a8e3ef70c7')\"\n",
              "            title=\"Convert this dataframe to an interactive table.\"\n",
              "            style=\"display:none;\">\n",
              "\n",
              "  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\" viewBox=\"0 -960 960 960\">\n",
              "    <path d=\"M120-120v-720h720v720H120Zm60-500h600v-160H180v160Zm220 220h160v-160H400v160Zm0 220h160v-160H400v160ZM180-400h160v-160H180v160Zm440 0h160v-160H620v160ZM180-180h160v-160H180v160Zm440 0h160v-160H620v160Z\"/>\n",
              "  </svg>\n",
              "    </button>\n",
              "\n",
              "  <style>\n",
              "    .colab-df-container {\n",
              "      display:flex;\n",
              "      gap: 12px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert {\n",
              "      background-color: #E8F0FE;\n",
              "      border: none;\n",
              "      border-radius: 50%;\n",
              "      cursor: pointer;\n",
              "      display: none;\n",
              "      fill: #1967D2;\n",
              "      height: 32px;\n",
              "      padding: 0 0 0 0;\n",
              "      width: 32px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert:hover {\n",
              "      background-color: #E2EBFA;\n",
              "      box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "      fill: #174EA6;\n",
              "    }\n",
              "\n",
              "    .colab-df-buttons div {\n",
              "      margin-bottom: 4px;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert {\n",
              "      background-color: #3B4455;\n",
              "      fill: #D2E3FC;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert:hover {\n",
              "      background-color: #434B5C;\n",
              "      box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n",
              "      filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n",
              "      fill: #FFFFFF;\n",
              "    }\n",
              "  </style>\n",
              "\n",
              "    <script>\n",
              "      const buttonEl =\n",
              "        document.querySelector('#df-c7d882e8-078c-4c10-83c6-b2a8e3ef70c7 button.colab-df-convert');\n",
              "      buttonEl.style.display =\n",
              "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "\n",
              "      async function convertToInteractive(key) {\n",
              "        const element = document.querySelector('#df-c7d882e8-078c-4c10-83c6-b2a8e3ef70c7');\n",
              "        const dataTable =\n",
              "          await google.colab.kernel.invokeFunction('convertToInteractive',\n",
              "                                                    [key], {});\n",
              "        if (!dataTable) return;\n",
              "\n",
              "        const docLinkHtml = 'Like what you see? Visit the ' +\n",
              "          '<a target=\"_blank\" href=https://colab.research.google.com/notebooks/data_table.ipynb>data table notebook</a>'\n",
              "          + ' to learn more about interactive tables.';\n",
              "        element.innerHTML = '';\n",
              "        dataTable['output_type'] = 'display_data';\n",
              "        await google.colab.output.renderOutput(dataTable, element);\n",
              "        const docLink = document.createElement('div');\n",
              "        docLink.innerHTML = docLinkHtml;\n",
              "        element.appendChild(docLink);\n",
              "      }\n",
              "    </script>\n",
              "  </div>\n",
              "\n",
              "\n",
              "<div id=\"df-2ba507ac-8916-4905-8ea3-7506841558e0\">\n",
              "  <button class=\"colab-df-quickchart\" onclick=\"quickchart('df-2ba507ac-8916-4905-8ea3-7506841558e0')\"\n",
              "            title=\"Suggest charts\"\n",
              "            style=\"display:none;\">\n",
              "\n",
              "<svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n",
              "     width=\"24px\">\n",
              "    <g>\n",
              "        <path d=\"M19 3H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zM9 17H7v-7h2v7zm4 0h-2V7h2v10zm4 0h-2v-4h2v4z\"/>\n",
              "    </g>\n",
              "</svg>\n",
              "  </button>\n",
              "\n",
              "<style>\n",
              "  .colab-df-quickchart {\n",
              "      --bg-color: #E8F0FE;\n",
              "      --fill-color: #1967D2;\n",
              "      --hover-bg-color: #E2EBFA;\n",
              "      --hover-fill-color: #174EA6;\n",
              "      --disabled-fill-color: #AAA;\n",
              "      --disabled-bg-color: #DDD;\n",
              "  }\n",
              "\n",
              "  [theme=dark] .colab-df-quickchart {\n",
              "      --bg-color: #3B4455;\n",
              "      --fill-color: #D2E3FC;\n",
              "      --hover-bg-color: #434B5C;\n",
              "      --hover-fill-color: #FFFFFF;\n",
              "      --disabled-bg-color: #3B4455;\n",
              "      --disabled-fill-color: #666;\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart {\n",
              "    background-color: var(--bg-color);\n",
              "    border: none;\n",
              "    border-radius: 50%;\n",
              "    cursor: pointer;\n",
              "    display: none;\n",
              "    fill: var(--fill-color);\n",
              "    height: 32px;\n",
              "    padding: 0;\n",
              "    width: 32px;\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart:hover {\n",
              "    background-color: var(--hover-bg-color);\n",
              "    box-shadow: 0 1px 2px rgba(60, 64, 67, 0.3), 0 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "    fill: var(--button-hover-fill-color);\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart-complete:disabled,\n",
              "  .colab-df-quickchart-complete:disabled:hover {\n",
              "    background-color: var(--disabled-bg-color);\n",
              "    fill: var(--disabled-fill-color);\n",
              "    box-shadow: none;\n",
              "  }\n",
              "\n",
              "  .colab-df-spinner {\n",
              "    border: 2px solid var(--fill-color);\n",
              "    border-color: transparent;\n",
              "    border-bottom-color: var(--fill-color);\n",
              "    animation:\n",
              "      spin 1s steps(1) infinite;\n",
              "  }\n",
              "\n",
              "  @keyframes spin {\n",
              "    0% {\n",
              "      border-color: transparent;\n",
              "      border-bottom-color: var(--fill-color);\n",
              "      border-left-color: var(--fill-color);\n",
              "    }\n",
              "    20% {\n",
              "      border-color: transparent;\n",
              "      border-left-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "    }\n",
              "    30% {\n",
              "      border-color: transparent;\n",
              "      border-left-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "      border-right-color: var(--fill-color);\n",
              "    }\n",
              "    40% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "    }\n",
              "    60% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "    }\n",
              "    80% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "      border-bottom-color: var(--fill-color);\n",
              "    }\n",
              "    90% {\n",
              "      border-color: transparent;\n",
              "      border-bottom-color: var(--fill-color);\n",
              "    }\n",
              "  }\n",
              "</style>\n",
              "\n",
              "  <script>\n",
              "    async function quickchart(key) {\n",
              "      const quickchartButtonEl =\n",
              "        document.querySelector('#' + key + ' button');\n",
              "      quickchartButtonEl.disabled = true;  // To prevent multiple clicks.\n",
              "      quickchartButtonEl.classList.add('colab-df-spinner');\n",
              "      try {\n",
              "        const charts = await google.colab.kernel.invokeFunction(\n",
              "            'suggestCharts', [key], {});\n",
              "      } catch (error) {\n",
              "        console.error('Error during call to suggestCharts:', error);\n",
              "      }\n",
              "      quickchartButtonEl.classList.remove('colab-df-spinner');\n",
              "      quickchartButtonEl.classList.add('colab-df-quickchart-complete');\n",
              "    }\n",
              "    (() => {\n",
              "      let quickchartButtonEl =\n",
              "        document.querySelector('#df-2ba507ac-8916-4905-8ea3-7506841558e0 button');\n",
              "      quickchartButtonEl.style.display =\n",
              "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "    })();\n",
              "  </script>\n",
              "</div>\n",
              "    </div>\n",
              "  </div>\n"
            ],
            "text/plain": [
              "       0        1        2          3\n",
              "0  chr11  3050030  3050031  33.333333\n",
              "1  chr11  3050031  3050032  50.000000\n",
              "2  chr11  3050047  3050048  50.000000\n",
              "3  chr11  3050048  3050049  57.142857\n",
              "4  chr11  3050223  3050224  36.363636"
            ]
          },
          "execution_count": 101,
          "metadata": {},
          "output_type": "execute_result"
        }
      ],
      "source": [
        "df = pd.read_csv(\"icm_cell.deduplicated.bedGraph\", delimiter='\\t', skiprows=1, header=None)\n",
        "df.head()"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 547
        },
        "id": "DN7B_W728QYd",
        "outputId": "f6665567-235c-42ae-8bd7-dbdbc894fc42"
      },
      "outputs": [
        {
          "data": {
            "image/png": "iVBORw0KGgoAAAANSUhEUgAAA9UAAAISCAYAAADC0OynAAAAOXRFWHRTb2Z0d2FyZQBNYXRwbG90bGliIHZlcnNpb24zLjcuMSwgaHR0cHM6Ly9tYXRwbG90bGliLm9yZy/bCgiHAAAACXBIWXMAAA9hAAAPYQGoP6dpAABERUlEQVR4nO3de5zVdZ0/8NccEA1RZAfENUxTA5GL4j3UvMSaaZpCmm6oi7RYqW2lK2rrZUzDyrtZXlAjMon1tqmo6XrdRGkTU0ktJUuUBAcUEVZgZn5/8JvRcQaY8wWZAzyfjwePh/O9nO/7e877zPg638/3c6oaGhoaAgAAAJSt1N4FAAAAwJpKqAYAAICChGoAAAAoSKgGAACAgoRqAAAAKEioBgAAgIKEagAAAChIqAYAAICChGoAAAAoqGN7FwDwUenTp0+zn6uqqtKlS5f07t07hx9+eL70pS+lqqqqnaoDWDccc8wxmTJlSn7+859n9913b7G+oaEh99xzTyZNmpRnn302c+bMSceOHbP55ptnl112ybBhwzJw4MCm7Z988skce+yxSZIuXbrk8ccfz/rrr9/qsc8999zcfPPNSZLDDz88F1544UdwhsC6TqgG1nqHH354kqSuri6vvvpqnnrqqfz+97/P5MmTc8kll7RzdQDrrlmzZuWkk07KH/7wh3To0CH9+vXLoEGDsnjx4rz00kuZMGFCJkyYkG9+85s58cQTW+w/f/78PPTQQznwwANbrFu8eHHuueee1XEawDpOqAbWeh++MvHb3/42o0aNyt13351DDjkk++23XztVBrDumj9/fo455pi88sor2XfffXPOOedk8803b7bNs88+mx/96Ed59dVXW+zfu3fvvPzyy/n1r3/daqh+9NFH89Zbb6Vfv36ZNm3aR3YeAO6pBtY5e+65Zw499NAkyQMPPNDO1QCsmy6++OK88sorGTx4cH7yk5+0CNRJMmDAgPzsZz/Ll7/85RbrqqurM3jw4Dz66KN5++23W6z/9a9/nVKplC984QsfSf0AjYRqYJ20/fbbJ0n+/ve/Ny2bNWtWrrvuugwfPjx77713+vfvnz333DMnnXRSnnnmmWU+1oIFC3Lttddm6NCh2WmnnbLjjjvmwAMPTE1NTf7yl780bXfbbbelT58+K/x35ZVXNnv8Y445Jn369MmMGTPyX//1Xxk6dGh22GGHfPrTn87o0aPzxhtvLLO2Rx99NKNGjcoee+yR/v3757Of/WzGjBmTuXPnLnOfFdV52223tbrfzJkzc95552XIkCEZMGBAdtttt5xwwgl56qmnlnmsJ598sqznIkkWLlyYa665JocddlgGDRqUQYMG5cgjj8ztt9/e6jH69OmT/ffff5k1tLa+8Tlo7fj33HPPcp+Lcutblg++Dt///vdb3eahhx5q2ub0009vsb6hoSF33XVXjj322Oy6664ZMGBAPv/5z+fKK6/MwoULm23blt5s/NfoyiuvXObzMHbs2Kbtn3zyyVbrX94xjjnmmGbbnn766ct8rJqamqb9ZsyY0bR8xowZTY81f/78nH/++dlnn32anoef/exnqa+vb7W2mTNn5uyzz85+++2X/v3759Of/vQyfxc0HueD/xrfb2eccUZeeeWVFvs8//zz+eEPf5ihQ4c2e3+ee+65rb6nP3gurVnW+pV5je64444cffTR2XnnnbPddts1O7/lvafa4q233sqtt96aJPmP//iPdOjQYZnblkqlDBo0qNV1hx56aKvDvBuHhe+2227p2bPnStUKsCKGfwPrpHfffTdJst566zUt++///u9cdNFF+eQnP5k+ffqkS5cu+etf/5r7778/Dz/8cK6++urstddezR5n1qxZOf744/PnP/85Xbt2zW677ZZOnTplxowZmTBhQrbccst88pOfbLbPdtttl759+7ao6a9//etyA+gNN9yQX/7yl9lll13y2c9+Nk8//XTuuOOOPPHEE/nVr36VzTbbrNn2F110Ua677rqst956GTBgQHr06JEXX3wxP/vZz/Lggw/m5ptvTvfu3Zd5vA/X+fzzz+eFF15oddupU6fmhBNOyNtvv51PfvKT2XfffTNnzpz8z//8Tx577LFcdNFFOeigg5Z5rE984hPZeeedV/hc1NbWZsSIEXnxxRfTo0eP7LrrrmloaMjUqVNz+umn57nnnstZZ521zOOsrEWLFuWiiy5a5vqPqr477rgjp5xySovJmBonYGpNfX19/v3f/z133XVXOnfunP79+6dr16557rnn8uMf/ziPPvpoxo8fnw022CDJ+3MPNGp8vXfaaadsueWWZdU7Z86cXH311W3atnPnzvnc5z7X9POCBQty3333tflYL730UiZOnLjcbRYtWpRjjz02r776avbYY48sXrw4kydPzpgxY/LCCy+0uEXkxRdfzHHHHZe5c+fmk5/8ZA444IC8/vrruf/++/PQQw/loosuyuc///kWx+nevXv23nvvJMl7772XZ599NrfddlseeOCB3HXXXc3C3bXXXpvf/OY36dOnT1PvP//887n55pvzwAMP5NZbb/1Iw+CKXqNx48bl+9//fqqqqjJo0KD06tWrKfiW+wFRa5544om899572X777bPNNtsUfpwhQ4akc+fOufPOO3PUUUc1Lb/vvvvy3nvv5ZBDDlnpWgFWRKgG1jkNDQ15+OGHkzSfIXynnXbKXXfdlU996lPNtn/sscfy9a9/PTU1NfnNb37TbMbw0047LX/+85/z+c9/PhdccEE23HDDpnUzZszI/PnzWxx/yJAhOfnkk1ssv+2225Ybqn/1q1/lmmuuyT777JNk6SQ8Z5xxRu68886cd955+clPftK07T333JPrrrsuvXv3zo9//OOmUNTQ0JArr7wyV111VS644IJceumlLY7TeOXuw3VeeeWVrYbq+fPn5+STT878+fPzox/9qGlofbL0fsiRI0fmu9/9bvbYY4/8wz/8Q6vH2mWXXTJmzJgVPhdnnHFGXnzxxRx77LH593//93Tq1ClJ8uabb+aEE07IL37xi+yzzz75zGc+s8zncWX87Gc/y4wZM7LZZps1G+XwUda366675ne/+10mTZrULPi+9tpreeyxx7LbbrtlypQpLfa74YYbctddd2W33XbLJZdckh49eiRZGjBrampyyy235Mc//nFOPfXUJC3nHmh8vY844ogMHTq0zfUmyeWXX5533nlnmc9TsnTiwCTp1q1bs2PPmDGjrFB94YUXpr6+Pj169Mjs2bNb3ebpp59Onz59ct999zX14N/+9rd85Stfye23354hQ4ZkyJAhSZa+R0499dTMnTs3X/3qV3Pqqac2vefvu+++fOtb38qZZ56ZnXfeOZtuummz42y99dbNzqW+vj7f/OY3c//99+c3v/lNs6vIX/7yl/Pd73632Qdb9fX1+clPfpIrr7wyl112WbP3xKq2otdo3LhxSZJLLrmkxQdiqyJUP//880neHzVUVOfOnfPZz342d911V1577bV8/OMfT7J06Pf666+fAw88MI888shK1wuwPIZ/A+uMurq6vPLKKznzzDMzderUdOrUKcOGDWta36dPnxaBOkn23nvvHHjggfnb3/6WP/3pT03Ln3nmmUyePDnV1dU5//zzmwXqJOnVq1e22267VVb/gQce2BSok6VX2b/73e/mYx/7WB588MHMnDmzaV3jFaiLL7642VXGqqqqnHzyyenbt2/uu+++zJkzp8Vx3nvvvSRJx45t+9z1lltuyezZs3Pcccc1C9TJ0vshv/GNb2TBggX59a9/3WLfxYsXt/lYzz//fB555JEMGDAgZ5xxRlNgTZZeIfze976XZPlXb1dGbW1trrnmmmy99dY57LDDVlt9u+++e7beeutMmDCh2fJf/epXqaqqyhFHHNFinyVLlmTs2LHp3LlzLr300qZAnSSdOnXKWWedlR49emTixInLHP5c1J/+9Kf853/+ZwYPHpxPf/rTy9xuyZIlSZqPFinXI488ksceeyxDhw5tMSLkw0aPHt3sQ51PfOIT+cY3vpEk+cUvftG0/Mknn8yf/vSnbL755vnWt77V7EO0z33ucxkyZEgWLFjQNHR5eUqlUnbZZZck73+I0GiPPfZoMVKkVCrlpJNOSs+ePfPggw+u8PGLastr1PgBxcoO816Wt956K0lafNBWxCGHHNJ0q0OSvPHGG5kyZUr222+/dOnSZaUfH2BFhGpgrdd4D+D222+fz33uc7ntttuy4YYb5pJLLsknPvGJZtsuWrQoDzzwQC699NKcddZZOf3003P66ac3hem//vWvTds+/vjjSZKDDz54tfyP28EHH9xiWbdu3bLnnnumoaEhv//975MsDX8vvPBCttpqq/Tu3bvFPlVVVdlpp51SV1fX6oy4//d//5ckTcOCV+S3v/1tkuSf/umfWl3fOLS1tXtRFy1alCTNAuiy/M///E+SpVfQS6WWf7623377dO7cOc8++2yb6i7XZZddlvnz5+f0009v9UOAj7K+L3/5y3n66aebRgosXrw4t956a/bdd99Whwj/8Y9/zNy5czNo0KBWh/hvsMEG6devX95+++1W7/ddGY1Xalu7x/uDGj+8actr35olS5bkBz/4QTbccMN8+9vfXu62m2yySfbcc88WyxsnsJo6dWrThwv/+7//m2Tph1itBf4vfvGLzbZblvfeey9PPvlkxo0bl/XWW6/ZB2KN5s6dm1tvvTUXXnhhzjzzzKbfN0uWLMlbb73VFDxXtba8Ro1XfO+///6PpIZVac8990x1dXXuvPPOJMmdd96Z+vr6Fh/yAXxUDP8G1nqNQ2arqqrSpUuX9O7dOwcccEC6du3abLsXX3wxX//61/Paa68t87Ea78VO0nRl+MPB/KPS2sy4yfv/8ztr1qwkaar/lVdeaTa8vTWtTVjWeIXqw8/PsjQe7+ijjy77WI2hYaONNmrzcS699NJWh603agzqH953Rc/F8rzwwgu55ZZbstdee2WfffZp9QOClalvRQ4//PBceumlufnmm1NTU5P7778/b775ZrN7SD+ocbKu3/72t4V6oKiHH344v/3tb/PlL395hcdtnK25La99a371q1/l5Zdfzne+853lzg2QLPu9s9FGG2XjjTfOvHnz8vbbb6dbt25N76NevXq1us+H328fNGXKlBbnvdVWW+UXv/hFiyvpd911V84666wsWLBgmXW/++672WSTTVZ4jHK09TX62te+ltNPPz2nnnpqbrrppmyxxRbLnUysXI3n1dpomXJ17NgxBx10UMaPH5/nn38+v/71r7PJJpt8ZLeBAHyYUA2s9T58r2hrGhoa8q1vfSuvvfZajjrqqBx99NHp1atXNtxww1RVVeWSSy7JNddck4aGhtVQ8cppvOLWo0ePFhOrfVhrYaMxHC4riCzreJ/73OfSuXPnZW639dZbt1jWGOBXFIo+eJydd9657A8yPjwZ1ge15f7QCy+8MFVVVcu9srcy9a1I165dc9BBB+XOO+/Maaedlptvvjm9evXK3nvv3er91I19uuWWW2annXZa7mN/OLQV1XjluEuXLvm3f/u3FW7fGErb8tp/2Lx583LllVfm4x//eEaMGFH2/ivjg8PBP+yDE5UtXrw4M2bMyNNPP53Ro0fn2muvbboV47XXXmvqpTPPPLNpxEHj6JCjjjoqU6dObfX3zQeP8UFtmeCtnNfosMMOy2abbZaf/vSneeKJJzJ16tTlbl+uxkkQ//jHP66Sxzv00EMzfvz4XHzxxXnxxRdz9NFHr9StBQDlEKoBkkyfPj3Tp09P//79U1NT02L9q6++2mLZP/7jPyZZOuHR6vD666+3eo/266+/niRNkyY1zgL+4Qmg2uq5555LVVVVm+8H32yzzfKXv/wlo0aNSv/+/cs6VuNXjrUWuFs7TrJ0ePXxxx9f1nGW91ysKFT/93//dyZPnpyjjz661XvuV0V9bXH00Ufntttuy+WXX54pU6bklFNOWWbAaxwS/uGJsz5Kv/zlLzN9+vSceuqpqa6uXuH2ja/9iu6Fbs1VV12VuXPn5uyzz27T8PHG98iHzZ8/P/PmzcsGG2yQjTfeOMn776NljVhpHAXw4UnKktaf70mTJuXb3/52zjvvvFx//fVJlt4Lvnjx4hx//PE57rjjWjxOa79vlneMxrpWFKrLfY322GOPpvvNL7300qYJy1bmSvkHH3v99dfPH//4x7z88ssrNQN4kgwcODBbbbVVHnvssSQx9BtYrdxTDZD3h6J++GupGtc13j/9QYMHD06S3H333c2GhX9UPvw9rMnS4dO//e1vm+6TTpaew9Zbb52XXnqp2fdkt8ULL7yQ119/Pdtvv32br2A2Pg/l3ntZX1+fKVOmpGPHjm0K4433xK7OezyXLFmSH/7wh9loo43yzW9+c7nbftT1DRw4MP369Wu6R/eDk+y1tu1GG22UKVOmfGT35X7QvHnzctVVV2WLLbZoNSS25oknnkiSZX7/8LL89a9/zU033ZSdd955uV/T9kFvvfVWJk+e3GL53XffnSTZcccdm4Y2N04sdu+997aYXCxJ04R7jdutyJAhQ1JVVdVsNvt58+Ylaf33ze9+97u8+eabbXrschR5jf7rv/4r999/fw488MA2P9dttckmmzT18Pnnn9/qc92ooaEhTz/99Aofc9iwYdlkk03St2/fFY7QAFiVhGqALB0mWyqV8sQTTzSbuOm9997LOeec02owGThwYHbffffU1tbm7LPPbnFv5IwZM/Liiy+ushrvueeepqswydLAN2bMmCxYsCD77rtvs+Ha3/jGN5q+zqfxq2s+aO7cuS2+27ehoSFXXXVVkiw3sH3YUUcdlerq6lx//fX51a9+1WI26SVLluSxxx5rNnN6fX19LrvsssycOTP77LNP01XC5dlhhx2y55575qmnnkpNTU2rX1f2wgsv5NFHH21z7Sty22235ZVXXsk3vvGNFc5SvDrq+9rXvpZ99903o0aNWu6Vxk6dOuWrX/1q3n333Zx88smtXvl84403cscddxSu5YOuv/76vPXWW82+Rmx5Hnzwwdxzzz3p3r1704cybXXllVdmyZIlOeOMM8ra7wc/+EGz+8dfffXVpn7/yle+0rR89913T+/evfPaa6/liiuuaDYE+/7778/999+fzp07t/k9cv/996ehoaHZHAVbbbVVkqUB/YO/N954442cc845ZZ1XW5X7Gr3xxhu54IILUl1d/ZHVdMopp2SrrbbK448/nhNPPLHZNxg0euGFF3L88ce3mP2+NaNGjcqTTz65yvoaoK0M/wZIUl1dnS996UuZOHFivvjFLzYNTfz973+furq6DB06NLfddluL/X70ox/luOOOy1133ZX/+Z//yU477ZROnTrl1VdfzfPPP5/Ro0evkqGSSXLkkUfmX//1X7PrrrumR48e+cMf/pAZM2Zk0003zdlnn91s20MOOSQvvfRSrr766gwdOjR9+/bNFltskYaGhrz66qt58cUX07lz5xx55JFJls5kfNFFFzXdNzllypQWs1Q3hvP//M//zJ///OeMHj06SbLxxhvnJz/5Sb72ta/l7LPPzk9/+tN86lOfysYbb5w333wzf/zjH5uukvXu3TtPPvlkzjzzzMyYMSM9e/YsKxz96Ec/yle/+tX88pe/zF133ZXtttsum266aebPn58XX3wxM2fOzLHHHrvKJiiaNWtWttxyywwfPrwi6jvggANywAEHtGnbUaNGZfr06fmv//qvfP7zn8/222+fXr16ZfHixfnLX/6Sl156KX369Gn168HKNWvWrOy6667LvG+90Zw5czJq1Kg8++yz6dSpU84///yyZ/+eNWtWDjvssAwYMKDN++y4445ZvHhxDjjggOyxxx5ZvHhxnnjiiSxcuDCHHnpos+e0qqoqF110UY499thcffXVuf/++9O3b9+8/vrreeqpp9KxY8dccMEFrQ7/nj59etO90kuWLMmrr77adIX1X//1X5u223///fOpT30qzz33XA444IDstNNOTbOFb7fddhk0aNAqv4e5ra9Ro+9+97t5++23c+WVV66Sr71qTZcuXTJ+/PiceOKJeeihh/Loo4+mf//++fjHP57Fixc3G22zohneAdqTUA3w/5177rnZeuutc8stt2Ty5MnZaKON8ulPfzrf/va3Ww3UydJ7V2+55ZaMGzcu9913Xx5//PGUSqVsttlm+ed//ufst99+q6y+448/Pv3798/Pf/7z/OEPf8jHPvaxfPGLX8x3vvOdVoeRfvvb385ee+2VX/ziF3nqqafypz/9KRtuuGF69uyZo48+OgceeGDTtn/729+a/U/8vffeu8w6nnrqqbzxxhtNoTpZGlruvPPO/OxnP8sjjzyS3/3ud0mWTpa266675p/+6Z+avg/373//ezp16pSRI0fm+OOPL2uiqurq6kyYMCETJ07M3Xffneeffz5Tp05N9+7ds8UWW+SYY45p9avHVsZpp53W5uDXHvUtS6lUyg9/+MN87nOfy8SJE/Pss8/mj3/8YzbeeONsttlmGTly5Cob0ltVVdWmD0cWLlyY2bNn59BDD83IkSMLfY/7xz72sXznO98pa59OnTpl7NixueSSS/LAAw9k7ty56dWrV4488shWh0L36dMnt99+e37605/msccey3333ZcuXbpkyJAhOeGEEzJw4MBWj/Pmm2823aPfoUOH9OjRI3vvvXeLD1I6deqUm266KZdeemkeffTRPPTQQ+nZs2eGDx+eE088MaNGjSrr/Nqira9RkkycODGPPfZYvvCFL7T5Q5yiNt100/zqV7/KPffck0mTJjX16XrrrZfNN988Rx99dL70pS+VPV8DwOpU1bAmTGULsA475phjMmXKlPz3f//3Mr/mZ2XddtttOeOMM9p0jNNPPz1TpkzJgw8++JHUAqvKjBkz8tnPfja77bZbxo8f397lALCWck81AAAAFGT4NwDZeeedM2bMmHTr1m2F237pS1/Kvvvu+9EXBQCwBhCqAciWW26ZLbfcsk3btvWrhAAA1gXuqQYAAICC3FMNAAAABQnVAAAAUJBQDQAAAAUJ1QAAAFDQGjX7d23tO6nkadWqqpLq6o0qvk7WXXqUSqdHqXR6lEqnR6l0a1KPNta6ImtUqG5oSMU/8cmaUyfrLj1KpdOjVDo9SqXTo1S6talHDf8GAACAgoRqAAAAKEioBgAAgIKEagAAAChIqAYAAICChGoAAAAoSKgGAACAgoRqAAAAKEioBgAAgIKEagAAAChIqAYAAICChGoAAAAoSKgGAACAgoRqAAAAKEioBgAAgIKEagAAAChIqAYAAICChGoAAAAoqGN7F7A26tBh2Z9V1Nc3pL6+YTVWAwAAwEdFqF6FSqWq1NU3pFu3DZe5zZK6+rz91gLBGgAAYC0gVK9CVVVV6VCqyr9NmJqXZs1vsX7bTbvk8qMGpVSqEqoBAADWAkL1R+ClWfMz7fV57V0GAAAAHzETlQEAAEBBQjUAAAAUJFQDAABAQUI1AAAAFCRUAwAAQEFCNQAAABQkVAMAAEBBQjUAAAAUJFQDAABAQYVC9U033ZT9998/AwYMyBFHHJFnnnlmudvPmzcvNTU12WuvvdK/f/987nOfyyOPPFKoYAAAAKgUHcvdYdKkSRkzZkxqamqyww47ZNy4cRk5cmTuvffeVFdXt9h+0aJFGTFiRKqrq3P55ZenZ8+eef3117PxxhuvkhMAAACA9lJ2qL7xxhtz5JFHZtiwYUmSmpqaPPzww7n11lszatSoFtvfeuutefvttzNhwoSst956SZJevXqtZNkAAADQ/soa/r1o0aJMmzYtgwcPfv8BSqUMHjw4U6dObXWfBx98MDvuuGPOO++8DB48OF/4whdy9dVXp66ubuUqBwAAgHZW1pXquXPnpq6ursUw7+rq6kyfPr3VfV599dU88cQTOeSQQ3Lttdfmb3/7W2pqarJkyZKcdNJJZRVbVVXW5qtdOfVV+rmwdmrsO/1HpdKjVDo9SqXTo1S6NalH21pj2cO/y9XQ0JDq6up873vfS4cOHdK/f/+88cYbuf7668sO1dXVG31EVa5e3bpt2N4lsI5bW95LrL30KJVOj1Lp9CiVbm3q0bJCdbdu3dKhQ4fU1tY2W15bW5vu3bu3uk+PHj3SsWPHdOjQoWnZ1ltvndmzZ2fRokXp1KlTm49fW/tOGhrKqXj16tixlE02WXFgnjv33dTV1a+GiqC5qqqlv8Aq/b3EukuPUun0KJVOj1Lp1qQebax1RcoK1Z06dUq/fv0yefLkDBkyJElSX1+fyZMnZ/jw4a3us9NOO+Wuu+5KfX19SqWlt3C/8sor6dGjR1mBOkkaGlLRT3w5tVXyebD2q/T3EuhRKp0epdLpUSrd2tSjZX9P9YgRIzJx4sTcfvvtefnll3Puuedm4cKFGTp0aJLktNNOy8UXX9y0/dFHH5233norF1xwQf7yl7/k4YcfzjXXXJOvfOUrq+4sAAAAoB2UfU/1QQcdlDlz5uSKK67I7Nmz07dv34wdO7Zp+PfMmTObrkgnyT/+4z/m+uuvz5gxY3LooYemZ8+eOfbYY/Ov//qvq+4sAAAAoB0Umqhs+PDhyxzuPX78+BbLBg0alIkTJxY5FAAAAFSssod/AwAAAEsJ1QAAAFCQUA0AAAAFCdUAAABQkFANAAAABQnVAAAAUJBQDQAAAAUJ1QAAAFCQUA0AAAAFCdUAAABQkFANAAAABQnVAAAAUJBQDQAAAAUJ1QAAAFCQUA0AAAAFCdUAAABQkFANAAAABQnVAAAAUJBQDQAAAAUJ1QAAAFCQUA0AAAAFCdUAAABQkFANAAAABQnVAAAAUJBQDQAAAAUJ1QAAAFCQUA0AAAAFCdUAAABQkFANAAAABQnVAAAAUJBQDQAAAAUJ1QAAAFCQUA0AAAAFCdUAAABQkFANAAAABQnVAAAAUJBQDQAAAAUJ1QAAAFCQUA0AAAAFCdUAAABQkFANAAAABQnVAAAAUJBQDQAAAAUJ1QAAAFCQUA0AAAAFCdUAAABQkFANAAAABQnVAAAAUJBQDQAAAAUJ1QAAAFCQUA0AAAAFCdUAAABQkFANAAAABQnVAAAAUJBQDQAAAAUJ1QAAAFCQUA0AAAAFCdUAAABQkFANAAAABQnVAAAAUFChUH3TTTdl//33z4ABA3LEEUfkmWeeWea2t912W/r06dPs34ABAwoXDAAAAJWiY7k7TJo0KWPGjElNTU122GGHjBs3LiNHjsy9996b6urqVvfp0qVL7r333qafq6qqilcMAAAAFaLsK9U33nhjjjzyyAwbNizbbrttampqssEGG+TWW29d5j5VVVXp0aNH07/u3buvVNEAAABQCcq6Ur1o0aJMmzYtJ5xwQtOyUqmUwYMHZ+rUqcvcb8GCBdlvv/1SX1+f7bffPt/5znfyqU99quxiK/0Cdzn1Vfq5sHZq7Dv9R6XSo1Q6PUql06NUujWpR9taY1mheu7cuamrq2sxzLu6ujrTp09vdZ9PfvKT+f73v58+ffrknXfeyQ033JCjjjoqd999dzbbbLNyDp/q6o3K2r5Sdeu2YXuXwDpubXkvsfbSo1Q6PUql06NUurWpR8u+p7pcgwYNyqBBg5r9fNBBB2XChAn51re+VdZj1da+k4aGVVzgKtSxYymbbLLiwDx37rupq6tfDRVBc1VVS3+BVfp7iXWXHqXS6VEqnR6l0q1JPdpY64qUFaq7deuWDh06pLa2ttny2traNt8nvd5666Vv377529/+Vs6hkyQNDanoJ76c2ir5PFj7Vfp7CfQolU6PUun0KJVuberRsiYq69SpU/r165fJkyc3Lauvr8/kyZObXY1enrq6uvzpT39Kjx49yqsUAAAAKkzZw79HjBiR0aNHp3///hk4cGDGjRuXhQsXZujQoUmS0047LT179swpp5ySJPnxj3+cHXfcMVtuuWXmzZuX66+/Pq+//nqOOOKIVXsmAAAAsJqVHaoPOuigzJkzJ1dccUVmz56dvn37ZuzYsU3Dv2fOnJlS6f0L4PPmzctZZ52V2bNnp2vXrunXr18mTJiQbbfddtWdBQAAALSDqoaGNWck+5tvVvbN7B07ltKt24Y5+IrHMu31eS3W99t849z9zb0zd+67WbLERGWsflVVSffuG1X8e4l1lx6l0ulRKp0epdKtST3aWOuKlHVPNQAAAPA+oRoAAAAKEqoBAACgIKEaAAAAChKqAQAAoCChGgAAAAoSqgEAAKAgoRoAAAAKEqoBAACgIKEaAAAAChKqAQAAoCChGgAAAAoSqgEAAKAgoRoAAAAKEqoBAACgIKEaAAAAChKqAQAAoCChGgAAAAoSqgEAAKAgoRoAAAAKEqoBAACgIKEaAAAAChKqAQAAoCChGgAAAAoSqgEAAKAgoRoAAAAKEqoBAACgIKEaAAAAChKqAQAAoCChGgAAAAoSqgEAAKAgoRoAAAAKEqoBAACgIKEaAAAAChKqAQAAoCChGgAAAAoSqgEAAKAgoRoAAAAKEqoBAACgIKEaAAAAChKqAQAAoCChGgAAAAoSqgEAAKAgoRoAAAAKEqoBAACgIKEaAAAAChKqAQAAoCChGgAAAAoSqgEAAKAgoRoAAAAKEqoBAACgIKEaAAAAChKqAQAAoCChGgAAAAoSqgEAAKAgoRoAAAAKEqoBAACgIKEaAAAAChKqAQAAoKBCofqmm27K/vvvnwEDBuSII47IM88806b97r777vTp0yff+MY3ihwWAAAAKkrZoXrSpEkZM2ZMTjzxxNx+++3ZbrvtMnLkyNTW1i53vxkzZuQHP/hBdtlll8LFAgAAQCUpO1TfeOONOfLIIzNs2LBsu+22qampyQYbbJBbb711mfvU1dXl1FNPzcknn5wttthipQoGAACASlFWqF60aFGmTZuWwYMHv/8ApVIGDx6cqVOnLnO/q666KtXV1TniiCOKVwoAAAAVpmM5G8+dOzd1dXWprq5utry6ujrTp09vdZ///d//zS233JI77rijcJGNqqpW+iE+UuXUV+nnwtqpse/0H5VKj1Lp9CiVTo9S6dakHm1rjWWF6nLNnz8/p512Wr73ve/lH/7hH1b68aqrN1oFVbW/bt02bO8SWMetLe8l1l56lEqnR6l0epRKtzb1aFmhulu3bunQoUOLSclqa2vTvXv3Ftu/+uqree211/L1r3+9aVl9fX2SZPvtt8+9996bT3ziE20+fm3tO2loKKfi1atjx1I22WTFgXnu3HdTV1e/GiqC5qqqlv4Cq/T3EusuPUql06NUOj1KpVuTerSx1hUpK1R36tQp/fr1y+TJkzNkyJAkS0Py5MmTM3z48Bbbb7311rnzzjubLbvsssvy7rvv5rvf/W4222yzcg6fhoZU9BNfTm2VfB6s/Sr9vQR6lEqnR6l0epRKtzb1aNnDv0eMGJHRo0enf//+GThwYMaNG5eFCxdm6NChSZLTTjstPXv2zCmnnJL1118/vXv3brb/xhtvnCQtlgMAAMCapuxQfdBBB2XOnDm54oorMnv27PTt2zdjx45tGv49c+bMlEplf1MXAAAArHEKTVQ2fPjwVod7J8n48eOXu++FF15Y5JAAAABQcVxSBgAAgIKEagAAAChIqAYAAICChGoAAAAoSKgGAACAgoRqAAAAKEioBgAAgIKEagAAAChIqAYAAICChGoAAAAoSKgGAACAgoRqAAAAKEioBgAAgIKEagAAAChIqAYAAICChGoAAAAoSKgGAACAgoRqAAAAKEioBgAAgIKEagAAAChIqAYAAICChGoAAAAoSKgGAACAgoRqAAAAKEioBgAAgIKEagAAAChIqAYAAICChGoAAAAoSKgGAACAgoRqAAAAKEioBgAAgIKEagAAAChIqAYAAICChGoAAAAoSKgGAACAgoRqAAAAKEioBgAAgIKEagAAAChIqAYAAICChGoAAAAoSKgGAACAgoRqAAAAKEioBgAAgIKEagAAAChIqAYAAICChGoAAAAoSKgGAACAgoRqAAAAKEioBgAAgIKEagAAAChIqAYAAICChGoAAAAoSKgGAACAgoRqAAAAKEioBgAAgIKEagAAAChIqAYAAICChGoAAAAoSKgGAACAgoRqAAAAKKhQqL7pppuy//77Z8CAATniiCPyzDPPLHPb3/zmNxk6dGh22WWX7LjjjvniF7+YO+64o2i9AAAAUDE6lrvDpEmTMmbMmNTU1GSHHXbIuHHjMnLkyNx7772prq5usX3Xrl3z9a9/PVtvvXXWW2+9PPTQQznzzDNTXV2dvffee5WcBAAAALSHsq9U33jjjTnyyCMzbNiwbLvttqmpqckGG2yQW2+9tdXtd9999/zTP/1Tttlmm3ziE5/Icccdlz59+uT3v//9ShcPAAAA7amsK9WLFi3KtGnTcsIJJzQtK5VKGTx4cKZOnbrC/RsaGvLEE0/kL3/5S0499dSyi62qKnuX1aqc+ir9XFg7Nfad/qNS6VEqnR6l0ulRKt2a1KNtrbGsUD137tzU1dW1GOZdXV2d6dOnL3O/d955J5/5zGeyaNGilEqlnHPOOdlzzz3LOfT/P85GZe9Tibp127C9S2Adt7a8l1h76VEqnR6l0ulRKt3a1KNl31NdxIYbbpg77rgjCxYsyOTJk3PhhRdmiy22yO67717W49TWvpOGho+oyFWgY8dSNtlkxYF57tx3U1dXvxoqguaqqpb+Aqv09xLrLj1KpdOjVDo9SqVbk3q0sdYVKStUd+vWLR06dEhtbW2z5bW1tenevfsy9yuVStlyyy2TJH379s3LL7+ca6+9tuxQ3dCQin7iy6mtks+DtV+lv5dAj1Lp9CiVTo9S6damHi1rorJOnTqlX79+mTx5ctOy+vr6TJ48OYMGDWrz49TX12fRokXlHBoAAAAqTtnDv0eMGJHRo0enf//+GThwYMaNG5eFCxdm6NChSZLTTjstPXv2zCmnnJIkueaaa9K/f/984hOfyKJFi/LII4/k17/+dc4999xVeiIAAACwupUdqg866KDMmTMnV1xxRWbPnp2+fftm7NixTcO/Z86cmVLp/QvgCxYsSE1NTf7+979ngw02yNZbb50f/ehHOeigg1bdWQAAAEA7qGpoWHNGsr/5ZmXfzN6xYyndum2Yg694LNNen9difb/NN87d39w7c+e+myVLTFTG6ldVlXTvvlHFv5dYd+lRKp0epdLpUSrdmtSjjbWuSFn3VAMAAADvE6oBAACgIKEaAAAAChKqAQAAoCChGgAAAAoSqgEAAKAgoRoAAAAKEqoBAACgIKEaAAAAChKqAQAAoCChGgAAAAoSqgEAAKAgoRoAAAAKEqoBAACgIKEaAAAAChKqAQAAoCChGgAAAAoSqgEAAKAgoRoAAAAKEqoBAACgIKEaAAAAChKqAQAAoCChGgAAAAoSqgEAAKAgoRoAAAAKEqoBAACgIKEaAAAAChKqAQAAoCChGgAAAAoSqgEAAKAgoRoAAAAKEqoBAACgIKEaAAAAChKqAQAAoCChGgAAAAoSqgEAAKAgoRoAAAAKEqoBAACgIKEaAAAAChKqAQAAoCChGgAAAAoSqgEAAKAgoRoAAAAKEqoBAACgIKEaAAAAChKqAQAAoCChGgAAAAoSqgEAAKAgoRoAAAAKEqoBAACgIKEaAAAAChKqAQAAoCChGgAAAAoSqgEAAKAgoRoAAAAKEqoBAACgIKEaAAAAChKqAQAAoCChGgAAAAoSqgEAAKCgQqH6pptuyv77758BAwbkiCOOyDPPPLPMbSdOnJh//ud/zq677ppdd901//Iv/7Lc7QEAAGBNUXaonjRpUsaMGZMTTzwxt99+e7bbbruMHDkytbW1rW7/5JNP5uCDD87Pf/7zTJgwIf/4j/+Y448/Pm+88cZKFw8AAADtqexQfeONN+bII4/MsGHDsu2226ampiYbbLBBbr311la3v/jii/OVr3wlffv2zTbbbJPzzz8/9fX1mTx58koXDwAAAO2pYzkbL1q0KNOmTcsJJ5zQtKxUKmXw4MGZOnVqmx5j4cKFWbJkSbp27VpepUmqqsreZbUqp75KPxfWTo19p/+oVHqUSqdHqXR6lEq3JvVoW2ssK1TPnTs3dXV1qa6ubra8uro606dPb9NjXHTRRdl0000zePDgcg79/4+zUdn7VKJu3TZs7xJYx60t7yXWXnqUSqdHqXR6lEq3NvVoWaF6ZV177bWZNGlSfv7zn2f99dcve//a2nfS0PARFLaKdOxYyiabrDgwz537burq6ldDRdBcVdXSX2CV/l5i3aVHqXR6lEqnR6l0a1KPNta6ImWF6m7duqVDhw4tJiWrra1N9+7dl7vv9ddfn2uvvTY33nhjtttuu3IO26ShIRX9xJdTWyWfB2u/Sn8vgR6l0ulRKp0epdKtTT1a1kRlnTp1Sr9+/ZpNMtY46digQYOWud91112Xn/zkJxk7dmwGDBhQvFoAAACoIGUP/x4xYkRGjx6d/v37Z+DAgRk3blwWLlyYoUOHJklOO+209OzZM6ecckqSpUO+r7jiilx88cX5+Mc/ntmzZydJOnfunA03dG8xAAAAa66yQ/VBBx2UOXPm5Iorrsjs2bPTt2/fjB07tmn498yZM1MqvX8BfMKECVm8eHG++c1vNnuck046KSeffPJKlg8AAADtp9BEZcOHD8/w4cNbXTd+/PhmPz/44INFDgEAAAAVr6x7qgEAAID3CdUAAABQkFANAAAABQnVAAAAUJBQDQAAAAUJ1QAAAFCQUA0AAAAFCdUAAABQkFANAAAABQnVAAAAUJBQDQAAAAUJ1QAAAFCQUA0AAAAFCdUAAABQkFANAAAABQnVAAAAUJBQDQAAAAUJ1QAAAFCQUA0AAAAFCdUAAABQkFANAAAABQnVAAAAUJBQDQAAAAUJ1QAAAFCQUA0AAAAFCdUAAABQkFANAAAABQnVAAAAUJBQDQAAAAUJ1QAAAFCQUA0AAAAFCdUAAABQkFANAAAABQnVAAAAUJBQDQAAAAUJ1QAAAFCQUA0AAAAFCdUAAABQkFANAAAABQnVAAAAUJBQDQAAAAUJ1QAAAFCQUA0AAAAFCdUAAABQkFANAAAABQnVAAAAUJBQDQAAAAUJ1QAAAFCQUA0AAAAFCdUAAABQkFANAAAABQnVAAAAUJBQDQAAAAUJ1QAAAFCQUA0AAAAFCdUAAABQkFANAAAABQnVAAAAUJBQDQAAAAUVCtU33XRT9t9//wwYMCBHHHFEnnnmmWVu++c//zknn3xy9t9///Tp0yc/+9nPitYKAAAAFaXsUD1p0qSMGTMmJ554Ym6//fZst912GTlyZGpra1vdfuHChenVq1dOOeWU9OjRY6ULBgAAgEpRdqi+8cYbc+SRR2bYsGHZdtttU1NTkw022CC33nprq9sPHDgwo0ePzsEHH5xOnTqtdMEAAABQKcoK1YsWLcq0adMyePDg9x+gVMrgwYMzderUVV4cAAAAVLKO5Ww8d+7c1NXVpbq6utny6urqTJ8+fZUW1pqqqo/8ECulnPoq/VxYOzX2nf6jUulRKp0epdLpUSrdmtSjba2xrFDd3qqrN2rvElaJbt02bO8SWMetLe8l1l56lEqnR6l0epRKtzb1aFmhulu3bunQoUOLSclqa2vTvXv3VVpYa2pr30lDw0d+mMI6dixlk01WHJjnzn03dXX1q6EiaK6qaukvsEp/L7Hu0qNUOj1KpdOjVLo1qUcba12RskJ1p06d0q9fv0yePDlDhgxJktTX12fy5MkZPnx4sUrL0NCQin7iy6mtks+DtV+lv5dAj1Lp9CiVTo9S6damHi17+PeIESMyevTo9O/fPwMHDsy4ceOycOHCDB06NEly2mmnpWfPnjnllFOSLJ3c7OWXX2767zfeeCPPP/98OnfunC233HIVngoAAACsXmWH6oMOOihz5szJFVdckdmzZ6dv374ZO3Zs0/DvmTNnplR6f1LxWbNm5bDDDmv6+YYbbsgNN9yQ3XbbLePHj1/5MwAAAIB2UmiisuHDhy9zuPeHg3KvXr3y4osvFjkMAAAAVLSyvqcaAAAAeJ9QDQAAAAUJ1QAAAFCQUA0AAAAFCdUAAABQUKHZvwGANUepVJVSqWqZ6+vrG1Jf37AaKwKAtYdQDQBrsVKpKl036ZyOHZY9OG1JXX3efmuBYA0ABQjVALAWK5Wq0rFDKf82YWpemjW/xfptN+2Sy48alFKpSqgGgAKEagBYB7w0a36mvT6vvcsAgLWOicoAAACgIKEaAAAAChKqAQAAoCChGgAAAAoSqgEAAKAgoRoAAAAKEqoBAACgIKEaAAAAChKqAQAAoCChGgAAAAoSqgEAAKCgju1dAAArVipVpVSqWub6+vqG1Nc3rMaKAABIhGqAilcqVaXrJp3TscOyBxctqavP228tEKwBAFYzoRqgwpVKVenYoZR/mzA1L82a32L9tpt2yeVHDUqpVCVUAwDtbkUj7NY2QjXAGuKlWfMz7fV57V0GAMAytWWEXV19Q0qlqtTVrR0XA4RqAAAAVom2jrCrqqpKIlQDAABAC+vSCDtfqQUAAAAFCdUAAABQkFANAAAABQnVAAAAUJBQDQAAAAUJ1QAAAFCQUA0AAAAFCdUAAABQkFANAAAABXVs7wIAWlMqVaVUqlruNvX1Damvb1hNFQEAQEtCNVBxSqWqdN2kczp2WP5gmiV19Xn7rQWCNQAA7UaoBipOqVSVjh1K+bcJU/PSrPmtbrPtpl1y+VGDUipVCdUAALQboRqoWC/Nmp9pr89r7zIAAGCZTFQGAAAABQnVAAAAUJBQDQAAAAUJ1QAAAFCQUA0AAAAFCdUAAABQkFANAAAABfmeagCAdVSpVJVSqWqZ6+vrG1Jf37AaKwJY8wjVAADroFKpKl036ZyOHZY9cHFJXX3efmuBYA2wHEI1rIM6LOd/oBJXJgDWBaVSVTp2KOXfJkzNS7Pmt1i/7aZdcvlRg1IqVfmbALAcQjWsQ0qlqtTVN6Rbtw2Xu50rEwDrjpdmzc+01+e1dxkAayyhGtYhVVVV6VCqWuZVicSVCQAAKIdQDesgVyUAAGDV8JVaAAAAUJBQDQAAAAUZ/g1AxTFDPQCwphCqYTUplapSKlUtdxtBgXWdGeoBgDWNUA2rQalUla6bdE7HFVx9ExRY15mhHgBY0wjVsBqUSlXp2KEkKEAbmaEeAFhTCNWwGgkKAGsH9/2vm9zKBbRGqAYAaCP3/a+73MoFLEuhUH3TTTfl+uuvz+zZs7PddtvlrLPOysCBA5e5/T333JPLL788r732Wrbaaquceuqp2WeffQoXDaubT6bXXit6bb2uwAe573/d5VYuYFnKDtWTJk3KmDFjUlNTkx122CHjxo3LyJEjc++996a6urrF9k899VROOeWUfOc738l+++2XO++8MyeeeGJuu+229O7de5WcBCzPygZin0yvvdry2npd10w+COOj5naeddea8tq7RQFWn7JD9Y033pgjjzwyw4YNS5LU1NTk4Ycfzq233ppRo0a12P7nP/959t5773z1q19NknzrW9/K448/nl/84hc577zzVrJ8WP7/PFdVVaXLRhusVCD2yXRLa8vV3RW9tuva69oWa8Jr74Owddua0KPwUXKLAqx+ZYXqRYsWZdq0aTnhhBOalpVKpQwePDhTp05tdZ+nn346//Iv/9Js2V577ZUHHnig7GJLpaShgt/3Vf//b3i/zTfOxzp1aLF+6+5Lf7mt6JPDhob3H6vI+nXpMdoamq9++OW8/vbCVtdt3vVj+dq+22S99Tqkrq6+xfrG12v9jqVWX9fGdR/cdlmPsazeSNrWHyv7fHXsuPJ1tOU5X1JXn3fn/99y/1Avr87V9Xyt6LVd0eva1uOs7Ou2oudjdf1uWR2v/aro0Q4dSunYobRS7/sV1dmW9ZXyGJXSP6viMdaWHl3RMdq6zdryu2VNeYzV9bdpVfyt71Cq+sh/B7ZlmzXlMdaUOteUx2jr75aqqqX5rpKt6Hlo2q6hoe0x9Y033shnPvOZTJgwIYMGDWpa/sMf/jC/+93v8p//+Z8t9unfv38uvPDCfOELX2hadtNNN+Wqq67K448/3tZDAwAAQMWp8M8GAAAAoHKVFaq7deuWDh06pLa2ttny2tradO/evdV9unfvnjfffLPN2wMAAMCaoqxQ3alTp/Tr1y+TJ09uWlZfX5/Jkyc3Gw7+QTvuuGOeeOKJZssef/zx7LjjjuVXCwAAABWk7OHfI0aMyMSJE3P77bfn5ZdfzrnnnpuFCxdm6NChSZLTTjstF198cdP2xx57bB577LHccMMNefnll3PllVfmueeey/Dhw1fdWQAAAEA7KPsrtQ466KDMmTMnV1xxRWbPnp2+fftm7NixTcO5Z86cmdIHpnHbaaedctFFF+Wyyy7LJZdckq222ipXXXWV76gGAABgjVfW7N8AAADA+8z+DQAAAAUJ1QAAAFCQUA0AAAAFCdUAAABQkFC9itx0003Zf//9M2DAgBxxxBF55pln2rsk1lHXXHNNhg0blkGDBuXTn/50vvGNb2T69OnNtnnvvfdSU1OT3XffPYMGDcrJJ5+cN998s50qZl137bXXpk+fPrngggualulR2tsbb7yRU089NbvvvnsGDhyYQw45JM8++2zT+oaGhlx++eXZa6+9MnDgwPzLv/xLXnnllfYrmHVOXV1dLrvssuy///4ZOHBghgwZkquuuiofnINYn7I6/e53v8vXvva17LXXXunTp08eeOCBZuvb0o9vvfVWTjnllOy0007ZZZddcuaZZ+bdd99djWdRjFC9CkyaNCljxozJiSeemNtvvz3bbbddRo4cmdra2vYujXXQlClT8pWvfCUTJ07MjTfemCVLlmTkyJFZsGBB0zbf//7389BDD+Wyyy7L+PHjM2vWrJx00kntWDXrqmeeeSYTJkxInz59mi3Xo7Snt99+O0cffXTWW2+9XHfddbn77rszevTodO3atWmb6667LuPHj8+5556biRMn5mMf+1hGjhyZ9957rx0rZ11y3XXX5eabb87ZZ5+dSZMm5dRTT83YsWMzfvz4ZtvoU1aXBQsWpE+fPjnnnHNaXd+Wfjz11FPz0ksv5cYbb8zVV1+d//3f/83ZZ5+9uk6huAZW2pe+9KWGmpqapp/r6uoa9tprr4ZrrrmmHauCpWpraxt69+7dMGXKlIaGhoaGefPmNfTr16/hnnvuadrmpZdeaujdu3fD1KlT26lK1kXz589vOOCAAxp++9vfNgwfPrzh/PPPb2ho0KO0vx/96EcNRx999DLX19fXN+y5554NY8eObVo2b968hv79+zfcddddq6NEaBg1alTDGWec0WzZSSed1HDKKac0NDToU9pX7969G+6///6mn9vSj41/65955pmmbR555JGGPn36NPz9739ffcUX4Er1Slq0aFGmTZuWwYMHNy0rlUoZPHhwpk6d2o6VwVLvvPNOkjRdYXnuueeyePHiZj27zTbbZPPNN8/TTz/dHiWyjjrvvPOyzz77NOvFRI/S/h588MH0798/3/zmN/PpT386hx12WCZOnNi0fsaMGZk9e3azHt1oo42yww47+NvPajNo0KA88cQT+ctf/pIkeeGFF/L73/8+n/nMZ5LoUypLW/px6tSp2XjjjTNgwICmbQYPHpxSqVTxt9Z2bO8C1nRz585NXV1dqqurmy2vrq5ucR8rrG719fX5/ve/n5122im9e/dOkrz55ptZb731svHGGzfbtrq6OrNnz26PMlkH3X333fnjH/+YW265pcU6PUp7e/XVV3PzzTdnxIgR+drXvpZnn302559/ftZbb70cfvjhTX3Y2t9+9/6zuowaNSrz58/P5z//+XTo0CF1dXX59re/nUMPPTRJ9CkVpS39+Oabb+Yf/uEfmq3v2LFjunbtWvF//4VqWIvV1NTkz3/+c375y1+2dynQZObMmbngggtyww03ZP3112/vcqCFhoaG9O/fP9/5zneSJNtvv33+/Oc/Z8KECTn88MPbuTpY6p577smdd96Ziy++ONtuu22ef/75jBkzJptuuqk+hdXM8O+V1K1bt3To0KHFpGS1tbXp3r17O1UFS4fWPvzwwxk3blw222yzpuXdu3fP4sWLM2/evGbb19bWpkePHqu7TNZB06ZNS21tbYYOHZrtt98+22+/faZMmZLx48dn++2316O0ux49emSbbbZptmzrrbfO66+/3rQ+ib/9tKsf/vCHGTVqVA4++OD06dMnhx12WI477rhcc801SfQplaUt/di9e/fMmTOn2folS5bk7bffrvi//0L1SurUqVP69euXyZMnNy2rr6/P5MmTM2jQoHasjHVVQ0NDzjvvvNx///0ZN25ctthii2br+/fvn/XWW69Zz06fPj2vv/56dtxxx9VcLeuiPfbYI3feeWfuuOOOpn/9+/fPIYcc0vTfepT2tNNOOzXdp9rolVdeycc//vEkSa9evdKjR49mPTp//vz84Q9/8Lef1eb//u//UlVV1WxZhw4dmr5SS59SSdrSj4MGDcq8efPy3HPPNW3zxBNPpL6+PgMHDlztNZfD8O9VYMSIERk9enT69++fgQMHZty4cVm4cGGGDh3a3qWxDqqpqcldd92Vn/zkJ9lwww2b7kHZaKONssEGG2SjjTbKsGHDcuGFF6Zr167p0qVLzj///AwaNEhgYbXo0qVL0z3+jTp37pxNNtmkabkepT0dd9xxOfroo3P11Vfn85//fJ555plMnDgx5513XpKkqqoqxx57bH76059myy23TK9evXL55Zdn0003zZAhQ9q5etYV++23X66++upsvvnmTcO/b7zxxgwbNiyJPmX1e/fdd/O3v/2t6ecZM2bk+eefT9euXbP55puvsB+32Wab7L333jnrrLNSU1OTxYsX53vf+14OPvjg9OzZs71Oq02qGho+8A3xFPaLX/wi119/fWbPnp2+ffvmP/7jP7LDDju0d1msgz78fb+NxowZ0/RBz3vvvZcLL7wwd999dxYtWpS99tor55xzTsUPrWHtdcwxx2S77bbLd7/73SR6lPb30EMP5ZJLLskrr7ySXr16ZcSIETnyyCOb1jc0NOSKK67IxIkTM2/evOy8884555xz8slPfrIdq2ZdMn/+/Fx++eV54IEHUltbm0033TQHH3xwTjzxxHTq1CmJPmX1evLJJ3Pssce2WH744YfnwgsvbFM/vvXWW/ne976XBx98MKVSKQcccED+4z/+IxtuuOHqPJWyCdUAAABQkHuqAQAAoCChGgAAAAoSqgEAAKAgoRoAAAAKEqoBAACgIKEaAAAAChKqAQAAoCChGgAAAAoSqgEAAKAgoRoAAAAKEqoBAACgIKEaAAAACvp/NiIt52mjJqEAAAAASUVORK5CYII=\n",
            "text/plain": [
              "<Figure size 1200x600 with 1 Axes>"
            ]
          },
          "metadata": {},
          "output_type": "display_data"
        }
      ],
      "source": [
        "sns.set_style(\"darkgrid\")\n",
        "plt.figure(figsize=[12, 6])\n",
        "plt.title(\"–†–∞—Å–ø—Ä–µ–¥–µ–ª–µ–Ω–∏–µ –º–µ—Ç–∏–ª–∏—Ä–æ–≤–∞–Ω–∏—è ICM\", fontsize=15)\n",
        "plt.hist(df[3], bins=100, density=True)\n",
        "plt.show()"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 206
        },
        "id": "0TLtuw707sNK",
        "outputId": "68ccbce1-b94e-49ea-88a9-3142a14f931e"
      },
      "outputs": [
        {
          "data": {
            "text/html": [
              "\n",
              "  <div id=\"df-f08b5e8d-d11a-4cff-b994-a020b57d56b1\" class=\"colab-df-container\">\n",
              "    <div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>0</th>\n",
              "      <th>1</th>\n",
              "      <th>2</th>\n",
              "      <th>3</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>0</th>\n",
              "      <td>chr11</td>\n",
              "      <td>3050030</td>\n",
              "      <td>3050031</td>\n",
              "      <td>75.000</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1</th>\n",
              "      <td>chr11</td>\n",
              "      <td>3050031</td>\n",
              "      <td>3050032</td>\n",
              "      <td>100.000</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2</th>\n",
              "      <td>chr11</td>\n",
              "      <td>3050047</td>\n",
              "      <td>3050048</td>\n",
              "      <td>100.000</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3</th>\n",
              "      <td>chr11</td>\n",
              "      <td>3050048</td>\n",
              "      <td>3050049</td>\n",
              "      <td>100.000</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>4</th>\n",
              "      <td>chr11</td>\n",
              "      <td>3050223</td>\n",
              "      <td>3050224</td>\n",
              "      <td>96.875</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "</div>\n",
              "    <div class=\"colab-df-buttons\">\n",
              "\n",
              "  <div class=\"colab-df-container\">\n",
              "    <button class=\"colab-df-convert\" onclick=\"convertToInteractive('df-f08b5e8d-d11a-4cff-b994-a020b57d56b1')\"\n",
              "            title=\"Convert this dataframe to an interactive table.\"\n",
              "            style=\"display:none;\">\n",
              "\n",
              "  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\" viewBox=\"0 -960 960 960\">\n",
              "    <path d=\"M120-120v-720h720v720H120Zm60-500h600v-160H180v160Zm220 220h160v-160H400v160Zm0 220h160v-160H400v160ZM180-400h160v-160H180v160Zm440 0h160v-160H620v160ZM180-180h160v-160H180v160Zm440 0h160v-160H620v160Z\"/>\n",
              "  </svg>\n",
              "    </button>\n",
              "\n",
              "  <style>\n",
              "    .colab-df-container {\n",
              "      display:flex;\n",
              "      gap: 12px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert {\n",
              "      background-color: #E8F0FE;\n",
              "      border: none;\n",
              "      border-radius: 50%;\n",
              "      cursor: pointer;\n",
              "      display: none;\n",
              "      fill: #1967D2;\n",
              "      height: 32px;\n",
              "      padding: 0 0 0 0;\n",
              "      width: 32px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert:hover {\n",
              "      background-color: #E2EBFA;\n",
              "      box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "      fill: #174EA6;\n",
              "    }\n",
              "\n",
              "    .colab-df-buttons div {\n",
              "      margin-bottom: 4px;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert {\n",
              "      background-color: #3B4455;\n",
              "      fill: #D2E3FC;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert:hover {\n",
              "      background-color: #434B5C;\n",
              "      box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n",
              "      filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n",
              "      fill: #FFFFFF;\n",
              "    }\n",
              "  </style>\n",
              "\n",
              "    <script>\n",
              "      const buttonEl =\n",
              "        document.querySelector('#df-f08b5e8d-d11a-4cff-b994-a020b57d56b1 button.colab-df-convert');\n",
              "      buttonEl.style.display =\n",
              "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "\n",
              "      async function convertToInteractive(key) {\n",
              "        const element = document.querySelector('#df-f08b5e8d-d11a-4cff-b994-a020b57d56b1');\n",
              "        const dataTable =\n",
              "          await google.colab.kernel.invokeFunction('convertToInteractive',\n",
              "                                                    [key], {});\n",
              "        if (!dataTable) return;\n",
              "\n",
              "        const docLinkHtml = 'Like what you see? Visit the ' +\n",
              "          '<a target=\"_blank\" href=https://colab.research.google.com/notebooks/data_table.ipynb>data table notebook</a>'\n",
              "          + ' to learn more about interactive tables.';\n",
              "        element.innerHTML = '';\n",
              "        dataTable['output_type'] = 'display_data';\n",
              "        await google.colab.output.renderOutput(dataTable, element);\n",
              "        const docLink = document.createElement('div');\n",
              "        docLink.innerHTML = docLinkHtml;\n",
              "        element.appendChild(docLink);\n",
              "      }\n",
              "    </script>\n",
              "  </div>\n",
              "\n",
              "\n",
              "<div id=\"df-0078fda1-f15f-468a-99c7-83d5d03ab093\">\n",
              "  <button class=\"colab-df-quickchart\" onclick=\"quickchart('df-0078fda1-f15f-468a-99c7-83d5d03ab093')\"\n",
              "            title=\"Suggest charts\"\n",
              "            style=\"display:none;\">\n",
              "\n",
              "<svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n",
              "     width=\"24px\">\n",
              "    <g>\n",
              "        <path d=\"M19 3H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zM9 17H7v-7h2v7zm4 0h-2V7h2v10zm4 0h-2v-4h2v4z\"/>\n",
              "    </g>\n",
              "</svg>\n",
              "  </button>\n",
              "\n",
              "<style>\n",
              "  .colab-df-quickchart {\n",
              "      --bg-color: #E8F0FE;\n",
              "      --fill-color: #1967D2;\n",
              "      --hover-bg-color: #E2EBFA;\n",
              "      --hover-fill-color: #174EA6;\n",
              "      --disabled-fill-color: #AAA;\n",
              "      --disabled-bg-color: #DDD;\n",
              "  }\n",
              "\n",
              "  [theme=dark] .colab-df-quickchart {\n",
              "      --bg-color: #3B4455;\n",
              "      --fill-color: #D2E3FC;\n",
              "      --hover-bg-color: #434B5C;\n",
              "      --hover-fill-color: #FFFFFF;\n",
              "      --disabled-bg-color: #3B4455;\n",
              "      --disabled-fill-color: #666;\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart {\n",
              "    background-color: var(--bg-color);\n",
              "    border: none;\n",
              "    border-radius: 50%;\n",
              "    cursor: pointer;\n",
              "    display: none;\n",
              "    fill: var(--fill-color);\n",
              "    height: 32px;\n",
              "    padding: 0;\n",
              "    width: 32px;\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart:hover {\n",
              "    background-color: var(--hover-bg-color);\n",
              "    box-shadow: 0 1px 2px rgba(60, 64, 67, 0.3), 0 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "    fill: var(--button-hover-fill-color);\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart-complete:disabled,\n",
              "  .colab-df-quickchart-complete:disabled:hover {\n",
              "    background-color: var(--disabled-bg-color);\n",
              "    fill: var(--disabled-fill-color);\n",
              "    box-shadow: none;\n",
              "  }\n",
              "\n",
              "  .colab-df-spinner {\n",
              "    border: 2px solid var(--fill-color);\n",
              "    border-color: transparent;\n",
              "    border-bottom-color: var(--fill-color);\n",
              "    animation:\n",
              "      spin 1s steps(1) infinite;\n",
              "  }\n",
              "\n",
              "  @keyframes spin {\n",
              "    0% {\n",
              "      border-color: transparent;\n",
              "      border-bottom-color: var(--fill-color);\n",
              "      border-left-color: var(--fill-color);\n",
              "    }\n",
              "    20% {\n",
              "      border-color: transparent;\n",
              "      border-left-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "    }\n",
              "    30% {\n",
              "      border-color: transparent;\n",
              "      border-left-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "      border-right-color: var(--fill-color);\n",
              "    }\n",
              "    40% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "    }\n",
              "    60% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "    }\n",
              "    80% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "      border-bottom-color: var(--fill-color);\n",
              "    }\n",
              "    90% {\n",
              "      border-color: transparent;\n",
              "      border-bottom-color: var(--fill-color);\n",
              "    }\n",
              "  }\n",
              "</style>\n",
              "\n",
              "  <script>\n",
              "    async function quickchart(key) {\n",
              "      const quickchartButtonEl =\n",
              "        document.querySelector('#' + key + ' button');\n",
              "      quickchartButtonEl.disabled = true;  // To prevent multiple clicks.\n",
              "      quickchartButtonEl.classList.add('colab-df-spinner');\n",
              "      try {\n",
              "        const charts = await google.colab.kernel.invokeFunction(\n",
              "            'suggestCharts', [key], {});\n",
              "      } catch (error) {\n",
              "        console.error('Error during call to suggestCharts:', error);\n",
              "      }\n",
              "      quickchartButtonEl.classList.remove('colab-df-spinner');\n",
              "      quickchartButtonEl.classList.add('colab-df-quickchart-complete');\n",
              "    }\n",
              "    (() => {\n",
              "      let quickchartButtonEl =\n",
              "        document.querySelector('#df-0078fda1-f15f-468a-99c7-83d5d03ab093 button');\n",
              "      quickchartButtonEl.style.display =\n",
              "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "    })();\n",
              "  </script>\n",
              "</div>\n",
              "    </div>\n",
              "  </div>\n"
            ],
            "text/plain": [
              "       0        1        2        3\n",
              "0  chr11  3050030  3050031   75.000\n",
              "1  chr11  3050031  3050032  100.000\n",
              "2  chr11  3050047  3050048  100.000\n",
              "3  chr11  3050048  3050049  100.000\n",
              "4  chr11  3050223  3050224   96.875"
            ]
          },
          "execution_count": 103,
          "metadata": {},
          "output_type": "execute_result"
        }
      ],
      "source": [
        "df = pd.read_csv(\"epiblast_cell.deduplicated.bedGraph\", delimiter='\\t', skiprows=1, header=None)\n",
        "df.head()"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 547
        },
        "id": "MhRqf-KG8cai",
        "outputId": "4104e5c2-d031-42b4-8997-3372e70bb838"
      },
      "outputs": [
        {
          "data": {
            "image/png": "iVBORw0KGgoAAAANSUhEUgAAA9UAAAISCAYAAADC0OynAAAAOXRFWHRTb2Z0d2FyZQBNYXRwbG90bGliIHZlcnNpb24zLjcuMSwgaHR0cHM6Ly9tYXRwbG90bGliLm9yZy/bCgiHAAAACXBIWXMAAA9hAAAPYQGoP6dpAABBpElEQVR4nO3deZxd8/0/8NdMIsgi0klQgloqCUlI7EGVptYuRGm1QYlvqK21VCy1RGm0dqq1k5IitfVrF7WWECX2SK2tEMRIRCRfWWZ+f+Q3w5iZZOYkMjfJ8/l4eDzk3M85533v/dw793XO53xOWXV1dXUAAACAZitv6QIAAABgcSVUAwAAQEFCNQAAABQkVAMAAEBBQjUAAAAUJFQDAABAQUI1AAAAFCRUAwAAQEFCNQAAABQkVAMtrlu3bnX+6969ezbZZJP89Kc/zd/+9rdUV1e3dIkAS7x99tmn3vfxl//bfvvtF3g/t9xyS7p165aLLrqozvLjjjsu3bp1y5NPPtnkbT355JPp1q1bjjvuuAWuC6Co1i1dAECN3XffPUkyZ86cvP3223nmmWfy9NNPZ/To0Tn33HNbuDqApcPWW2+dLl26NPhYp06dFnE1pWn77bfPO++8k/Hjx7d0KUAJEKqBknHmmWfW+fdjjz2WwYMH584778z3v//9bLfddi1UGcDSY/Dgwdl8882/su1/97vfzYYbbiigA0sMw7+BkrXVVlvlBz/4QZLk/vvvb+FqAFgYOnTokHXWWSdf+9rXWroUgIVCqAZK2vrrr58kee+992qXffDBB7n88sszcODAbLPNNunZs2e22mqrHHbYYXn++ecb3db06dNz2WWXZcCAAenbt2822mij7LTTThk6dGjefPPN2nY11/vN778vXw9Ycz3ihAkT8ve//z0DBgzIhhtumC233DJDhgzJ+++/32htjzzySAYPHpwtttgiPXv2zHe+850MGzYskydPbnSd+dV5yy23NLjexIkTc9ppp6V///7p1atXNttssxx00EF55plnGt1XzXWLTX0tkmTGjBm59NJLs9tuu6VPnz7p06dP9tprr9x6660N7mN+12s29Hhj12Ymyd133z3P16K59TXmi+/D7373uwbbPPjgg7VtGrr2s7q6OnfccUf23XffbLrppunVq1d23nnnXHTRRZkxY0adtk3pmzX/1bjooosafR2uuOKK2vaNXcs6r33ss88+ddrO67rYoUOH1q43YcKE2uUTJkyo3da0adNy+umnZ9ttt619Ha655ppUVVU1WNvEiRNz8sknZ7vttkvPnj2z5ZZbNvpdULOfL/5X83k7/vjj89Zbb9VbZ9y4cfnDH/6QAQMG1Pl8nnrqqQ1+pr/4XBrS2OML8h7ddttt2XvvvbPxxhune/fuC/0a6Hkp8r03r89tjYcffjh77713+vTpk0033TSHHXZYXn/99SbXNXXq1Fx77bUZNGhQbd/YfPPNM2jQoDz22GMNrvPpp5/m0ksvzQ9+8INsvPHG6dOnT/r3758jjjgijz76aJLPvwvfeeedJFmkrzVQugz/Bkrap59+miRZZpllapf94x//yNlnn5211lor3bp1S/v27fOf//wno0aNykMPPZRLLrkkW2+9dZ3tfPDBBznggAPy6quvpmPHjtlss83Spk2bTJgwITfccEPWXHPNrLXWWnXW6d69e3r06FGvpv/85z/zDKBXXXVV/vrXv2aTTTbJd77znTz77LO57bbb8sQTT+TGG2/MKqusUqf92WefncsvvzzLLLNMevXqlS5dumT8+PG55ppr8sADD+T6669P586dG93fl+scN25cXnnllQbbjh07NgcddFA+/vjjrLXWWvn2t7+djz76KP/85z/z6KOP5uyzz84uu+zS6L7WWGONbLzxxvN9LSorK7P//vtn/Pjx6dKlSzbddNNUV1dn7NixOe644/Liiy/mpJNOanQ/C2rmzJk5++yzG338q6rvtttuy9FHH51ll122zvLrr7++0XWqqqry61//OnfccUfatm2bnj17pmPHjnnxxRfzxz/+MY888kiuvfbaLLfcckk+n3ugRs373bdv36y55prNqvejjz7KJZdc0qS2bdu2zY477lj77+nTp+fee+9t8r5ee+21jBw5cp5tZs6cmX333Tdvv/12tthii8yaNSujR4/OsGHD8sorr9S7RGT8+PHZb7/9Mnny5Ky11lrZYYcd8u6772bUqFF58MEHc/bZZ2fnnXeut5/OnTtnm222SZJ89tlneeGFF3LLLbfk/vvvzx133JGVV165tu1ll12W++67L926davt++PGjcv111+f+++/PzfffHOd9gvb/N6j4cOH53e/+13KysrSp0+fdO3aNa1atUqSZh8gWhDN/d6bl3vuuSfXX399evbsme222y7jx4/PqFGj8sQTT+S6665L9+7d57uNZ599NqeffnpWW221rLXWWtloo40yceLEPPbYY3nsscdy+umn50c/+lFt+zlz5mT//ffPc889l06dOmWzzTbLsssum/feey+PPPJI2rZtm2222SadO3fO7rvvnnvvvTfTp0+v83k0nB2WXkI1ULKqq6vz0EMPJUmds259+/bNHXfckW9+85t12j/66KP5xS9+kaFDh+a+++5LWVlZ7WPHHntsXn311ey8884544wz0q5du9rHJkyYkGnTptXbf//+/XP44YfXW37LLbfMM1TfeOONufTSS7PtttsmSWbNmpXjjz8+t99+e0477bT86U9/qm1799135/LLL896662XP/7xj7WhqLq6OhdddFEuvvjinHHGGTnvvPPq7afmzN2X67zooosaDNXTpk3L4YcfnmnTpuWss86qHVqfJC+88EIGDRqUE088MVtssUW9YZk1+9pkk00ybNiw+b4Wxx9/fMaPH5999903v/71r9OmTZskyYcffpiDDjoo1113Xbbddtt861vfavR1XBDXXHNNJkyYkFVWWaXOKIevsr5NN900Tz31VO666646P7TfeeedPProo9lss80yZsyYeutdddVVueOOO7LZZpvl3HPPrZ0gaubMmRk6dGhuuumm/PGPf8wxxxyTpP7cAzXv95577pkBAwY0ud4kueCCC/LJJ580+jolc8NGMjcwfHHfEyZMaFaoPvPMM1NVVZUuXbpk0qRJDbZ59tln061bt9x77721ffC///1vfvazn+XWW29N//79079//yRzPyPHHHNMJk+enAMPPDDHHHNM7Wf+3nvvza9+9auccMIJ2XjjjbPSSivV2c/aa69d57lUVVXliCOOyKhRo3LffffVOYv84x//OCeeeGKdA1tVVVX505/+lIsuuijnn39+nc/Ewja/92j48OFJknPPPbfeAbFFGaqb8703P3/961/z29/+NnvttVeSue/1Oeeck8svvzzHHXdcbrvttvluY6211sqNN96YjTbaqM7yl19+Ofvtt1+GDRuWnXfeufZvwVNPPZXnnnsuvXr1yogRI+ocGJs2bVrtKIZ11lknZ555ZsaMGZPp06fX+zwCSyfDv4GSM2fOnLz11ls54YQTMnbs2LRp0yZ77LFH7ePdunWrF6iTZJtttslOO+2U//73v/n3v/9du/z555/P6NGjU1FRkdNPP71OoE6Srl27NunMR1PttNNOtT8sk7ln2U888cQsv/zyeeCBBzJx4sTax2rOQJ1zzjl1zjKWlZXl8MMPT48ePXLvvffmo48+qrefzz77LEnSunXTjo/edNNNmTRpUvbbb786gTpJevXqlUMOOSTTp0/P//7v/9Zbd9asWU3e17hx4/Lwww+nV69eOf7442sDazL3DOFvf/vbJPM+e7sgKisrc+mll2bttdfObrvttsjq23zzzbP22mvnhhtuqLP8xhtvTFlZWfbcc89668yePTtXXHFF2rZtm/POO6/OjMtt2rTJSSedlC5dumTkyJGNDn8u6t///nf+9re/pV+/ftlyyy0bbTd79uwkdUeLNNfDDz+cRx99NAMGDKg3IuTLhgwZUuegzhprrJFDDjkkSXLdddfVLn/yySfz73//O6uuump+9atf1TmItuOOO6Z///6ZPn16br755vnWV15enk022STJ5wcRamyxxRb1RoqUl5fnsMMOy8orr5wHHnhgvtsvqinvUc0BioU59HjfffdtdLj/GWec0eA6zfnem5+aSzFqlJWV5Ze//GVWWWWVjBs3Lv/617/mu43VV1+9XqBO5l5S9NOf/jTTpk2rM5S+5ju2b9++9UaatG/fPj179mxy/cDSx5lqoGR88Wx0jXbt2uX3v/991lhjjTrLZ86cmUceeSQvvPBCPvroo9rQVxOm//Of/9Ru7/HHH0+S7Lrrrmnfvv1X+RRq9/NlnTp1ylZbbZX7778/Tz/9dL73ve+lsrIyr7zySr7xjW9kvfXWq7dOWVlZ+vbtm3HjxuWll16qHa5a4//+7/+SpHZY8PzUXEf43e9+t8HHa4a2NnQt6syZM5OkTgBtzD//+c8kc8+gl5fXP3a7/vrrp23btnnhhReaVHdznX/++Zk2bVrOPffcBp/LV1nfj3/849qhyt27d8+sWbNy880359vf/naDQ4RffvnlTJ48OVtttVWDQ/yXW265bLDBBnnooYfy1ltvZe211252TY2pOcN23HHH5eqrr260Xc3Bm6a89w2ZPXt2fv/736ddu3Y58sgjc+SRRzbadsUVV8xWW21Vb/n3vve9nHrqqRk7dmyqqqpSXl5eG6x22mmnBgP/D3/4w9x3333zDWCfffZZnn322QwfPjzLLLNMnWBYY/LkyXnggQfy6quvZurUqbUHOGbPnp0pU6ZkypQpWXHFFee5nyKa8h6tttpqefPNNzNq1Kh8//vfXyj7ndcttXr37t3g8qZ+7zVFQ9taZpllsuOOO2b48OF5+umnaw+CzMucOXMyevTojB07NpMmTar9Hqs56/yf//yntm2PHj1SXl6eW265Jeuuu26++93vGs4NNJlQDZSMmiGzZWVlad++fdZbb73ssMMO6dixY51248ePzy9+8YvaiWIaUnMtdpLaMyRfDuZflVVXXbXB5auttlqSudd3J6mt/6233mrwgMIXNTRhWc0Zqi+/Po2p2d/ee+/d7H1NmTIlydxZe5u6n/POO6/BYes1an7gfnnd+b0W8/LKK6/kpptuytZbb51tt922wVC9IPXNz+67757zzjsv119/fYYOHZpRo0blww8/zE9+8pMG29dM1vXYY48V6gNFPfTQQ3nsscfy4x//eL77/fjjj5M07b1vyI033pjXX389Rx111DznBkga/+x06NAhK6ywQqZOnZqPP/44nTp1qv0cde3atcF1vvx5+6IxY8bUe97f+MY3ct1119U7k37HHXfkpJNOyvTp0xut+9NPP60XqhvaR3M09T06+OCDc9xxx+WYY47JiBEjsvrqq9deU11UkVtqNfV7b1Ft67333stBBx3U6PwSSd2/E2uttVZ+/etf59xzz81JJ52UU045Jd/85jez5ZZbZvfdd1+oo5mAJY9QDZSMplybVl1dnV/96ld555138pOf/CR77713unbtmnbt2qWsrCznnntuLr300lRXVy+CihdMzdmuLl261JtY7csa+pFZEw4b+wHa2P523HHHtG3bttF2DZ0NrQnw8wtFX9zPxhtv3OwDGV+eDOuLmnJ96JlnnpmysrIGZ9heGPXNT8eOHbPLLrvk9ttvz7HHHpvrr78+Xbt2zTbbbNPg9dQ1/XTNNddM375957nthXUmtObMcfv27fPLX/5yvu1rAkxT3vsvmzp1ai666KKsttpq2X///Zu9/oL44nDwL/viRGWzZs3KhAkT8uyzz2bIkCG57LLLai/FeOedd2r70gknnFA74qBmdMhPfvKTjB07tsHvmy/u44uaMsFbc96j3XbbLausskr+/Oc/54knnsjYsWPn2X5pceKJJ+aVV17JjjvumAMPPDBrrbVW2rVrl/Ly8tx44405+eST671vBxxwQHbeeefcf//9eeyxx/L000/nmmuuyfDhw3P88cdnv/32a6FnA5Q6oRpYrLzxxht544030rNnzwwdOrTe42+//Xa9ZV//+teTzJ3waFF49913Gzyr8e677yZJ7aRJNbPhfnkCqKZ68cUXU1ZW1uQzKKusskrefPPNDB48uNnXB9bccqwpw49rnlf//v1zwAEHNGs/83ot5heq//GPf2T06NHZe++9G7zmfmHU1xR77713brnlllxwwQUZM2ZMjj766EYDXs2Q8C9PnPVV+utf/5o33ngjxxxzTCoqKubbvua9n9+10A25+OKLM3ny5Jx88slNGj5e8xn5smnTpmXq1KlZbrnlssIKKyT5/HPU2IiVmlEAX56kLGn49b7rrrty5JFH5rTTTsuVV16ZZO614LNmzcoBBxzQYKBq6PtmXvuoqWt+obq579EWW2xRe735eeedVzth2YKcKW+upn7vNXVb81o+v21Nnz49jz/+eDp37pzzzjuv3pn7eb1vX//617PPPvtkn332yezZs3PnnXfmhBNOyFlnnZXddtutySODgKWLicqAxUrNUNSGbs/y8ccf114//UX9+vVLktx55511hvt9Ve6+++56y6ZMmZLHHnus9jrpZO5zWHvttfPaa6/VuU92U7zyyit59913s/766zf5DGbN6zBq1Khm7auqqipjxoxJ69atmxTGa66Jbe5+FsTs2bPzhz/8IR06dMgRRxwxz7ZfdX29e/fOBhtsUHuN7hcn2WuobYcOHTJmzJjaIfZfpalTp+biiy/O6quv3uSzbk888USSuZNHNcd//vOfjBgxIhtvvPE8b9P2RVOmTMno0aPrLb/zzjuTJBtttFFtQKq5pvaee+6pN7lYktoJ95py7W0y9yBLWVlZndnsp06dmqTh75unnnoqH374YZO23RxF3qO///3vGTVqVHbaaacmv9YLW1O/94pua/bs2bnvvvuSpM5t/RryySef1M40/+VAPWvWrCZ/9lu3bp0f/vCH6dWrV2bNmlXnGuya6/hrJvIDlm5CNbBYWXPNNVNeXp4nnniidrKZZO5kQ6ecckqDwaR3797ZfPPNU1lZmZNPPrnetZETJkzI+PHjF1qNd999dx599NHaf8+ePTvDhg3L9OnT8+1vf7vOcO1DDjmk9nY+48aNq7etyZMn17u3b3V1dS6++OIkmWdg+7Kf/OQnqaioyJVXXpkbb7yx3mzSs2fPzqOPPlpn5vSqqqqcf/75mThxYrbddtvas4TzsuGGG2arrbbKM888k6FDhzZ4u7JXXnkljzzySJNrn59bbrklb731Vg455JB6twNrifoOPvjgfPvb387gwYPneaaxTZs2OfDAA/Ppp5/m8MMPb/AM2vvvv9+kWwg1xZVXXpkpU6bUuY3YvDzwwAO5++6707lz59qDMk110UUXZfbs2Tn++OObtd7vf//7OtePv/3227X9/Wc/+1nt8s033zzrrbde3nnnnVx44YV1hvKOGjUqo0aNStu2bZv8GRk1alSqq6vrnIn8xje+kWRuQP/i98b777+fU045pVnPq6ma+x69//77OeOMM1JRUfGV1dQUzfnem5+nn346N910U51lF110Ud59991069ZtvgdKKioq0qFDh7z66qt5+umna5fPmTMnZ599dp2/HTWeeOKJPP744/W+F99+++28/vrrKSsrqzPZYM3Z8uYeEAWWTIZ/A4uVioqK/OhHP8rIkSPzwx/+MFtssUWWXXbZPP3005kzZ04GDBiQW265pd56Z511Vvbbb7/ccccd+ec//5m+ffumTZs2efvttzNu3LgMGTJkoQ2V3GuvvfI///M/2XTTTdOlS5c899xzmTBhQlZaaaWcfPLJddp+//vfz2uvvZZLLrkkAwYMSI8ePbL66qunuro6b7/9dsaPH5+2bdvW3l7mX//6V84+++za6ybHjBlTb5bqmnD+t7/9La+++mqGDBmSJFlhhRXypz/9KQcffHBOPvnk/PnPf843v/nNrLDCCvnwww/z8ssv154lW2+99fLkk0/mhBNOyIQJE7Lyyis3KxydddZZOfDAA/PXv/41d9xxR7p3756VVlop06ZNy/jx4zNx4sTsu+++C+0+1R988EHWXHPNDBw4sCTq22GHHbLDDjs0qe3gwYPzxhtv5O9//3t23nnnrL/++unatWtmzZqVN998M6+99lq6devW4O3BmuuDDz7Ipptu2uh16zU++uijDB48OC+88ELatGmT008/vdmzf3/wwQfZbbfd0qtXryavs9FGG2XWrFnZYYcdssUWW2TWrFl54oknMmPGjPzgBz+o85qWlZXl7LPPzr777ptLLrkko0aNSo8ePfLuu+/mmWeeSevWrXPGGWc0OFT4jTfeqL1Wevbs2Xn77bfz7LPPJkn+53/+p7bd9ttvn29+85t58cUXs8MOO6Rv37757LPP8uSTT6Z79+7p06fPQr+GuanvUY0TTzwxH3/8cS666KL5HlBqqssuu2yel1uccsopWX755essa8733vzsvffe+c1vfpMbb7wxa6yxRsaPH59XX3017du3b9JlEq1bt86BBx6Y8847L/vss0+22GKLdOzYMc8991wqKyvzs5/9LCNGjKizziuvvJJhw4bla1/7WjbYYIOsuOKKmTx5csaMGZOZM2dmn332qROqt99++4wZMyY///nPs/nmm2f55ZdPp06dau8nDyxdhGpgsXPqqadm7bXXzk033ZTRo0enQ4cO2XLLLXPkkUc2GKiTudeu3nTTTRk+fHjuvffePP744ykvL88qq6ySn/70p9luu+0WWn0HHHBAevbsmb/85S957rnnsvzyy+eHP/xhjjrqqAaHkR555JHZeuutc9111+WZZ57Jv//977Rr1y4rr7xy9t577+y00061bf/73//W+RF/zz33NFrHM888k/fff782VCdzQ8vtt9+ea665Jg8//HCeeuqpJHMnS9t0003z3e9+t/Z+uO+9917atGmTQYMG5YADDmjWRFUVFRW54YYbMnLkyNx5550ZN25cxo4dm86dO2f11VfPPvvs0+BtcxbEscce2+Tg1xL1Naa8vDx/+MMfsuOOO2bkyJF54YUX8vLLL2eFFVbIKquskkGDBi20Ib1lZWVNOjgyY8aMTJo0KT/4wQ8yaNCgQjMfL7/88jnqqKOatU6bNm1yxRVX5Nxzz83999+fyZMnp2vXrtlrr70aHArdrVu33Hrrrfnzn/+cRx99NPfee2/at2+f/v3756CDDmr09k8ffvhhbWhs1apVunTpkm222abegZQ2bdpkxIgROe+88/LII4/kwQcfzMorr5yBAwfm0EMPzeDBg5v1/Jqiqe9RkowcOTKPPvpovve97zX5IE5T1Nx2rjEnnHBCvVDd3O+9edl5552z7bbb5tJLL80//vGPtG7dOt/5zndy1FFHZd11123SNg4++OCsssoqGT58eJ555pksu+yy2XjjjXPEEUfk5Zdfrtd+u+22y5QpU/Lkk0/mlVdeyZQpU/K1r30tG2+8cX7605/WuxXhPvvsk48//jh33nln7rvvvsyaNSurrbaaUA1LqbLqxWGKXIDFwD777JMxY8bkH//4R6O3+VlQt9xyS44//vgm7eO4447LmDFj8sADD3wltcDCMmHChHznO9/JZpttlmuvvbaly6EZFsX3HkCpc001AAAAFGT4N8BiZOONN86wYcPSqVOn+bb90Y9+lG9/+9tffVEAAEsxoRpgMbLmmmtmzTXXbFLbpt5KCACA4lxTDQAAAAW5phoAAAAKEqoBAACgIKEaAAAAChKqAQAAoKDFavbvyspPUsrTqpWVJRUVHUq+TpZe+iilTh+l1OmjlDp9lFK3OPXRmlrnZ7EK1dXVKfkXPll86mTppY9S6vRRSp0+SqnTRyl1S1IfNfwbAAAAChKqAQAAoCChGgAAAAoSqgEAAKAgoRoAAAAKEqoBAACgIKEaAAAAChKqAQAAoCChGgAAAAoSqgEAAKAgoRoAAAAKEqoBAACgIKEaAAAAChKqAQAAoCChGgAAAAoSqgEAAKAgoRoAAAAKEqoBAACgoNYtXQAAAABLjvLyspSXl7V0GYuMUA0AAMBCUV5elo4rtk3rVo0Pip5TVZ3y8rLMmVO9CCv76gjVAAAALBTl5WVp3ao8v7xhbF77YFq9x9ddqX0u+EmflJWVJRGqAQAAoJ7XPpiWl96d2tJlLBImKgMAAICChGoAAAAoSKgGAACAgoRqAAAAKEioBgAAgIKEagAAAChIqAYAAICChGoAAAAoSKgGAACAgoRqAAAAKEioBgAAgIKEagAAAChIqAYAAICChGoAAAAoSKgGAACAgoRqAAAAKEioBgAAgIKEagAAAChIqAYAAICChGoAAAAoSKgGAACAgoRqAAAAKEioBgAAgIKEagAAAChIqAYAAICChGoAAAAoSKgGAACAgoRqAAAAKEioBgAAgIKEagAAAChIqAYAAICChGoAAAAoSKgGAACAgoRqAAAAKEioBgAAgIKEagAAAChIqAYAAICChGoAAAAoSKgGAACAgoRqAAAAKEioBgAAgIKEagAAAChIqAYAAICChGoAAAAoSKgGAACAgoRqAAAAKEioBgAAgIKEagAAAChIqAYAAICChGoAAAAoSKgGAACAgoRqAAAAKEioBgAAgIIKheoRI0Zk++23T69evbLnnnvm+eefb9J6d955Z7p165ZDDjmkyG4BAACgpDQ7VN91110ZNmxYDj300Nx6663p3r17Bg0alMrKynmuN2HChPz+97/PJptsUrhYAAAAKCXNDtVXX3119tprr+yxxx5Zd911M3To0Cy33HK5+eabG11nzpw5OeaYY3L44Ydn9dVXX6CCAQAAoFQ0K1TPnDkzL730Uvr16/f5BsrL069fv4wdO7bR9S6++OJUVFRkzz33LF4pAAAAlJjWzWk8efLkzJkzJxUVFXWWV1RU5I033mhwnX/961+56aabcttttxUuskZZ2QJv4itVU1+p18nSSx+l1OmjlDp9lFKnj7K4KCsr/X7a1PqaFaqba9q0aTn22GPz29/+Nl/72tcWeHsVFR0WQlVfvcWlTpZe+iilTh+l1OmjlDp9lFK34ortWrqEhaZZobpTp05p1apVvUnJKisr07lz53rt33777bzzzjv5xS9+UbusqqoqSbL++uvnnnvuyRprrNHk/VdWfpLq6uZUvGiVlc39Aiv1Oll66aOUOn2UUqePUur0UVpaq1bl6dRp/oF5ypRPM3t21SKoqLiaz9P8NCtUt2nTJhtssEFGjx6d/v37J5kbkkePHp2BAwfWa7/22mvn9ttvr7Ps/PPPz6effpoTTzwxq6yySnN2n+rqLBZfDotLnSy99FFKnT5KqdNHKXX6KKVuSeqjzR7+vf/++2fIkCHp2bNnevfuneHDh2fGjBkZMGBAkuTYY4/NyiuvnKOPPjrLLrts1ltvvTrrr7DCCklSbzkAAAAsbpodqnfZZZd89NFHufDCCzNp0qT06NEjV1xxRe3w74kTJ6a8vNl36gIAAIDFTqGJygYOHNjgcO8kufbaa+e57plnnllklwAAAFBynFIGAACAgoRqAAAAKEioBgAAgIKEagAAAChIqAYAAICChGoAAAAoSKgGAACAgoRqAAAAKEioBgAAgIKEagAAAChIqAYAAICChGoAAAAoSKgGAACAgoRqAAAAKEioBgAAgIKEagAAAChIqAYAAICChGoAAAAoSKgGAACAgoRqAAAAKEioBgAAgIKEagAAAChIqAYAAICChGoAAAAoSKgGAACAgoRqAAAAKEioBgAAgIKEagAAAChIqAYAAICChGoAAAAoSKgGAACAgoRqAAAAKEioBgAAgIKEagAAAChIqAYAAICChGoAAAAoSKgGAACAgoRqAAAAKEioBgAAgIKEagAAAChIqAYAAICChGoAAAAoSKgGAACAgoRqAAAAKEioBgAAgIKEagAAAChIqAYAAICChGoAAAAoSKgGAACAgoRqAAAAKEioBgAAgIKEagAAAChIqAYAAICChGoAAAAoSKgGAACAgoRqAAAAKEioBgAAgIKEagAAAChIqAYAAICChGoAAAAoSKgGAACAgoRqAAAAKEioBgAAgIKEagAAAChIqAYAAICChGoAAAAoSKgGAACAgoRqAAAAKEioBgAAgIKEagAAAChIqAYAAICChGoAAAAoSKgGAACAgoRqAAAAKEioBgAAgIKEagAAAChIqAYAAICChGoAAAAoSKgGAACAgoRqAAAAKEioBgAAgIKEagAAAChIqAYAAICChGoAAAAoqFCoHjFiRLbffvv06tUre+65Z55//vlG2953330ZMGBANtlkk2y00Ub54Q9/mNtuu61ovQAAAFAyWjd3hbvuuivDhg3L0KFDs+GGG2b48OEZNGhQ7rnnnlRUVNRr37Fjx/ziF7/I2muvnWWWWSYPPvhgTjjhhFRUVGSbbbZZKE8CAAAAWkKzz1RfffXV2WuvvbLHHntk3XXXzdChQ7Pccsvl5ptvbrD95ptvnu9+97tZZ511ssYaa2S//fZLt27d8vTTTy9w8QAAANCSmhWqZ86cmZdeein9+vX7fAPl5enXr1/Gjh073/Wrq6szevTovPnmm9l0002bXy0AAACUkGYN/548eXLmzJlTb5h3RUVF3njjjUbX++STT/Ktb30rM2fOTHl5eU455ZRstdVWzS62rKzZqyxSNfWVep0svfRRSp0+SqnTRyl1+iiLi7Ky0u+nTa2v2ddUF9GuXbvcdtttmT59ekaPHp0zzzwzq6++ejbffPNmbaeiosNXVOHCtbjUydJLH6XU6aOUOn2UUqePUupWXLFdS5ew0DQrVHfq1CmtWrVKZWVlneWVlZXp3Llzo+uVl5dnzTXXTJL06NEjr7/+ei677LJmh+rKyk9SXd2sVRapsrK5X2ClXidLL32UUqePUur0UUqdPkpLa9WqPJ06zT8wT5nyaWbPrloEFRVX83man2aF6jZt2mSDDTbI6NGj079//yRJVVVVRo8enYEDBzZ5O1VVVZk5c2Zzdp0kqa7OYvHlsLjUydJLH6XU6aOUOn2UUqePUuqWpD7a7OHf+++/f4YMGZKePXumd+/eGT58eGbMmJEBAwYkSY499tisvPLKOfroo5Mkl156aXr27Jk11lgjM2fOzMMPP5z//d//zamnnrpQnwgAAAAsas0O1bvssks++uijXHjhhZk0aVJ69OiRK664onb498SJE1Ne/vmk4tOnT8/QoUPz3nvvZbnllsvaa6+ds846K7vsssvCexYAAADQAsqqqxefk+4fflja14aUlSWdO3co+TpZeumjlDp9lFKnj1Lq9FFaWuvWc6+p3vXCR/PSu1PrPb7BqivkziO2yeTJi8c11Z07z/+a6mbdpxoAAAD4nFANAAAABQnVAAAAUJBQDQAAAAUJ1QAAAFCQUA0AAAAFCdUAAABQkFANAAAABQnVAAAAUJBQDQAAAAUJ1QAAAFCQUA0AAAAFCdUAAABQkFANAAAABQnVAAAAUJBQDQAAAAUJ1QAAAFCQUA0AAAAFCdUAAABQkFANAAAABQnVAAAAUJBQDQAAAAUJ1QAAAFCQUA0AAAAFCdUAAABQkFANAAAABQnVAAAAUJBQDQAAAAUJ1QAAAFCQUA0AAAAFCdUAAABQkFANAAAABQnVAAAAUJBQDQAAAAUJ1QAAAFCQUA0AAAAFCdUAAABQkFANAAAABQnVAAAAUJBQDQAAAAUJ1QAAAFCQUA0AAAAFCdUAAABQkFANAAAABQnVAAAAUJBQDQAAAAUJ1QAAAFCQUA0AAAAFCdUAAABQkFANAAAABQnVAAAAUJBQDQAAAAUJ1QAAAFCQUA0AAAAFCdUAAABQkFANAAAABQnVAAAAUJBQDQAAAAUJ1QAAAFCQUA0AAAAFCdUAAABQkFANAAAABQnVAAAAUJBQDQAAAAUJ1QAAAFCQUA0AAAAFCdUAAABQkFANAAAABQnVAAAAUJBQDQAAAAUJ1QAAAFCQUA0AAAAFCdUAAABQkFANAAAABQnVAAAAUJBQDQAAAAUJ1QAAAFCQUA0AAAAFCdUAAABQkFANAAAABQnVAAAAUJBQDQAAAAUJ1QAAAFCQUA0AAAAFFQrVI0aMyPbbb59evXplzz33zPPPP99o25EjR+anP/1pNt1002y66ab5+c9/Ps/2AAAAsLhodqi+6667MmzYsBx66KG59dZb07179wwaNCiVlZUNtn/yySez66675i9/+UtuuOGGfP3rX88BBxyQ999/f4GLBwAAgJbU7FB99dVXZ6+99soee+yRddddN0OHDs1yyy2Xm2++ucH255xzTn72s5+lR48eWWeddXL66aenqqoqo0ePXuDiAQAAoCW1bk7jmTNn5qWXXspBBx1Uu6y8vDz9+vXL2LFjm7SNGTNmZPbs2enYsWPzKk1SVtbsVRapmvpKvU6WXvoopU4fpdTpo5Q6fZTFRVlZ6ffTptbXrFA9efLkzJkzJxUVFXWWV1RU5I033mjSNs4+++ystNJK6devX3N2/f/306HZ67SExaVOll76KKVOH6XU6aOUOn2UUrfiiu1auoSFplmhekFddtllueuuu/KXv/wlyy67bLPXr6z8JNXVX0FhC0lZ2dwvsFKvk6WXPkqp00cpdfoopU4fpaW1alWeTp3mH5inTPk0s2dXLYKKiqv5PM1Ps0J1p06d0qpVq3qTklVWVqZz587zXPfKK6/MZZddlquvvjrdu3dvzm5rVVdnsfhyWFzqZOmlj1Lq9FFKnT5KqdNHKXVLUh9t1kRlbdq0yQYbbFBnkrGaScf69OnT6HqXX355/vSnP+WKK65Ir169ilcLAAAAJaTZw7/333//DBkyJD179kzv3r0zfPjwzJgxIwMGDEiSHHvssVl55ZVz9NFHJ5k75PvCCy/MOeeck9VWWy2TJk1KkrRt2zbt2i054+gBAABY+jQ7VO+yyy756KOPcuGFF2bSpEnp0aNHrrjiitrh3xMnTkx5+ecnwG+44YbMmjUrRxxxRJ3tHHbYYTn88MMXsHwAAABoOYUmKhs4cGAGDhzY4GPXXnttnX8/8MADRXYBAAAAJa9Z11QDAAAAnxOqAQAAoCChGgAAAAoSqgEAAKAgoRoAAAAKEqoBAACgIKEaAAAAChKqAQAAoCChGgAAAAoSqgEAAKAgoRoAAAAKEqoBAACgIKEaAAAAChKqAQAAoCChGgAAAAoSqgEAAKAgoRoAAAAKEqoBAACgIKEaAAAAChKqAQAAoCChGgAAAAoSqgEAAKAgoRoAAAAKEqoBAACgIKEaAAAAChKqAQAAoCChGgAAAAoSqgEAAKAgoRoAAAAKEqoBAACgIKEaAAAAChKqAQAAoCChGgAAAAoSqgEAAKAgoRoAAAAKEqoBAACgIKEaAAAAChKqAQAAoCChGgAAAAoSqgEAAKAgoRoAAAAKEqoBAACgIKEaAAAAChKqAQAAoCChGgAAAAoSqgEAAKAgoRoAAAAKEqoBAACgIKEaAAAAChKqAQAAoCChGgAAAAoSqgEAAKAgoRoAAAAKEqoBAACgIKEaAAAAChKqAQAAoCChGgAAAAoSqgEAAKAgoRoAAAAKEqoBAACgIKEaAAAAChKqAQAAoCChGgAAAAoSqgEAAKAgoRoAAAAKEqoBAACgIKEaAAAACmrd0gUsiVq1avxYRVVVdaqqqhdhNQAAAHxVhOqFqLy8LHOqqtOpU7tG28yeU5WPp0wXrAEAAJYAQvVCVFZWllblZfnlDWPz2gfT6j2+7krtc8FP+qS8vEyoBgAAWAII1V+B1z6YlpfendrSZQAAAPAVM1EZAAAAFCRUAwAAQEFCNQAAABQkVAMAAEBBQjUAAAAUJFQDAABAQUI1AAAAFCRUAwAAQEFCNQAAABQkVAMAAEBBQjUAAAAUJFQDAABAQUI1AAAAFFQoVI8YMSLbb799evXqlT333DPPP/98o21fffXVHH744dl+++3TrVu3XHPNNUVrBQAAgJLS7FB91113ZdiwYTn00ENz6623pnv37hk0aFAqKysbbD9jxox07do1Rx99dLp06bLABQMAAECpaHaovvrqq7PXXntljz32yLrrrpuhQ4dmueWWy80339xg+969e2fIkCHZdddd06ZNmwUuGAAAAEpF6+Y0njlzZl566aUcdNBBtcvKy8vTr1+/jB07dqEX92VlZV/5LhZIc+or9efCkqmm3+l/lCp9lFKnj1Lq9FEWF2Vlpd9Pm1pfs0L15MmTM2fOnFRUVNRZXlFRkTfeeKM5myqkoqLDV76PRaFTp3YtXQJLuSXls8SSSx+l1OmjlDp9lFK34opLTiZqVqhuaZWVn6S6uqWraFzr1uVN6hyTJ3+aOXOqFkFFUFdZ2dw/sqX+WWLppY9S6vRRSp0+Sktr1aq8SScRp0z5NLNnl3Ymqvk8zU+zQnWnTp3SqlWrepOSVVZWpnPnzs2rsIDq6pT0l0Nzaivl58GSr9Q/S6CPUur0UUqdPkqpW5L6aLMmKmvTpk022GCDjB49unZZVVVVRo8enT59+iz04gAAAKCUNXv49/77758hQ4akZ8+e6d27d4YPH54ZM2ZkwIABSZJjjz02K6+8co4++ugkcyc3e/3112v///3338+4cePStm3brLnmmgvxqQAAAMCi1exQvcsuu+Sjjz7KhRdemEmTJqVHjx654ooraod/T5w4MeXln58A/+CDD7LbbrvV/vuqq67KVVddlc022yzXXnvtgj8DAAAAaCGFJiobOHBgBg4c2OBjXw7KXbt2zfjx44vsBgAAAEpas66pBgAAAD4nVAMAAEBBQjUAAAAUJFQDAABAQUI1AAAAFCRUAwAAQEFCNQAAABQkVAMAAEBBQjUAAAAUJFQDAABAQUI1AAAAFCRUAwAAQEFCNQAAABQkVAMAAEBBQjUAAAAUJFQDAABAQUI1AAAAFCRUAwAAQEFCNQAAABQkVAMAAEBBQjUAAAAUJFQDAABAQUI1AAAAFCRUAwAAQEFCNQAAABQkVAMAAEBBQjUAAAAUJFQDAABAQUI1AAAAFCRUAwAAQEFCNQAAABQkVAMAAEBBQjUAAAAUJFQDAABAQUI1AAAAFCRUAwAAQEFCNQAAABQkVAMAAEBBQjUAAAAUJFQDAABAQUI1AAAAFCRUAwAAQEGtW7oAAABaRnl5WcrLyxp9vKqqOlVV1YuwIoDFj1ANALAUKi8vS8cV26Z1q8YHLs6eU5WPp0wXrP+/+R2ESByIgKWRUA0AsBQqLy9L61bl+eUNY/PaB9PqPb7uSu1zwU/6pLy8TEhM0w5CJA5EwNJIqAYAWIq99sG0vPTu1JYuo+TN7yBE4kAELK2EagAAaKLF5SBEq/mcUTdMHRYeoRoAlnAmo4KlR3l5WeZUVadTp3bzbGeYOiw8QjUALMFMRgVLl7KysrQqLzNMHRYhoRoAlmAmo4Kl0+IyTB2WBEI1ACwF/MAGgK/GvGcwAAAAABolVAMAAEBBQjUAAAAU5JpqAACg2dyuD+YSqgEAgGZxuz74nFANAAA0i9v1weeEagAAoBC36wMTlQEAAEBhQjUAAAAUJFQDAABAQa6pBgAAWoTbcrEkEKoBAIBFzm25WFII1QAAwCLntlwsKYRqAACgxbgtF4s7E5UBAABAQUI1AAAAFCRUAwAAQEFCNQAAABRkojIAAGCx5V7XtDShGgAAWCy51zWlQKgGAAAWS+51TSkQqgEAmqnVPM6KJYabwqLmXte0JKEaAKCJysvLMqeqOp06tZtnO8NNAZYeQjXAYsAkLFAaysrK0qq8rNGhponhpgBLG6EaoMSZhAVKj6GmsHiZ1yUbDkyzoIRqgBJnEhZgaTe/0TqJYETDurRfNnOqqrPCCss32qYpB6aNGGNehGqAxYQzY8DSqCmjdRIjdmjYCsu3nuclG005MG3EGPMjVAMAULLmN1onMWKH+VuQA9NGjDE/QjUAACXPaB1amj5IY4RqoCS5fg4AgMWBUA2UHNfPMa9ZWhMHVIDFlwmvYMkjVAMlx/VzS6/y8rLMqapOp07t5tnOARUaI7BQykx4BUsmoRqWQovLWUDXLi19ysrK5jlLa+KACo1bnAKLS1yWTia8giVToVA9YsSIXHnllZk0aVK6d++ek046Kb179260/d13350LLrgg77zzTr7xjW/kmGOOybbbblu4aFgclcIPKGcBWVw4oEIRi0tgcYkLpfIdN6+D7A7qQNM1O1TfddddGTZsWIYOHZoNN9www4cPz6BBg3LPPfekoqKiXvtnnnkmRx99dI466qhst912uf3223PooYfmlltuyXrrrbdQngSUulL5AeUsIEuLUjiIRcsplcDSGJe41Devz+z8RlfRfF3aL5s5VdVZYYXlG23joE7zLWhf9Xdp8dXsUH311Vdnr732yh577JEkGTp0aB566KHcfPPNGTx4cL32f/nLX7LNNtvkwAMPTJL86le/yuOPP57rrrsup5122gKWD/O3MH5cL+g2Su0HVKn/4FyUXH/ZPIvD67WwDmItDs91SbO0veZN+S5eGs4kNvUz2xSl8no1VkdTQ9dXfZnWCsu3nudB9oX5m2Rez2VhvB6lcNClKQcpkmROVXVazeM7bvacqkz75P9SXd34a76kfO6XNM0K1TNnzsxLL72Ugw46qHZZeXl5+vXrl7Fjxza4zrPPPpuf//zndZZtvfXWuf/++5tdbHl5Mo8+1uLK/v9nZINVV8jybVrVe3ztznOH3M7vw19d/fm2ijxuG58rKytL+w7LNenH9afT/q/BL6mFsY2a93zZ1uUN9o2ax77YtiEL+nq1/v/7aKyPJk3rp1/1+1qz76+6zqa8t/N6X5u6n6/69VhU3y2L4vVaGH20VavytG5Vnkseej3vfjyjwTardlw+B397nSyzTKvMmVNV7/HFpW805fFS6T/za7MwXvNF8VwXVh+d3zY2XmPFJp1JXJA+WCp9Y36f2V6rdcyem6xe8q9XU+uY1zaauv68nktT+ug6Xdonafx3SVN+k8zv9Wjqc/mqt1HzXBekn8/vufZZY8W0Ki+b59+dmn7cWJtvrtQ+P918zay4YttG60jm349L4fd5U79bysrm5rtSNr/XobZd9bwOhXzJ+++/n29961u54YYb0qdPn9rlf/jDH/LUU0/lb3/7W711evbsmTPPPDPf+973apeNGDEiF198cR5//PGm7hoAAABKTokfGwAAAIDS1axQ3alTp7Rq1SqVlZV1lldWVqZz584NrtO5c+d8+OGHTW4PAAAAi4tmheo2bdpkgw02yOjRo2uXVVVVZfTo0XWGg3/RRhttlCeeeKLOsscffzwbbbRR86sFAACAEtLs4d/7779/Ro4cmVtvvTWvv/56Tj311MyYMSMDBgxIkhx77LE555xzatvvu+++efTRR3PVVVfl9ddfz0UXXZQXX3wxAwcOXHjPAgAAAFpAs2+ptcsuu+Sjjz7KhRdemEmTJqVHjx654ooraodzT5w4MeVfmMatb9++Ofvss3P++efn3HPPzTe+8Y1cfPHF7lENAADAYq9Zs38DAAAAnzP7NwAAABQkVAMAAEBBQjUAAAAUJFQDAABAQUL1QjJixIhsv/326dWrV/bcc888//zzLV0SS6lLL700e+yxR/r06ZMtt9wyhxxySN544406bT777LMMHTo0m2++efr06ZPDDz88H374YQtVzNLusssuS7du3XLGGWfULtNHaWnvv/9+jjnmmGy++ebp3bt3vv/97+eFF16ofby6ujoXXHBBtt566/Tu3Ts///nP89Zbb7VcwSx15syZk/PPPz/bb799evfunf79++fiiy/OF+cg1k9ZlJ566qkcfPDB2XrrrdOtW7fcf//9dR5vSn+cMmVKjj766PTt2zebbLJJTjjhhHz66aeL8FkUI1QvBHfddVeGDRuWQw89NLfeemu6d++eQYMGpbKysqVLYyk0ZsyY/OxnP8vIkSNz9dVXZ/bs2Rk0aFCmT59e2+Z3v/tdHnzwwZx//vm59tpr88EHH+Swww5rwapZWj3//PO54YYb0q1btzrL9VFa0scff5y99947yyyzTC6//PLceeedGTJkSDp27Fjb5vLLL8+1116bU089NSNHjszyyy+fQYMG5bPPPmvBylmaXH755bn++utz8skn56677soxxxyTK664Itdee22dNvopi8r06dPTrVu3nHLKKQ0+3pT+eMwxx+S1117L1VdfnUsuuST/+te/cvLJJy+qp1BcNQvsRz/6UfXQoUNr/z1nzpzqrbfeuvrSSy9twapgrsrKyur11luvesyYMdXV1dXVU6dOrd5ggw2q77777to2r732WvV6661XPXbs2BaqkqXRtGnTqnfYYYfqxx57rHrgwIHVp59+enV1tT5KyzvrrLOq995770Yfr6qqqt5qq62qr7jiitplU6dOre7Zs2f1HXfcsShKhOrBgwdXH3/88XWWHXbYYdVHH310dXW1fkrLWm+99apHjRpV+++m9Meav/XPP/98bZuHH364ulu3btXvvffeoiu+AGeqF9DMmTPz0ksvpV+/frXLysvL069fv4wdO7YFK4O5PvnkkySpPcPy4osvZtasWXX67DrrrJNVV101zz77bEuUyFLqtNNOy7bbblunLyb6KC3vgQceSM+ePXPEEUdkyy23zG677ZaRI0fWPj5hwoRMmjSpTh/t0KFDNtxwQ3/7WWT69OmTJ554Im+++WaS5JVXXsnTTz+db33rW0n0U0pLU/rj2LFjs8IKK6RXr161bfr165fy8vKSv7S2dUsXsLibPHly5syZk4qKijrLKyoq6l3HCotaVVVVfve736Vv375Zb731kiQffvhhlllmmaywwgp12lZUVGTSpEktUSZLoTvvvDMvv/xybrrppnqP6aO0tLfffjvXX3999t9//xx88MF54YUXcvrpp2eZZZbJ7rvvXtsPG/rb79p/FpXBgwdn2rRp2XnnndOqVavMmTMnRx55ZH7wgx8kiX5KSWlKf/zwww/zta99rc7jrVu3TseOHUv+779QDUuwoUOH5tVXX81f//rXli4Fak2cODFnnHFGrrrqqiy77LItXQ7UU11dnZ49e+aoo45Kkqy//vp59dVXc8MNN2T33Xdv4epgrrvvvju33357zjnnnKy77roZN25chg0blpVWWkk/hUXM8O8F1KlTp7Rq1arepGSVlZXp3LlzC1UFc4fWPvTQQxk+fHhWWWWV2uWdO3fOrFmzMnXq1DrtKysr06VLl0VdJkuhl156KZWVlRkwYEDWX3/9rL/++hkzZkyuvfbarL/++vooLa5Lly5ZZ5116ixbe+218+6779Y+nsTfflrUH/7whwwePDi77rprunXrlt122y377bdfLr300iT6KaWlKf2xc+fO+eijj+o8Pnv27Hz88ccl//dfqF5Abdq0yQYbbJDRo0fXLquqqsro0aPTp0+fFqyMpVV1dXVOO+20jBo1KsOHD8/qq69e5/GePXtmmWWWqdNn33jjjbz77rvZaKONFnG1LI222GKL3H777bnttttq/+vZs2e+//3v1/6/PkpL6tu3b+11qjXeeuutrLbaakmSrl27pkuXLnX66LRp0/Lcc8/5288i83//938pKyurs6xVq1a1t9TSTyklTemPffr0ydSpU/Piiy/WtnniiSdSVVWV3r17L/Kam8Pw74Vg//33z5AhQ9KzZ8/07t07w4cPz4wZMzJgwICWLo2l0NChQ3PHHXfkT3/6U9q1a1d7DUqHDh2y3HLLpUOHDtljjz1y5plnpmPHjmnfvn1OP/309OnTR2BhkWjfvn3tNf412rZtmxVXXLF2uT5KS9pvv/2y995755JLLsnOO++c559/PiNHjsxpp52WJCkrK8u+++6bP//5z1lzzTXTtWvXXHDBBVlppZXSv3//Fq6epcV2222XSy65JKuuumrt8O+rr746e+yxRxL9lEXv008/zX//+9/af0+YMCHjxo1Lx44ds+qqq863P66zzjrZZpttctJJJ2Xo0KGZNWtWfvvb32bXXXfNyiuv3FJPq0nKqqu/cId4Crvuuuty5ZVXZtKkSenRo0d+85vfZMMNN2zpslgKffl+vzWGDRtWe6Dns88+y5lnnpk777wzM2fOzNZbb51TTjml5IfWsOTaZ5990r1795x44olJ9FFa3oMPPphzzz03b731Vrp27Zr9998/e+21V+3j1dXVufDCCzNy5MhMnTo1G2+8cU455ZSstdZaLVg1S5Np06blggsuyP3335/KysqstNJK2XXXXXPooYemTZs2SfRTFq0nn3wy++67b73lu+++e84888wm9ccpU6bkt7/9bR544IGUl5dnhx12yG9+85u0a9duUT6VZhOqAQAAoCDXVAMAAEBBQjUAAAAUJFQDAABAQUI1AAAAFCRUAwAAQEFCNQAAABQkVAMAAEBBQjUAAAAUJFQDAABAQUI1AAAAFCRUAwAAQEFCNQAAABT0/wAWLr1LPPFXnAAAAABJRU5ErkJggg==\n",
            "text/plain": [
              "<Figure size 1200x600 with 1 Axes>"
            ]
          },
          "metadata": {},
          "output_type": "display_data"
        }
      ],
      "source": [
        "sns.set_style(\"darkgrid\")\n",
        "plt.figure(figsize=[12, 6])\n",
        "plt.title(\"–†–∞—Å–ø—Ä–µ–¥–µ–ª–µ–Ω–∏–µ –º–µ—Ç–∏–ª–∏—Ä–æ–≤–∞–Ω–∏—è Epiblast\", fontsize=15)\n",
        "plt.hist(df[3], bins=100, density=True)\n",
        "plt.show()"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "fcS724jEncto",
        "outputId": "0857638e-0683-49c1-ca57-84f84df84a97"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "chr11\t3050031\t3050031\t50\t1\t1\n",
            "chr11\t3050048\t3050048\t66.6666666666667\t2\t1\n",
            "chr11\t3050049\t3050049\t50\t1\t1\n",
            "chr11\t3050224\t3050224\t87.5\t7\t1\n",
            "chr11\t3050225\t3050225\t53.3333333333333\t8\t7\n",
            "chr11\t3050286\t3050286\t66.6666666666667\t6\t3\n",
            "chr11\t3050287\t3050287\t33.3333333333333\t5\t10\n",
            "chr11\t3050310\t3050310\t62.5\t5\t3\n",
            "chr11\t3050311\t3050311\t50\t7\t7\n",
            "chr11\t3050349\t3050349\t55.5555555555556\t5\t4\n"
          ]
        }
      ],
      "source": [
        "! gzip -d s_8_cell.deduplicated.bismark.cov.gz\n",
        "! head s_8_cell.deduplicated.bismark.cov"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "PQw5vBYe9LGM",
        "outputId": "0cc58d81-0610-4a96-cd94-968db9ecc77e"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "chr11\t3050031\t3050031\t33.3333333333333\t1\t2\n",
            "chr11\t3050032\t3050032\t50\t3\t3\n",
            "chr11\t3050048\t3050048\t50\t2\t2\n",
            "chr11\t3050049\t3050049\t57.1428571428571\t4\t3\n",
            "chr11\t3050224\t3050224\t36.3636363636364\t4\t7\n",
            "chr11\t3050225\t3050225\t22.7272727272727\t5\t17\n",
            "chr11\t3050286\t3050286\t25\t2\t6\n",
            "chr11\t3050287\t3050287\t28.5714285714286\t6\t15\n",
            "chr11\t3050310\t3050310\t22.2222222222222\t2\t7\n",
            "chr11\t3050311\t3050311\t41.1764705882353\t7\t10\n"
          ]
        }
      ],
      "source": [
        "! gzip -d icm_cell.deduplicated.bismark.cov.gz\n",
        "! head icm_cell.deduplicated.bismark.cov"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "9xEqSQAq9Ukf",
        "outputId": "a788345c-faeb-4fb5-9516-35a2234247ee"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "chr11\t3050031\t3050031\t75\t3\t1\n",
            "chr11\t3050032\t3050032\t100\t4\t0\n",
            "chr11\t3050048\t3050048\t100\t5\t0\n",
            "chr11\t3050049\t3050049\t100\t9\t0\n",
            "chr11\t3050224\t3050224\t96.875\t31\t1\n",
            "chr11\t3050225\t3050225\t93.75\t30\t2\n",
            "chr11\t3050286\t3050286\t93.3333333333333\t28\t2\n",
            "chr11\t3050287\t3050287\t90\t36\t4\n",
            "chr11\t3050310\t3050310\t96.7741935483871\t30\t1\n",
            "chr11\t3050311\t3050311\t97.5609756097561\t40\t1\n"
          ]
        }
      ],
      "source": [
        "! gzip -d epiblast_cell.deduplicated.bismark.cov.gz\n",
        "! head epiblast_cell.deduplicated.bismark.cov"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 1000
        },
        "id": "b6k9REsM6NXn",
        "outputId": "badb96a6-9957-4cdc-90de-70e99026f50c"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "Collecting pyGenomeTracks\n",
            "  Downloading pyGenomeTracks-3.8-py2.py3-none-any.whl (113 kB)\n",
            "\u001b[2K     \u001b[90m‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ\u001b[0m \u001b[32m113.6/113.6 kB\u001b[0m \u001b[31m3.3 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hCollecting numpy>=1.20 (from pyGenomeTracks)\n",
            "  Downloading numpy-1.26.4-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (61 kB)\n",
            "\u001b[2K     \u001b[90m‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ\u001b[0m \u001b[32m61.0/61.0 kB\u001b[0m \u001b[31m5.6 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hCollecting matplotlib<=3.6.2,>=3.1.1 (from pyGenomeTracks)\n",
            "  Downloading matplotlib-3.6.2-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (11.8 MB)\n",
            "\u001b[2K     \u001b[90m‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ\u001b[0m \u001b[32m11.8/11.8 MB\u001b[0m \u001b[31m47.0 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hCollecting intervaltree>=2.1.0 (from pyGenomeTracks)\n",
            "  Downloading intervaltree-3.1.0.tar.gz (32 kB)\n",
            "  Preparing metadata (setup.py) ... \u001b[?25l\u001b[?25hdone\n",
            "Collecting pyBigWig>=0.3.16 (from pyGenomeTracks)\n",
            "  Downloading pyBigWig-0.3.22-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (209 kB)\n",
            "\u001b[2K     \u001b[90m‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ\u001b[0m \u001b[32m209.7/209.7 kB\u001b[0m \u001b[31m16.0 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hCollecting future>=0.17.0 (from pyGenomeTracks)\n",
            "  Downloading future-0.18.3.tar.gz (840 kB)\n",
            "\u001b[2K     \u001b[90m‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ\u001b[0m \u001b[32m840.9/840.9 kB\u001b[0m \u001b[31m33.8 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25h  Preparing metadata (setup.py) ... \u001b[?25l\u001b[?25hdone\n",
            "Collecting hicmatrix>=15 (from pyGenomeTracks)\n",
            "  Downloading HiCMatrix-17.1-py3-none-any.whl.metadata (2.0 kB)\n",
            "Collecting pysam>=0.14 (from pyGenomeTracks)\n",
            "  Downloading pysam-0.22.0-cp310-cp310-manylinux_2_28_x86_64.whl.metadata (1.5 kB)\n",
            "Collecting pytest (from pyGenomeTracks)\n",
            "  Downloading pytest-8.0.0-py3-none-any.whl.metadata (7.8 kB)\n",
            "Collecting gffutils>=0.9 (from pyGenomeTracks)\n",
            "  Downloading gffutils-0.12-py3-none-any.whl.metadata (1.5 kB)\n",
            "Collecting pybedtools>=0.8.1 (from pyGenomeTracks)\n",
            "  Downloading pybedtools-0.9.1.tar.gz (12.5 MB)\n",
            "\u001b[2K     \u001b[90m‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ\u001b[0m \u001b[32m12.5/12.5 MB\u001b[0m \u001b[31m38.7 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25h  Preparing metadata (setup.py) ... \u001b[?25l\u001b[?25hdone\n",
            "Requirement already satisfied: tqdm>=4.20 in /usr/local/lib/python3.10/site-packages (from pyGenomeTracks) (4.66.1)\n",
            "Collecting bx-python>=0.8.13 (from pyGenomeTracks)\n",
            "  Downloading bx_python-0.11.0-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (3.0 kB)\n",
            "Collecting pyfaidx>=0.1.3 (from pyGenomeTracks)\n",
            "  Downloading pyfaidx-0.8.1.1-py3-none-any.whl.metadata (25 kB)\n",
            "Collecting six>=1.12.0 (from gffutils>=0.9->pyGenomeTracks)\n",
            "  Downloading six-1.16.0-py2.py3-none-any.whl (11 kB)\n",
            "Collecting argh>=0.26.2 (from gffutils>=0.9->pyGenomeTracks)\n",
            "  Downloading argh-0.31.2-py3-none-any.whl.metadata (7.3 kB)\n",
            "Collecting argcomplete>=1.9.4 (from gffutils>=0.9->pyGenomeTracks)\n",
            "  Downloading argcomplete-3.2.2-py3-none-any.whl.metadata (16 kB)\n",
            "Collecting simplejson (from gffutils>=0.9->pyGenomeTracks)\n",
            "  Downloading simplejson-3.19.2-cp310-cp310-manylinux_2_5_x86_64.manylinux1_x86_64.manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (3.1 kB)\n",
            "Collecting scipy>=1.2 (from hicmatrix>=15->pyGenomeTracks)\n",
            "  Downloading scipy-1.12.0-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (60 kB)\n",
            "\u001b[2K     \u001b[90m‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ\u001b[0m \u001b[32m60.4/60.4 kB\u001b[0m \u001b[31m2.6 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hCollecting tables>=3.5 (from hicmatrix>=15->pyGenomeTracks)\n",
            "  Downloading tables-3.9.2-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (2.3 kB)\n",
            "Collecting pandas>=0.25 (from hicmatrix>=15->pyGenomeTracks)\n",
            "  Downloading pandas-2.2.0-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (19 kB)\n",
            "Collecting cooler>=0.8.9 (from hicmatrix>=15->pyGenomeTracks)\n",
            "  Downloading cooler-0.9.3-py3-none-any.whl.metadata (8.2 kB)\n",
            "Collecting sortedcontainers<3.0,>=2.0 (from intervaltree>=2.1.0->pyGenomeTracks)\n",
            "  Downloading sortedcontainers-2.4.0-py2.py3-none-any.whl (29 kB)\n",
            "Collecting contourpy>=1.0.1 (from matplotlib<=3.6.2,>=3.1.1->pyGenomeTracks)\n",
            "  Downloading contourpy-1.2.0-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (5.8 kB)\n",
            "Collecting cycler>=0.10 (from matplotlib<=3.6.2,>=3.1.1->pyGenomeTracks)\n",
            "  Downloading cycler-0.12.1-py3-none-any.whl.metadata (3.8 kB)\n",
            "Collecting fonttools>=4.22.0 (from matplotlib<=3.6.2,>=3.1.1->pyGenomeTracks)\n",
            "  Downloading fonttools-4.48.1-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (158 kB)\n",
            "\u001b[2K     \u001b[90m‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ\u001b[0m \u001b[32m158.9/158.9 kB\u001b[0m \u001b[31m14.2 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hCollecting kiwisolver>=1.0.1 (from matplotlib<=3.6.2,>=3.1.1->pyGenomeTracks)\n",
            "  Downloading kiwisolver-1.4.5-cp310-cp310-manylinux_2_12_x86_64.manylinux2010_x86_64.whl.metadata (6.4 kB)\n",
            "Requirement already satisfied: packaging>=20.0 in /usr/local/lib/python3.10/site-packages (from matplotlib<=3.6.2,>=3.1.1->pyGenomeTracks) (23.2)\n",
            "Collecting pillow>=6.2.0 (from matplotlib<=3.6.2,>=3.1.1->pyGenomeTracks)\n",
            "  Downloading pillow-10.2.0-cp310-cp310-manylinux_2_28_x86_64.whl.metadata (9.7 kB)\n",
            "Collecting pyparsing>=2.2.1 (from matplotlib<=3.6.2,>=3.1.1->pyGenomeTracks)\n",
            "  Downloading pyparsing-3.1.1-py3-none-any.whl.metadata (5.1 kB)\n",
            "Collecting python-dateutil>=2.7 (from matplotlib<=3.6.2,>=3.1.1->pyGenomeTracks)\n",
            "  Downloading python_dateutil-2.8.2-py2.py3-none-any.whl (247 kB)\n",
            "\u001b[2K     \u001b[90m‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ\u001b[0m \u001b[32m247.7/247.7 kB\u001b[0m \u001b[31m21.9 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hRequirement already satisfied: setuptools in /usr/local/lib/python3.10/site-packages (from pyfaidx>=0.1.3->pyGenomeTracks) (68.2.2)\n",
            "Collecting importlib-metadata (from pyfaidx>=0.1.3->pyGenomeTracks)\n",
            "  Downloading importlib_metadata-7.0.1-py3-none-any.whl.metadata (4.9 kB)\n",
            "Collecting iniconfig (from pytest->pyGenomeTracks)\n",
            "  Downloading iniconfig-2.0.0-py3-none-any.whl (5.9 kB)\n",
            "Requirement already satisfied: pluggy<2.0,>=1.3.0 in /usr/local/lib/python3.10/site-packages (from pytest->pyGenomeTracks) (1.3.0)\n",
            "Collecting exceptiongroup>=1.0.0rc8 (from pytest->pyGenomeTracks)\n",
            "  Downloading exceptiongroup-1.2.0-py3-none-any.whl.metadata (6.6 kB)\n",
            "Collecting tomli>=1.0.0 (from pytest->pyGenomeTracks)\n",
            "  Downloading tomli-2.0.1-py3-none-any.whl (12 kB)\n",
            "Collecting asciitree (from cooler>=0.8.9->hicmatrix>=15->pyGenomeTracks)\n",
            "  Downloading asciitree-0.3.3.tar.gz (4.0 kB)\n",
            "  Preparing metadata (setup.py) ... \u001b[?25l\u001b[?25hdone\n",
            "Collecting click>=7 (from cooler>=0.8.9->hicmatrix>=15->pyGenomeTracks)\n",
            "  Downloading click-8.1.7-py3-none-any.whl.metadata (3.0 kB)\n",
            "Collecting cytoolz (from cooler>=0.8.9->hicmatrix>=15->pyGenomeTracks)\n",
            "  Downloading cytoolz-0.12.3-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (4.6 kB)\n",
            "Collecting h5py>=2.5 (from cooler>=0.8.9->hicmatrix>=15->pyGenomeTracks)\n",
            "  Downloading h5py-3.10.0-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (2.5 kB)\n",
            "Collecting multiprocess (from cooler>=0.8.9->hicmatrix>=15->pyGenomeTracks)\n",
            "  Downloading multiprocess-0.70.16-py310-none-any.whl.metadata (7.2 kB)\n",
            "Collecting pyyaml (from cooler>=0.8.9->hicmatrix>=15->pyGenomeTracks)\n",
            "  Downloading PyYAML-6.0.1-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (2.1 kB)\n",
            "Collecting pytz>=2020.1 (from pandas>=0.25->hicmatrix>=15->pyGenomeTracks)\n",
            "  Downloading pytz-2024.1-py2.py3-none-any.whl.metadata (22 kB)\n",
            "Collecting tzdata>=2022.7 (from pandas>=0.25->hicmatrix>=15->pyGenomeTracks)\n",
            "  Downloading tzdata-2023.4-py2.py3-none-any.whl.metadata (1.4 kB)\n",
            "Collecting numexpr>=2.6.2 (from tables>=3.5->hicmatrix>=15->pyGenomeTracks)\n",
            "  Downloading numexpr-2.9.0-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (7.9 kB)\n",
            "Collecting py-cpuinfo (from tables>=3.5->hicmatrix>=15->pyGenomeTracks)\n",
            "  Downloading py_cpuinfo-9.0.0-py3-none-any.whl (22 kB)\n",
            "Collecting blosc2>=2.3.0 (from tables>=3.5->hicmatrix>=15->pyGenomeTracks)\n",
            "  Downloading blosc2-2.5.1-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (9.2 kB)\n",
            "Collecting zipp>=0.5 (from importlib-metadata->pyfaidx>=0.1.3->pyGenomeTracks)\n",
            "  Downloading zipp-3.17.0-py3-none-any.whl.metadata (3.7 kB)\n",
            "Collecting ndindex>=1.4 (from blosc2>=2.3.0->tables>=3.5->hicmatrix>=15->pyGenomeTracks)\n",
            "  Downloading ndindex-1.7-py3-none-any.whl (85 kB)\n",
            "\u001b[2K     \u001b[90m‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ\u001b[0m \u001b[32m85.7/85.7 kB\u001b[0m \u001b[31m7.2 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hCollecting msgpack (from blosc2>=2.3.0->tables>=3.5->hicmatrix>=15->pyGenomeTracks)\n",
            "  Downloading msgpack-1.0.7-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (9.1 kB)\n",
            "Collecting toolz>=0.8.0 (from cytoolz->cooler>=0.8.9->hicmatrix>=15->pyGenomeTracks)\n",
            "  Downloading toolz-0.12.1-py3-none-any.whl.metadata (5.1 kB)\n",
            "Collecting dill>=0.3.8 (from multiprocess->cooler>=0.8.9->hicmatrix>=15->pyGenomeTracks)\n",
            "  Downloading dill-0.3.8-py3-none-any.whl.metadata (10 kB)\n",
            "Downloading bx_python-0.11.0-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (5.6 MB)\n",
            "\u001b[2K   \u001b[90m‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ\u001b[0m \u001b[32m5.6/5.6 MB\u001b[0m \u001b[31m60.4 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading gffutils-0.12-py3-none-any.whl (1.6 MB)\n",
            "\u001b[2K   \u001b[90m‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ\u001b[0m \u001b[32m1.6/1.6 MB\u001b[0m \u001b[31m54.0 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading HiCMatrix-17.1-py3-none-any.whl (47 kB)\n",
            "\u001b[2K   \u001b[90m‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ\u001b[0m \u001b[32m47.4/47.4 kB\u001b[0m \u001b[31m4.2 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading numpy-1.26.4-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (18.2 MB)\n",
            "\u001b[2K   \u001b[90m‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ\u001b[0m \u001b[32m18.2/18.2 MB\u001b[0m \u001b[31m46.3 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading pyfaidx-0.8.1.1-py3-none-any.whl (28 kB)\n",
            "Downloading pysam-0.22.0-cp310-cp310-manylinux_2_28_x86_64.whl (21.9 MB)\n",
            "\u001b[2K   \u001b[90m‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ\u001b[0m \u001b[32m21.9/21.9 MB\u001b[0m \u001b[31m44.4 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading pytest-8.0.0-py3-none-any.whl (334 kB)\n",
            "\u001b[2K   \u001b[90m‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ\u001b[0m \u001b[32m334.0/334.0 kB\u001b[0m \u001b[31m24.7 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading argcomplete-3.2.2-py3-none-any.whl (42 kB)\n",
            "\u001b[2K   \u001b[90m‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ\u001b[0m \u001b[32m42.3/42.3 kB\u001b[0m \u001b[31m3.4 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading argh-0.31.2-py3-none-any.whl (44 kB)\n",
            "\u001b[2K   \u001b[90m‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ\u001b[0m \u001b[32m44.8/44.8 kB\u001b[0m \u001b[31m4.3 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading contourpy-1.2.0-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (310 kB)\n",
            "\u001b[2K   \u001b[90m‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ\u001b[0m \u001b[32m310.7/310.7 kB\u001b[0m \u001b[31m25.6 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading cooler-0.9.3-py3-none-any.whl (103 kB)\n",
            "\u001b[2K   \u001b[90m‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ\u001b[0m \u001b[32m103.7/103.7 kB\u001b[0m \u001b[31m10.4 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading cycler-0.12.1-py3-none-any.whl (8.3 kB)\n",
            "Downloading exceptiongroup-1.2.0-py3-none-any.whl (16 kB)\n",
            "Downloading fonttools-4.48.1-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (4.6 MB)\n",
            "\u001b[2K   \u001b[90m‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ\u001b[0m \u001b[32m4.6/4.6 MB\u001b[0m \u001b[31m86.6 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading kiwisolver-1.4.5-cp310-cp310-manylinux_2_12_x86_64.manylinux2010_x86_64.whl (1.6 MB)\n",
            "\u001b[2K   \u001b[90m‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ\u001b[0m \u001b[32m1.6/1.6 MB\u001b[0m \u001b[31m61.8 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading pandas-2.2.0-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (13.0 MB)\n",
            "\u001b[2K   \u001b[90m‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ\u001b[0m \u001b[32m13.0/13.0 MB\u001b[0m \u001b[31m57.4 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading pillow-10.2.0-cp310-cp310-manylinux_2_28_x86_64.whl (4.5 MB)\n",
            "\u001b[2K   \u001b[90m‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ\u001b[0m \u001b[32m4.5/4.5 MB\u001b[0m \u001b[31m56.7 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading pyparsing-3.1.1-py3-none-any.whl (103 kB)\n",
            "\u001b[2K   \u001b[90m‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ\u001b[0m \u001b[32m103.1/103.1 kB\u001b[0m \u001b[31m10.3 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading scipy-1.12.0-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (38.4 MB)\n",
            "\u001b[2K   \u001b[90m‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ\u001b[0m \u001b[32m38.4/38.4 MB\u001b[0m \u001b[31m10.4 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading tables-3.9.2-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (7.5 MB)\n",
            "\u001b[2K   \u001b[90m‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ\u001b[0m \u001b[32m7.5/7.5 MB\u001b[0m \u001b[31m93.9 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading importlib_metadata-7.0.1-py3-none-any.whl (23 kB)\n",
            "Downloading simplejson-3.19.2-cp310-cp310-manylinux_2_5_x86_64.manylinux1_x86_64.manylinux_2_17_x86_64.manylinux2014_x86_64.whl (137 kB)\n",
            "\u001b[2K   \u001b[90m‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ\u001b[0m \u001b[32m137.9/137.9 kB\u001b[0m \u001b[31m14.8 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading blosc2-2.5.1-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (4.4 MB)\n",
            "\u001b[2K   \u001b[90m‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ\u001b[0m \u001b[32m4.4/4.4 MB\u001b[0m \u001b[31m90.8 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading click-8.1.7-py3-none-any.whl (97 kB)\n",
            "\u001b[2K   \u001b[90m‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ\u001b[0m \u001b[32m97.9/97.9 kB\u001b[0m \u001b[31m11.0 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading h5py-3.10.0-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (4.8 MB)\n",
            "\u001b[2K   \u001b[90m‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ\u001b[0m \u001b[32m4.8/4.8 MB\u001b[0m \u001b[31m73.2 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading numexpr-2.9.0-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (375 kB)\n",
            "\u001b[2K   \u001b[90m‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ\u001b[0m \u001b[32m375.2/375.2 kB\u001b[0m \u001b[31m29.0 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading pytz-2024.1-py2.py3-none-any.whl (505 kB)\n",
            "\u001b[2K   \u001b[90m‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ\u001b[0m \u001b[32m505.5/505.5 kB\u001b[0m \u001b[31m37.1 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading tzdata-2023.4-py2.py3-none-any.whl (346 kB)\n",
            "\u001b[2K   \u001b[90m‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ\u001b[0m \u001b[32m346.6/346.6 kB\u001b[0m \u001b[31m25.4 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading zipp-3.17.0-py3-none-any.whl (7.4 kB)\n",
            "Downloading cytoolz-0.12.3-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (2.0 MB)\n",
            "\u001b[2K   \u001b[90m‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ\u001b[0m \u001b[32m2.0/2.0 MB\u001b[0m \u001b[31m59.1 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading multiprocess-0.70.16-py310-none-any.whl (134 kB)\n",
            "\u001b[2K   \u001b[90m‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ\u001b[0m \u001b[32m134.8/134.8 kB\u001b[0m \u001b[31m12.5 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading PyYAML-6.0.1-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (705 kB)\n",
            "\u001b[2K   \u001b[90m‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ\u001b[0m \u001b[32m705.5/705.5 kB\u001b[0m \u001b[31m40.3 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading dill-0.3.8-py3-none-any.whl (116 kB)\n",
            "\u001b[2K   \u001b[90m‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ\u001b[0m \u001b[32m116.3/116.3 kB\u001b[0m \u001b[31m11.8 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading toolz-0.12.1-py3-none-any.whl (56 kB)\n",
            "\u001b[2K   \u001b[90m‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ\u001b[0m \u001b[32m56.1/56.1 kB\u001b[0m \u001b[31m5.2 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading msgpack-1.0.7-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (530 kB)\n",
            "\u001b[2K   \u001b[90m‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ\u001b[0m \u001b[32m530.8/530.8 kB\u001b[0m \u001b[31m37.2 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hBuilding wheels for collected packages: future, intervaltree, pybedtools, asciitree\n",
            "  Building wheel for future (setup.py) ... \u001b[?25l\u001b[?25hdone\n",
            "  Created wheel for future: filename=future-0.18.3-py3-none-any.whl size=492024 sha256=d3b4504dcc467f1943c35a52bb66dbafde17397fcb6a79dcc091e626579beb1d\n",
            "  Stored in directory: /root/.cache/pip/wheels/5e/a9/47/f118e66afd12240e4662752cc22cefae5d97275623aa8ef57d\n",
            "  Building wheel for intervaltree (setup.py) ... \u001b[?25l\u001b[?25hdone\n",
            "  Created wheel for intervaltree: filename=intervaltree-3.1.0-py2.py3-none-any.whl size=26095 sha256=4f45ac386eafc84f18d434e99d736981362cbbf2236bff59fb184b4327a2889c\n",
            "  Stored in directory: /root/.cache/pip/wheels/fa/80/8c/43488a924a046b733b64de3fac99252674c892a4c3801c0a61\n",
            "  Building wheel for pybedtools (setup.py) ... \u001b[?25l\u001b[?25hdone\n",
            "  Created wheel for pybedtools: filename=pybedtools-0.9.1-cp310-cp310-linux_x86_64.whl size=12752426 sha256=7e6e95d44f9df9fa657cde05f0320a5692667410f05d7eec9f17fc9d6f3e15b9\n",
            "  Stored in directory: /root/.cache/pip/wheels/f5/79/76/c2cef660013cdf573214696aec4bc7ec352bf46141682230af\n",
            "  Building wheel for asciitree (setup.py) ... \u001b[?25l\u001b[?25hdone\n",
            "  Created wheel for asciitree: filename=asciitree-0.3.3-py3-none-any.whl size=5033 sha256=f0c0bae9c7deb57b527d6dc59ce350f3be6aa5626cdbcfe54ea4dbb13bc8a185\n",
            "  Stored in directory: /root/.cache/pip/wheels/7f/4e/be/1171b40f43b918087657ec57cf3b81fa1a2e027d8755baa184\n",
            "Successfully built future intervaltree pybedtools asciitree\n",
            "Installing collected packages: sortedcontainers, pytz, py-cpuinfo, asciitree, zipp, tzdata, toolz, tomli, six, simplejson, pyyaml, pysam, pyparsing, pyBigWig, pillow, numpy, ndindex, msgpack, kiwisolver, intervaltree, iniconfig, future, fonttools, exceptiongroup, dill, cycler, click, argh, argcomplete, scipy, python-dateutil, pytest, pybedtools, numexpr, multiprocess, importlib-metadata, h5py, cytoolz, contourpy, bx-python, blosc2, tables, pyfaidx, pandas, matplotlib, gffutils, cooler, hicmatrix, pyGenomeTracks\n",
            "Successfully installed argcomplete-3.2.2 argh-0.31.2 asciitree-0.3.3 blosc2-2.5.1 bx-python-0.11.0 click-8.1.7 contourpy-1.2.0 cooler-0.9.3 cycler-0.12.1 cytoolz-0.12.3 dill-0.3.8 exceptiongroup-1.2.0 fonttools-4.48.1 future-0.18.3 gffutils-0.12 h5py-3.10.0 hicmatrix-17.1 importlib-metadata-7.0.1 iniconfig-2.0.0 intervaltree-3.1.0 kiwisolver-1.4.5 matplotlib-3.6.2 msgpack-1.0.7 multiprocess-0.70.16 ndindex-1.7 numexpr-2.9.0 numpy-1.26.4 pandas-2.2.0 pillow-10.2.0 py-cpuinfo-9.0.0 pyBigWig-0.3.22 pyGenomeTracks-3.8 pybedtools-0.9.1 pyfaidx-0.8.1.1 pyparsing-3.1.1 pysam-0.22.0 pytest-8.0.0 python-dateutil-2.8.2 pytz-2024.1 pyyaml-6.0.1 scipy-1.12.0 simplejson-3.19.2 six-1.16.0 sortedcontainers-2.4.0 tables-3.9.2 tomli-2.0.1 toolz-0.12.1 tzdata-2023.4 zipp-3.17.0\n",
            "\u001b[33mWARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv\u001b[0m\u001b[33m\n",
            "\u001b[0m"
          ]
        },
        {
          "data": {
            "application/vnd.colab-display-data+json": {
              "pip_warning": {
                "packages": [
                  "cycler",
                  "dateutil",
                  "kiwisolver",
                  "numexpr",
                  "six"
                ]
              }
            }
          },
          "metadata": {},
          "output_type": "display_data"
        }
      ],
      "source": [
        "! pip install pyGenomeTracks"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 1000
        },
        "id": "fcsJWNmHo1i-",
        "outputId": "cb74241c-2ae5-4648-e1d6-a8f557d4809e"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "Collecting ipykernel\n",
            "  Downloading ipykernel-6.29.2-py3-none-any.whl.metadata (6.0 kB)\n",
            "Collecting comm>=0.1.1 (from ipykernel)\n",
            "  Downloading comm-0.2.1-py3-none-any.whl.metadata (3.7 kB)\n",
            "Collecting debugpy>=1.6.5 (from ipykernel)\n",
            "  Downloading debugpy-1.8.1-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (1.1 kB)\n",
            "Collecting ipython>=7.23.1 (from ipykernel)\n",
            "  Downloading ipython-8.21.0-py3-none-any.whl.metadata (5.9 kB)\n",
            "Collecting jupyter-client>=6.1.12 (from ipykernel)\n",
            "  Downloading jupyter_client-8.6.0-py3-none-any.whl.metadata (8.3 kB)\n",
            "Collecting jupyter-core!=5.0.*,>=4.12 (from ipykernel)\n",
            "  Downloading jupyter_core-5.7.1-py3-none-any.whl.metadata (3.4 kB)\n",
            "Collecting matplotlib-inline>=0.1 (from ipykernel)\n",
            "  Downloading matplotlib_inline-0.1.6-py3-none-any.whl (9.4 kB)\n",
            "Collecting nest-asyncio (from ipykernel)\n",
            "  Downloading nest_asyncio-1.6.0-py3-none-any.whl.metadata (2.8 kB)\n",
            "Requirement already satisfied: packaging in /usr/local/lib/python3.10/site-packages (from ipykernel) (23.2)\n",
            "Collecting psutil (from ipykernel)\n",
            "  Downloading psutil-5.9.8-cp36-abi3-manylinux_2_12_x86_64.manylinux2010_x86_64.manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (21 kB)\n",
            "Collecting pyzmq>=24 (from ipykernel)\n",
            "  Downloading pyzmq-25.1.2-cp310-cp310-manylinux_2_28_x86_64.whl.metadata (4.9 kB)\n",
            "Collecting tornado>=6.1 (from ipykernel)\n",
            "  Downloading tornado-6.4-cp38-abi3-manylinux_2_5_x86_64.manylinux1_x86_64.manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (2.5 kB)\n",
            "Collecting traitlets>=5.4.0 (from ipykernel)\n",
            "  Downloading traitlets-5.14.1-py3-none-any.whl.metadata (10 kB)\n",
            "Collecting decorator (from ipython>=7.23.1->ipykernel)\n",
            "  Downloading decorator-5.1.1-py3-none-any.whl (9.1 kB)\n",
            "Collecting jedi>=0.16 (from ipython>=7.23.1->ipykernel)\n",
            "  Downloading jedi-0.19.1-py2.py3-none-any.whl.metadata (22 kB)\n",
            "Collecting prompt-toolkit<3.1.0,>=3.0.41 (from ipython>=7.23.1->ipykernel)\n",
            "  Downloading prompt_toolkit-3.0.43-py3-none-any.whl.metadata (6.5 kB)\n",
            "Collecting pygments>=2.4.0 (from ipython>=7.23.1->ipykernel)\n",
            "  Downloading pygments-2.17.2-py3-none-any.whl.metadata (2.6 kB)\n",
            "Collecting stack-data (from ipython>=7.23.1->ipykernel)\n",
            "  Downloading stack_data-0.6.3-py3-none-any.whl.metadata (18 kB)\n",
            "Requirement already satisfied: exceptiongroup in /usr/local/lib/python3.10/site-packages (from ipython>=7.23.1->ipykernel) (1.2.0)\n",
            "Collecting pexpect>4.3 (from ipython>=7.23.1->ipykernel)\n",
            "  Downloading pexpect-4.9.0-py2.py3-none-any.whl.metadata (2.5 kB)\n",
            "Requirement already satisfied: python-dateutil>=2.8.2 in /usr/local/lib/python3.10/site-packages (from jupyter-client>=6.1.12->ipykernel) (2.8.2)\n",
            "Requirement already satisfied: platformdirs>=2.5 in /usr/local/lib/python3.10/site-packages (from jupyter-core!=5.0.*,>=4.12->ipykernel) (4.1.0)\n",
            "Collecting parso<0.9.0,>=0.8.3 (from jedi>=0.16->ipython>=7.23.1->ipykernel)\n",
            "  Downloading parso-0.8.3-py2.py3-none-any.whl (100 kB)\n",
            "\u001b[2K     \u001b[90m‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ\u001b[0m \u001b[32m100.8/100.8 kB\u001b[0m \u001b[31m4.2 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hCollecting ptyprocess>=0.5 (from pexpect>4.3->ipython>=7.23.1->ipykernel)\n",
            "  Downloading ptyprocess-0.7.0-py2.py3-none-any.whl (13 kB)\n",
            "Collecting wcwidth (from prompt-toolkit<3.1.0,>=3.0.41->ipython>=7.23.1->ipykernel)\n",
            "  Downloading wcwidth-0.2.13-py2.py3-none-any.whl.metadata (14 kB)\n",
            "Requirement already satisfied: six>=1.5 in /usr/local/lib/python3.10/site-packages (from python-dateutil>=2.8.2->jupyter-client>=6.1.12->ipykernel) (1.16.0)\n",
            "Collecting executing>=1.2.0 (from stack-data->ipython>=7.23.1->ipykernel)\n",
            "  Downloading executing-2.0.1-py2.py3-none-any.whl.metadata (9.0 kB)\n",
            "Collecting asttokens>=2.1.0 (from stack-data->ipython>=7.23.1->ipykernel)\n",
            "  Downloading asttokens-2.4.1-py2.py3-none-any.whl.metadata (5.2 kB)\n",
            "Collecting pure-eval (from stack-data->ipython>=7.23.1->ipykernel)\n",
            "  Downloading pure_eval-0.2.2-py3-none-any.whl (11 kB)\n",
            "Downloading ipykernel-6.29.2-py3-none-any.whl (116 kB)\n",
            "\u001b[2K   \u001b[90m‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ\u001b[0m \u001b[32m116.1/116.1 kB\u001b[0m \u001b[31m12.3 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading comm-0.2.1-py3-none-any.whl (7.2 kB)\n",
            "Downloading debugpy-1.8.1-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (3.0 MB)\n",
            "\u001b[2K   \u001b[90m‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ\u001b[0m \u001b[32m3.0/3.0 MB\u001b[0m \u001b[31m48.0 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading ipython-8.21.0-py3-none-any.whl (810 kB)\n",
            "\u001b[2K   \u001b[90m‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ\u001b[0m \u001b[32m810.0/810.0 kB\u001b[0m \u001b[31m54.1 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading jupyter_client-8.6.0-py3-none-any.whl (105 kB)\n",
            "\u001b[2K   \u001b[90m‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ\u001b[0m \u001b[32m105.9/105.9 kB\u001b[0m \u001b[31m11.5 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading jupyter_core-5.7.1-py3-none-any.whl (28 kB)\n",
            "Downloading pyzmq-25.1.2-cp310-cp310-manylinux_2_28_x86_64.whl (1.1 MB)\n",
            "\u001b[2K   \u001b[90m‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ\u001b[0m \u001b[32m1.1/1.1 MB\u001b[0m \u001b[31m48.4 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading tornado-6.4-cp38-abi3-manylinux_2_5_x86_64.manylinux1_x86_64.manylinux_2_17_x86_64.manylinux2014_x86_64.whl (435 kB)\n",
            "\u001b[2K   \u001b[90m‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ\u001b[0m \u001b[32m435.4/435.4 kB\u001b[0m \u001b[31m37.2 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading traitlets-5.14.1-py3-none-any.whl (85 kB)\n",
            "\u001b[2K   \u001b[90m‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ\u001b[0m \u001b[32m85.4/85.4 kB\u001b[0m \u001b[31m9.0 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading nest_asyncio-1.6.0-py3-none-any.whl (5.2 kB)\n",
            "Downloading psutil-5.9.8-cp36-abi3-manylinux_2_12_x86_64.manylinux2010_x86_64.manylinux_2_17_x86_64.manylinux2014_x86_64.whl (288 kB)\n",
            "\u001b[2K   \u001b[90m‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ\u001b[0m \u001b[32m288.2/288.2 kB\u001b[0m \u001b[31m23.8 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading jedi-0.19.1-py2.py3-none-any.whl (1.6 MB)\n",
            "\u001b[2K   \u001b[90m‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ\u001b[0m \u001b[32m1.6/1.6 MB\u001b[0m \u001b[31m48.2 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading pexpect-4.9.0-py2.py3-none-any.whl (63 kB)\n",
            "\u001b[2K   \u001b[90m‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ\u001b[0m \u001b[32m63.8/63.8 kB\u001b[0m \u001b[31m6.7 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading prompt_toolkit-3.0.43-py3-none-any.whl (386 kB)\n",
            "\u001b[2K   \u001b[90m‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ\u001b[0m \u001b[32m386.1/386.1 kB\u001b[0m \u001b[31m32.9 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading pygments-2.17.2-py3-none-any.whl (1.2 MB)\n",
            "\u001b[2K   \u001b[90m‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ\u001b[0m \u001b[32m1.2/1.2 MB\u001b[0m \u001b[31m47.5 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading stack_data-0.6.3-py3-none-any.whl (24 kB)\n",
            "Downloading asttokens-2.4.1-py2.py3-none-any.whl (27 kB)\n",
            "Downloading executing-2.0.1-py2.py3-none-any.whl (24 kB)\n",
            "Downloading wcwidth-0.2.13-py2.py3-none-any.whl (34 kB)\n",
            "Installing collected packages: wcwidth, pure-eval, ptyprocess, traitlets, tornado, pyzmq, pygments, psutil, prompt-toolkit, pexpect, parso, nest-asyncio, executing, decorator, debugpy, asttokens, stack-data, matplotlib-inline, jupyter-core, jedi, comm, jupyter-client, ipython, ipykernel\n",
            "Successfully installed asttokens-2.4.1 comm-0.2.1 debugpy-1.8.1 decorator-5.1.1 executing-2.0.1 ipykernel-6.29.2 ipython-8.21.0 jedi-0.19.1 jupyter-client-8.6.0 jupyter-core-5.7.1 matplotlib-inline-0.1.6 nest-asyncio-1.6.0 parso-0.8.3 pexpect-4.9.0 prompt-toolkit-3.0.43 psutil-5.9.8 ptyprocess-0.7.0 pure-eval-0.2.2 pygments-2.17.2 pyzmq-25.1.2 stack-data-0.6.3 tornado-6.4 traitlets-5.14.1 wcwidth-0.2.13\n",
            "\u001b[33mWARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv\u001b[0m\u001b[33m\n",
            "\u001b[0m"
          ]
        },
        {
          "data": {
            "application/vnd.colab-display-data+json": {
              "pip_warning": {
                "packages": [
                  "matplotlib_inline",
                  "pexpect",
                  "prompt_toolkit",
                  "wcwidth"
                ]
              }
            }
          },
          "metadata": {},
          "output_type": "display_data"
        }
      ],
      "source": [
        "!pip install -U ipykernel"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "UhIKyKz_t7yy",
        "outputId": "6769dd8c-a26b-40c9-cea3-2a5383444c05"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "Channels:\n",
            " - bioconda\n",
            " - conda-forge\n",
            "Platform: linux-64\n",
            "Collecting package metadata (repodata.json): - \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\bdone\n",
            "Solving environment: / \b\b- \b\b\\ \b\b| \u001b[33m\u001b[1mwarning  libmamba\u001b[m Added empty dependency for problem type SOLVER_RULE_UPDATE\n",
            "\b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\bfailed\n",
            "\n",
            "LibMambaUnsatisfiableError: Encountered problems while solving:\n",
            "  - package ucsc-fetchchromsizes-324-0 requires openssl >=1.1.0,<=1.1.1, but none of the providers can be installed\n",
            "\n",
            "Could not solve for environment specs\n",
            "The following packages are incompatible\n",
            "‚îú‚îÄ \u001b[32mmamba >=1.5.5 \u001b[0m is installable with the potential options\n",
            "‚îÇ  ‚îú‚îÄ \u001b[32mmamba [1.5.5|1.5.6]\u001b[0m would require\n",
            "‚îÇ  ‚îÇ  ‚îú‚îÄ \u001b[32mlibmambapy 1.5.6 py310h39ff949_0\u001b[0m, which requires\n",
            "‚îÇ  ‚îÇ  ‚îÇ  ‚îî‚îÄ \u001b[32mlibmamba 1.5.6 had39da4_0\u001b[0m, which requires\n",
            "‚îÇ  ‚îÇ  ‚îÇ     ‚îî‚îÄ \u001b[32mopenssl >=3.2.0,<4.0a0 \u001b[0m, which can be installed;\n",
            "‚îÇ  ‚îÇ  ‚îî‚îÄ \u001b[32mopenssl >=3.2.0,<4.0a0 \u001b[0m, which can be installed;\n",
            "‚îÇ  ‚îú‚îÄ \u001b[32mmamba [1.5.5|1.5.6]\u001b[0m would require\n",
            "‚îÇ  ‚îÇ  ‚îî‚îÄ \u001b[32mpython >=3.11,<3.12.0a0 \u001b[0m, which can be installed;\n",
            "‚îÇ  ‚îú‚îÄ \u001b[32mmamba [1.5.5|1.5.6]\u001b[0m would require\n",
            "‚îÇ  ‚îÇ  ‚îî‚îÄ \u001b[32mpython >=3.12,<3.13.0a0 \u001b[0m, which can be installed;\n",
            "‚îÇ  ‚îú‚îÄ \u001b[32mmamba [1.5.5|1.5.6]\u001b[0m would require\n",
            "‚îÇ  ‚îÇ  ‚îî‚îÄ \u001b[32mpython >=3.8,<3.9.0a0 \u001b[0m, which can be installed;\n",
            "‚îÇ  ‚îî‚îÄ \u001b[32mmamba [1.5.5|1.5.6]\u001b[0m would require\n",
            "‚îÇ     ‚îî‚îÄ \u001b[32mpython >=3.9,<3.10.0a0 \u001b[0m, which can be installed;\n",
            "‚îú‚îÄ \u001b[31mpin-2\u001b[0m is not installable because it requires\n",
            "‚îÇ  ‚îî‚îÄ \u001b[31mpython 3.10** \u001b[0m, which conflicts with any installable versions previously reported;\n",
            "‚îî‚îÄ \u001b[31mucsc-fetchchromsizes\u001b[0m is not installable because there are no viable options\n",
            "   ‚îú‚îÄ \u001b[31mucsc-fetchchromsizes [324|332|357]\u001b[0m would require\n",
            "   ‚îÇ  ‚îî‚îÄ \u001b[31mopenssl >=1.1.0,<=1.1.1 \u001b[0m, which conflicts with any installable versions previously reported;\n",
            "   ‚îú‚îÄ \u001b[31mucsc-fetchchromsizes 366\u001b[0m would require\n",
            "   ‚îÇ  ‚îî‚îÄ \u001b[31mopenssl >=1.0.2o,<1.0.3a \u001b[0m, which conflicts with any installable versions previously reported;\n",
            "   ‚îú‚îÄ \u001b[31mucsc-fetchchromsizes [366|377]\u001b[0m would require\n",
            "   ‚îÇ  ‚îî‚îÄ \u001b[31mopenssl >=1.0.2p,<1.0.3a \u001b[0m, which conflicts with any installable versions previously reported;\n",
            "   ‚îú‚îÄ \u001b[31mucsc-fetchchromsizes 377\u001b[0m would require\n",
            "   ‚îÇ  ‚îî‚îÄ \u001b[31mopenssl >=1.1.1j,<1.1.2a \u001b[0m, which conflicts with any installable versions previously reported;\n",
            "   ‚îú‚îÄ \u001b[31mucsc-fetchchromsizes 377\u001b[0m would require\n",
            "   ‚îÇ  ‚îî‚îÄ \u001b[31mopenssl >=1.1.1g,<1.1.2a \u001b[0m, which conflicts with any installable versions previously reported;\n",
            "   ‚îî‚îÄ \u001b[31mucsc-fetchchromsizes 377\u001b[0m would require\n",
            "      ‚îî‚îÄ \u001b[31mopenssl >=1.1.1l,<1.1.2a \u001b[0m, which conflicts with any installable versions previously reported.\n",
            "\n"
          ]
        }
      ],
      "source": [
        "! conda install -c bioconda ucsc-fetchchromsizes"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "LZpspVUvt4EY",
        "outputId": "a57c25ce-c9d4-4d69-c58e-9f24e85a67d2"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "/bin/bash: line 1: fetchChromSizes: command not found\n"
          ]
        }
      ],
      "source": [
        "# ! fetchChromSizes mm10 > m.chrom.sizes"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "2eoX34Q6IrdW",
        "outputId": "80e342b3-6f2c-4992-b9b7-c9f5ce18000e"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "–§–∞–π–ª 'm.chrom.sizes' —É—Å–ø–µ—à–Ω–æ —Å–æ—Ö—Ä–∞–Ω–µ–Ω.\n"
          ]
        }
      ],
      "source": [
        "import requests\n",
        "\n",
        "\n",
        "url = 'http://hgdownload.cse.ucsc.edu/goldenPath/mm10/bigZips/mm10.chrom.sizes'\n",
        "\n",
        "response = requests.get(url)\n",
        "\n",
        "with open(\"m.chrom.sizes\", \"wb\") as f:\n",
        "    f.write(response.content)\n",
        "print(\"–§–∞–π–ª 'm.chrom.sizes' —É—Å–ø–µ—à–Ω–æ —Å–æ—Ö—Ä–∞–Ω–µ–Ω.\")"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "svn9iAuGtVxX",
        "outputId": "e731a297-24fa-4315-901e-f355e8fea126"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "Channels:\n",
            " - bioconda\n",
            " - conda-forge\n",
            "Platform: linux-64\n",
            "Collecting package metadata (repodata.json): - \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\bdone\n",
            "Solving environment: | \b\b/ \b\b- \b\b\\ \b\bdone\n",
            "\n",
            "# All requested packages already installed.\n",
            "\n"
          ]
        }
      ],
      "source": [
        "! conda install -c bioconda ucsc-bedgraphtobigwig"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "rh0ba_Y_tcq5"
      },
      "outputs": [],
      "source": [
        "! bedGraphToBigWig  s_8_cell.deduplicated.bedGraph m.chrom.sizes cell8_methylation.bigWig"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "qofYnYciunf8",
        "outputId": "40e3f5bc-c200-42db-a55d-3b59eed9be67"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "track type=bedGraph\n",
            "chr11\t3050030\t3050031\t50\n",
            "chr11\t3050047\t3050048\t66.6666666666667\n",
            "chr11\t3050048\t3050049\t50\n",
            "chr11\t3050223\t3050224\t87.5\n",
            "chr11\t3050224\t3050225\t53.3333333333333\n",
            "chr11\t3050285\t3050286\t66.6666666666667\n",
            "chr11\t3050286\t3050287\t33.3333333333333\n",
            "chr11\t3050309\t3050310\t62.5\n",
            "chr11\t3050310\t3050311\t50\n"
          ]
        }
      ],
      "source": [
        "! head s_8_cell.deduplicated.bedGraph"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "XEWV8p4ku-Hy",
        "outputId": "d8d651ed-90d5-40a1-8442-6c71aad4e143"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "Reading package lists... Done\n",
            "Building dependency tree... Done\n",
            "Reading state information... Done\n",
            "The following NEW packages will be installed:\n",
            "  bedtools\n",
            "0 upgraded, 1 newly installed, 0 to remove and 32 not upgraded.\n",
            "Need to get 563 kB of archives.\n",
            "After this operation, 1,548 kB of additional disk space will be used.\n",
            "Get:1 http://archive.ubuntu.com/ubuntu jammy-updates/universe amd64 bedtools amd64 2.30.0+dfsg-2ubuntu0.1 [563 kB]\n",
            "Fetched 563 kB in 1s (1,090 kB/s)\n",
            "Selecting previously unselected package bedtools.\n",
            "(Reading database ... 121749 files and directories currently installed.)\n",
            "Preparing to unpack .../bedtools_2.30.0+dfsg-2ubuntu0.1_amd64.deb ...\n",
            "Unpacking bedtools (2.30.0+dfsg-2ubuntu0.1) ...\n",
            "Setting up bedtools (2.30.0+dfsg-2ubuntu0.1) ...\n"
          ]
        }
      ],
      "source": [
        "! apt-get install bedtools"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "ofhSsnSqvBbu",
        "outputId": "4fba31de-ca72-4f7d-c0a4-26255df590b9"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "\n",
            "*****\n",
            "*****WARNING: Genome (-g) files are ignored when BAM input is provided. \n",
            "*****\n"
          ]
        }
      ],
      "source": [
        "! bedtools genomecov -bg -ibam s_8_cell.deduplicated.bam -g m.chromSizes  >  s_8_cell.cover.bedGraph"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "N-8RwloKJxuL",
        "outputId": "8bfb0ee1-a20b-4032-8501-1de008a1f7b0"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "\n",
            "*****\n",
            "*****WARNING: Genome (-g) files are ignored when BAM input is provided. \n",
            "*****\n"
          ]
        }
      ],
      "source": [
        "! bedtools genomecov -bg -ibam icm_cell.deduplicated.bam -g m.chromSizes  >  icm_cell.cover.bedGraph"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "3lzJAYQMKDIG",
        "outputId": "887598d0-d82d-4a80-be9f-9504d0c6a778"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "\n",
            "*****\n",
            "*****WARNING: Genome (-g) files are ignored when BAM input is provided. \n",
            "*****\n"
          ]
        }
      ],
      "source": [
        "! bedtools genomecov -bg -ibam epiblast_cell.deduplicated.bam -g m.chromSizes  >  epiblast_cell.cover.bedGraph"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "RLpT4wFCvr29"
      },
      "outputs": [],
      "source": [
        "! bedGraphToBigWig   s_8_cell.cover.bedGraph m.chrom.sizes cell8_coverage.bigWig.bw"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "PT489lPDKZNt"
      },
      "outputs": [],
      "source": [
        "! bedGraphToBigWig   icm_cell.cover.bedGraph m.chrom.sizes icm_coverage.bigWig.bw"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "fMuGL2DFKhWg"
      },
      "outputs": [],
      "source": [
        "! bedGraphToBigWig   epiblast_cell.cover.bedGraph m.chrom.sizes epiblast_coverage.bigWig.bw"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "0egk8Q4VvXss",
        "outputId": "1665e215-13f5-4d5b-d519-24f78ee6f2c9"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "track type=bedGraph\n",
            "chr11\t3050030\t3050031\t50\n",
            "chr11\t3050047\t3050048\t66.6666666666667\n",
            "chr11\t3050048\t3050049\t50\n",
            "chr11\t3050223\t3050224\t87.5\n",
            "chr11\t3050224\t3050225\t53.3333333333333\n",
            "chr11\t3050285\t3050286\t66.6666666666667\n",
            "chr11\t3050286\t3050287\t33.3333333333333\n",
            "chr11\t3050309\t3050310\t62.5\n",
            "chr11\t3050310\t3050311\t50\n"
          ]
        }
      ],
      "source": [
        "! head s_8_cell.deduplicated.bedGraph"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "rAoU3fjzLdYX",
        "outputId": "23022722-2db7-4f47-959f-e273524a37e5"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "Adding bigwig file: cell8_coverage.bigWig.bw\n",
            "Adding bigwig file: epiblast_coverage.bigWig.bw\n",
            "Adding bigwig file: icm_coverage.bigWig.bw\n"
          ]
        }
      ],
      "source": [
        "! make_tracks_file --trackFiles cell8_coverage.bigWig.bw epiblast_coverage.bigWig.bw icm_coverage.bigWig.bw -o tracks.ini.txt"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "bAaOdZCKsVtX",
        "outputId": "fe9cffff-f9f3-4147-a9c8-573458700f1b"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "INFO:pygenometracks.tracksClass:initialize 1. [x-axis]\n",
            "INFO:pygenometracks.tracksClass:initialize 2. [spacer]\n",
            "INFO:pygenometracks.tracksClass:initialize 3. [cell8_coverage.bigWig]\n",
            "INFO:pygenometracks.tracksClass:initialize 4. [epiblast_coverage.bigWig]\n",
            "INFO:pygenometracks.tracksClass:initialize 5. [icm_coverage.bigWig]\n",
            "INFO:pygenometracks.tracksClass:time initializing track(s):\n",
            "INFO:pygenometracks.tracksClass:0.01100778579711914\n",
            "DEBUG:pygenometracks.tracksClass:Figure size in cm is 40 x 8.909574468085108. Dpi is set to 72\n",
            "\n",
            "INFO:pygenometracks.tracksClass:plotting 1. [x-axis]\n",
            "INFO:pygenometracks.tracksClass:plotting 2. [spacer]\n",
            "INFO:pygenometracks.tracksClass:plotting 3. [cell8_coverage.bigWig]\n",
            "INFO:pygenometracks.tracksClass:plotting 4. [epiblast_coverage.bigWig]\n",
            "INFO:pygenometracks.tracksClass:plotting 5. [icm_coverage.bigWig]\n"
          ]
        }
      ],
      "source": [
        "! pyGenomeTracks --tracks tracks.ini.txt --region chr11:3100030-3500030 -o image_cov.png"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "p7CjvnFhwpyL"
      },
      "outputs": [],
      "source": [
        "! bedGraphToBigWig   s_8_cell.deduplicated.bedGraph m.chrom.sizes cell8_methylation.bigWig.bw\n",
        "! bedGraphToBigWig   icm_cell.deduplicated.bedGraph m.chrom.sizes icm_methylation.bigWig.bw\n",
        "! bedGraphToBigWig   epiblast_cell.deduplicated.bedGraph m.chrom.sizes epiblast_methylation.bigWig.bw"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "z8Qceld3Pvv6",
        "outputId": "f131fa79-d9ae-40a0-b44a-bcca708d5b5f"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "Adding bigwig file: cell8_methylation.bigWig.bw\n",
            "Adding bigwig file: epiblast_methylation.bigWig.bw\n",
            "Adding bigwig file: icm_methylation.bigWig.bw\n"
          ]
        }
      ],
      "source": [
        "! make_tracks_file --trackFiles cell8_methylation.bigWig.bw epiblast_methylation.bigWig.bw icm_methylation.bigWig.bw -o tracks_met.ini.txt"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "vBzTYSeGQFwS",
        "outputId": "3aefba4d-0e80-4181-e89c-cbb2dd868324"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "INFO:pygenometracks.tracksClass:initialize 1. [x-axis]\n",
            "INFO:pygenometracks.tracksClass:initialize 2. [spacer]\n",
            "INFO:pygenometracks.tracksClass:initialize 3. [cell8_methylation.bigWig]\n",
            "INFO:pygenometracks.tracksClass:initialize 4. [epiblast_methylation.bigWig]\n",
            "INFO:pygenometracks.tracksClass:initialize 5. [icm_methylation.bigWig]\n",
            "INFO:pygenometracks.tracksClass:time initializing track(s):\n",
            "INFO:pygenometracks.tracksClass:0.004153251647949219\n",
            "DEBUG:pygenometracks.tracksClass:Figure size in cm is 40 x 8.909574468085108. Dpi is set to 72\n",
            "\n",
            "INFO:pygenometracks.tracksClass:plotting 1. [x-axis]\n",
            "INFO:pygenometracks.tracksClass:plotting 2. [spacer]\n",
            "INFO:pygenometracks.tracksClass:plotting 3. [cell8_methylation.bigWig]\n",
            "INFO:pygenometracks.tracksClass:plotting 4. [epiblast_methylation.bigWig]\n",
            "INFO:pygenometracks.tracksClass:plotting 5. [icm_methylation.bigWig]\n"
          ]
        }
      ],
      "source": [
        "! pyGenomeTracks --tracks tracks_met.ini.txt --region chr11:3100030-3500030 -o image_met.png"
      ]
    }
  ],
  "metadata": {
    "colab": {
      "provenance": []
    },
    "kernelspec": {
      "display_name": "Python 3",
      "name": "python3"
    },
    "language_info": {
      "name": "python"
    }
  },
  "nbformat": 4,
  "nbformat_minor": 0
}